{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "599c3288-b0bd-485c-824a-ce8029366466",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sb\n",
    "import numpy as np\n",
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=pd.errors.PerformanceWarning)\n",
    "from scipy.stats import chisquare\n",
    "from sklearn.linear_model import LinearRegression\n",
    "import statsmodels.api as sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "7ec97927-86fc-43e2-8f99-97f66d3a8281",
   "metadata": {},
   "outputs": [],
   "source": [
    "dir_fig = 'fragile_x_tables/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d0d6cacd-dc1d-46ef-8c84-685899205b10",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>region_id</th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>bp_dev_h1</th>\n",
       "      <th>bp_dev_h2</th>\n",
       "      <th>bp_dev_h0</th>\n",
       "      <th>rn_h1</th>\n",
       "      <th>rn_h2</th>\n",
       "      <th>rn_h0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147911920</td>\n",
       "      <td>147912179</td>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>.;.;.;91|95|100|99|96|100|99|98|98|96|96|97|94...</td>\n",
       "      <td>.;.;.;64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|...</td>\n",
       "      <td>.;0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3...</td>\n",
       "      <td>.;.;.;32;.;.;30;.;31,30;34;27,27,27,29;.;.;.;3...</td>\n",
       "      <td>.;.;.;31,43,30;.;.;30,28,28,31;.;30,30,32;34,2...</td>\n",
       "      <td>.;51,51,51,57,51,51,52;30;.;.;30;.;34,31;.;.;....</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/27724288/ccs;.;.;m...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/75497573/ccs,m5434...</td>\n",
       "      <td>.;m54340U_210523_044800/52758275/ccs,m54340U_2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149505139</td>\n",
       "      <td>149505306</td>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>.;.;.;8|2|2|1|0|0|0|0|0|1|0|7|1|0|3|2|1|4|0|0|...</td>\n",
       "      <td>.;.;.;62|51|80|91|75|95|89|98|99|98|97|99|99|9...</td>\n",
       "      <td>.;0|2|0|1|0|1|6|2|1|8|9|13|6|3|4|8|5|0|1|0|0|0...</td>\n",
       "      <td>.;.;.;1;.;.;.;.;.;2;.;.;.;.;0,0,2;.;.;-1,1,0;....</td>\n",
       "      <td>.;.;.;0;.;.;.;.;.;.;.;.;.;.;.;.;.;1,0,1;.;.;.;...</td>\n",
       "      <td>.;0,2;3;.;0,0;0;-1,1,0,1;0,-2,0;0,0,-1,0;0,0,1...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/128059025/ccs;.;.;...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/133105829/ccs;.;.;...</td>\n",
       "      <td>.;m54340U_200806_070905/88277013/ccs,m54340U_2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149631582</td>\n",
       "      <td>149631792</td>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;95|91|35|70|18|12|12|62|41|61|26|6...</td>\n",
       "      <td>.;.;.;.;73|60|29|4|2|7|3|1|1|1|1|0|0|0|8|17|11...</td>\n",
       "      <td>5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...</td>\n",
       "      <td>.;.;.;.;.;.;-21,-21,-21;.;-21,-15;.;15,9;.;0;....</td>\n",
       "      <td>.;.;.;.;-21;.;.,9,0;.;-21,-21;.;-21;.;-11;.;.;...</td>\n",
       "      <td>-21,-19;-21,-21,-21,-21,-18,-21,-21,-25;.;-19;...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_210703_193517/110166254/cc...</td>\n",
       "      <td>.;.;.;.;m54340U_200716_153006/94110941/ccs;.;m...</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs,m54340U_20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150363318</td>\n",
       "      <td>150363530</td>\n",
       "      <td>MAMLD1_5UTR</td>\n",
       "      <td>.;.;.;3|2|5|5|4|1|3|3|1|0|0|0|0|0|0|0|0|0|0|4|...</td>\n",
       "      <td>.;.;.;.;.;.;71|65|46|44|7|4|0|0|0|0|0|2|64|66|...</td>\n",
       "      <td>0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...</td>\n",
       "      <td>.;.;.;0;.;.;-1;.;0,0;32,33,35,29,29,34,34;1,-1...</td>\n",
       "      <td>.;.;.;.;.;.;1,3;.;.;1,2,2;0,1,3;.;1,1;.;1,0;.;...</td>\n",
       "      <td>-1;0,0,0;.;.;0,3;4,1;1,0,1,1;0,1,0,0;.;.;.;37,...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/71501671/ccs;.;.;m...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_200806_223416/98175724/ccs...</td>\n",
       "      <td>m54340U_200716_001732/24969595/ccs;m54340U_210...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150568622</td>\n",
       "      <td>150568662</td>\n",
       "      <td>MTM1_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;3|13|4|2|0|0,98|99|98|95|97|...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;100|100|100|100|100|100,45|5...</td>\n",
       "      <td>1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;0,0;0,0;.;0,0,0;.;.;.;.;0,0;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;0,2;0;.;0,-1;.;.;.;.;1;.;.;....</td>\n",
       "      <td>0,0,0,0,0;0,0,0;0;.;0,0;.;2,0,0,0;.;.,0,0,1,0,...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/158271...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/107350...</td>\n",
       "      <td>m54340U_200716_001732/162859070/ccs,m54340U_20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150693397</td>\n",
       "      <td>150693530</td>\n",
       "      <td>MTMR1_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;2|1|2|10|11|5|3|2|3|1|1|5|18...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;2|0|0|5|2|5|12|23|34|4...</td>\n",
       "      <td>6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;1,0,1,0,2,0;1,2,2,1,1;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;-1,1,-1;.;0,2,3;.;.;0,...</td>\n",
       "      <td>1,0,0;0,0,0,0,2,0,1;.;0;0,0;0;0,0,0,1,1;0,-1,0...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/609751...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;m54340U_200611_224317/...</td>\n",
       "      <td>m54340U_200716_001732/53740679/ccs,m54340U_200...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150898590</td>\n",
       "      <td>150898668</td>\n",
       "      <td>CD99L2_5UTR</td>\n",
       "      <td>.;.;.;8|7|9|52|43|81|48|25;.;.;88|47|83|82|98|...</td>\n",
       "      <td>.;.;.;2|0|0|1|0|0|0|0;.;.;0|0|0|0|0|0|0|13,62|...</td>\n",
       "      <td>0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69;0|0|0|0|1|1...</td>\n",
       "      <td>.;.;.;1;.;.;0,1;.;0;0,0,-1;0,0;.;0,0,1;.;.;.;....</td>\n",
       "      <td>.;.;.;1;.;.;0,0,0;.;.;0,0,1;2,1;.;1,0;.;.;.;.;...</td>\n",
       "      <td>0,0;0,0;.;1,3;2,0,1;.;.;-2,2,0;.;.;.;1,0;.;0;0...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/24904108/ccs;.;.;m...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/134351559/ccs;.;.;...</td>\n",
       "      <td>m54340U_200716_001732/156107929/ccs,m54340U_20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150983336</td>\n",
       "      <td>150983376</td>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;100|100|100|100|100,11|61|12|40|64...</td>\n",
       "      <td>.;.;.;.;.;.;29|50|50|41;.;93|98|98|85|95;70|67...</td>\n",
       "      <td>.;2|11|5|2|6,2|1|15|3|11,5|9|6|2|0;.;.;80|76|5...</td>\n",
       "      <td>.;.;.;.;.;.;0,1,0;.;.;0,0,0;0,0,0;.;0,0;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;0;0;0,0,0;.;2,0,0,0;.;0;0;.;-1...</td>\n",
       "      <td>.;0,0,0;.;.;0;0,0,0;.;0,0,0;0;0,0;.;0,0,0;.;0,...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_210703_193517/127796270/cc...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_200806_223416/168034751/cc...</td>\n",
       "      <td>.;m54340U_210523_044800/100468860/ccs,m54340U_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom        pos        end      region_id  \\\n",
       "1   chrX  147911920  147912179      FMR1_5UTR   \n",
       "7   chrX  149505139  149505306       IDS_5UTR   \n",
       "8   chrX  149631582  149631792  TMEM185A_5UTR   \n",
       "9   chrX  150363318  150363530    MAMLD1_5UTR   \n",
       "10  chrX  150568622  150568662      MTM1_5UTR   \n",
       "11  chrX  150693397  150693530     MTMR1_5UTR   \n",
       "12  chrX  150898590  150898668    CD99L2_5UTR   \n",
       "13  chrX  150983336  150983376     HMGB3_5UTR   \n",
       "\n",
       "                                            methyl_h1  \\\n",
       "1   .;.;.;91|95|100|99|96|100|99|98|98|96|96|97|94...   \n",
       "7   .;.;.;8|2|2|1|0|0|0|0|0|1|0|7|1|0|3|2|1|4|0|0|...   \n",
       "8   .;.;.;.;.;.;95|91|35|70|18|12|12|62|41|61|26|6...   \n",
       "9   .;.;.;3|2|5|5|4|1|3|3|1|0|0|0|0|0|0|0|0|0|0|4|...   \n",
       "10  .;.;.;.;.;.;.;.;.;3|13|4|2|0|0,98|99|98|95|97|...   \n",
       "11  .;.;.;.;.;.;.;.;.;2|1|2|10|11|5|3|2|3|1|1|5|18...   \n",
       "12  .;.;.;8|7|9|52|43|81|48|25;.;.;88|47|83|82|98|...   \n",
       "13  .;.;.;.;.;.;100|100|100|100|100,11|61|12|40|64...   \n",
       "\n",
       "                                            methyl_h2  \\\n",
       "1   .;.;.;64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|...   \n",
       "7   .;.;.;62|51|80|91|75|95|89|98|99|98|97|99|99|9...   \n",
       "8   .;.;.;.;73|60|29|4|2|7|3|1|1|1|1|0|0|0|8|17|11...   \n",
       "9   .;.;.;.;.;.;71|65|46|44|7|4|0|0|0|0|0|2|64|66|...   \n",
       "10  .;.;.;.;.;.;.;.;.;100|100|100|100|100|100,45|5...   \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;2|0|0|5|2|5|12|23|34|4...   \n",
       "12  .;.;.;2|0|0|1|0|0|0|0;.;.;0|0|0|0|0|0|0|13,62|...   \n",
       "13  .;.;.;.;.;.;29|50|50|41;.;93|98|98|85|95;70|67...   \n",
       "\n",
       "                                            methyl_h0  \\\n",
       "1   .;0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3...   \n",
       "7   .;0|2|0|1|0|1|6|2|1|8|9|13|6|3|4|8|5|0|1|0|0|0...   \n",
       "8   5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...   \n",
       "9   0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...   \n",
       "10  1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...   \n",
       "11  6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...   \n",
       "12  0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69;0|0|0|0|1|1...   \n",
       "13  .;2|11|5|2|6,2|1|15|3|11,5|9|6|2|0;.;.;80|76|5...   \n",
       "\n",
       "                                            bp_dev_h1  \\\n",
       "1   .;.;.;32;.;.;30;.;31,30;34;27,27,27,29;.;.;.;3...   \n",
       "7   .;.;.;1;.;.;.;.;.;2;.;.;.;.;0,0,2;.;.;-1,1,0;....   \n",
       "8   .;.;.;.;.;.;-21,-21,-21;.;-21,-15;.;15,9;.;0;....   \n",
       "9   .;.;.;0;.;.;-1;.;0,0;32,33,35,29,29,34,34;1,-1...   \n",
       "10  .;.;.;.;.;.;.;.;.;0,0;0,0;.;0,0,0;.;.;.;.;0,0;...   \n",
       "11  .;.;.;.;.;.;.;.;.;1,0,1,0,2,0;1,2,2,1,1;.;.;.;...   \n",
       "12  .;.;.;1;.;.;0,1;.;0;0,0,-1;0,0;.;0,0,1;.;.;.;....   \n",
       "13  .;.;.;.;.;.;0,1,0;.;.;0,0,0;0,0,0;.;0,0;.;.;.;...   \n",
       "\n",
       "                                            bp_dev_h2  \\\n",
       "1   .;.;.;31,43,30;.;.;30,28,28,31;.;30,30,32;34,2...   \n",
       "7   .;.;.;0;.;.;.;.;.;.;.;.;.;.;.;.;.;1,0,1;.;.;.;...   \n",
       "8   .;.;.;.;-21;.;.,9,0;.;-21,-21;.;-21;.;-11;.;.;...   \n",
       "9   .;.;.;.;.;.;1,3;.;.;1,2,2;0,1,3;.;1,1;.;1,0;.;...   \n",
       "10  .;.;.;.;.;.;.;.;.;0,2;0;.;0,-1;.;.;.;.;1;.;.;....   \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;-1,1,-1;.;0,2,3;.;.;0,...   \n",
       "12  .;.;.;1;.;.;0,0,0;.;.;0,0,1;2,1;.;1,0;.;.;.;.;...   \n",
       "13  .;.;.;.;.;.;.;.;0;0;0,0,0;.;2,0,0,0;.;0;0;.;-1...   \n",
       "\n",
       "                                            bp_dev_h0  \\\n",
       "1   .;51,51,51,57,51,51,52;30;.;.;30;.;34,31;.;.;....   \n",
       "7   .;0,2;3;.;0,0;0;-1,1,0,1;0,-2,0;0,0,-1,0;0,0,1...   \n",
       "8   -21,-19;-21,-21,-21,-21,-18,-21,-21,-25;.;-19;...   \n",
       "9   -1;0,0,0;.;.;0,3;4,1;1,0,1,1;0,1,0,0;.;.;.;37,...   \n",
       "10  0,0,0,0,0;0,0,0;0;.;0,0;.;2,0,0,0;.;.,0,0,1,0,...   \n",
       "11  1,0,0;0,0,0,0,2,0,1;.;0;0,0;0;0,0,0,1,1;0,-1,0...   \n",
       "12  0,0;0,0;.;1,3;2,0,1;.;.;-2,2,0;.;.;.;1,0;.;0;0...   \n",
       "13  .;0,0,0;.;.;0;0,0,0;.;0,0,0;0;0,0;.;0,0,0;.;0,...   \n",
       "\n",
       "                                                rn_h1  \\\n",
       "1   .;.;.;m54340U_200716_153006/27724288/ccs;.;.;m...   \n",
       "7   .;.;.;m54340U_200716_153006/128059025/ccs;.;.;...   \n",
       "8   .;.;.;.;.;.;m54340U_210703_193517/110166254/cc...   \n",
       "9   .;.;.;m54340U_200716_153006/71501671/ccs;.;.;m...   \n",
       "10  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/158271...   \n",
       "11  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/609751...   \n",
       "12  .;.;.;m54340U_200716_153006/24904108/ccs;.;.;m...   \n",
       "13  .;.;.;.;.;.;m54340U_210703_193517/127796270/cc...   \n",
       "\n",
       "                                                rn_h2  \\\n",
       "1   .;.;.;m54340U_200716_153006/75497573/ccs,m5434...   \n",
       "7   .;.;.;m54340U_200716_153006/133105829/ccs;.;.;...   \n",
       "8   .;.;.;.;m54340U_200716_153006/94110941/ccs;.;m...   \n",
       "9   .;.;.;.;.;.;m54340U_200806_223416/98175724/ccs...   \n",
       "10  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/107350...   \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;m54340U_200611_224317/...   \n",
       "12  .;.;.;m54340U_200716_153006/134351559/ccs;.;.;...   \n",
       "13  .;.;.;.;.;.;m54340U_200806_223416/168034751/cc...   \n",
       "\n",
       "                                                rn_h0  \n",
       "1   .;m54340U_210523_044800/52758275/ccs,m54340U_2...  \n",
       "7   .;m54340U_200806_070905/88277013/ccs,m54340U_2...  \n",
       "8   m54340U_200716_001732/174262380/ccs,m54340U_20...  \n",
       "9   m54340U_200716_001732/24969595/ccs;m54340U_210...  \n",
       "10  m54340U_200716_001732/162859070/ccs,m54340U_20...  \n",
       "11  m54340U_200716_001732/53740679/ccs,m54340U_200...  \n",
       "12  m54340U_200716_001732/156107929/ccs,m54340U_20...  \n",
       "13  .;m54340U_210523_044800/100468860/ccs,m54340U_...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of samples: 270\n"
     ]
    }
   ],
   "source": [
    "file_in = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/METHYLATION/snoopsv_methyl/FMR1_ALL/methyl_data.tsv'\n",
    "df = pd.read_table(file_in, sep='\\t', header=0, keep_default_na=False)\n",
    "df = df.loc[df.region_id.str.contains('_5UTR')].copy()\n",
    "#df = df.reindex(index=[i for i in range(df.shape[0])])\n",
    "#.reindex(labels=[])\n",
    "print('df:')\n",
    "display(df)\n",
    "\n",
    "file_sam = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/METHYLATION/snoopsv_methyl/FMR1_ALL/methyl_data_sample_order.txt'\n",
    "with open(file_sam, 'r') as fh:\n",
    "    sample_order = fh.readline().split(',')[:-1] # last one is an empty string\n",
    "print(f'number of samples: {len(sample_order)}')\n",
    "#print(f'sample order: {sample_order}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "30341c5d-2104-46fc-943c-29edb826a4cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>region_id</th>\n",
       "      <th>REACH000026_methyl_h1</th>\n",
       "      <th>REACH000026_methyl_h2</th>\n",
       "      <th>REACH000026_methyl_h0</th>\n",
       "      <th>REACH000026_bp_dev_h1</th>\n",
       "      <th>REACH000026_bp_dev_h2</th>\n",
       "      <th>REACH000026_bp_dev_h0</th>\n",
       "      <th>...</th>\n",
       "      <th>REACH000684_rn_h0</th>\n",
       "      <th>REACH000685_methyl_h1</th>\n",
       "      <th>REACH000685_methyl_h2</th>\n",
       "      <th>REACH000685_methyl_h0</th>\n",
       "      <th>REACH000685_bp_dev_h1</th>\n",
       "      <th>REACH000685_bp_dev_h2</th>\n",
       "      <th>REACH000685_bp_dev_h0</th>\n",
       "      <th>REACH000685_rn_h1</th>\n",
       "      <th>REACH000685_rn_h2</th>\n",
       "      <th>REACH000685_rn_h0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147911920</td>\n",
       "      <td>147912179</td>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>13,17,33,29,16</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>658567b8-0bb0-48a8-a85d-619f26f01f60,ec6386b7-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149505139</td>\n",
       "      <td>149505306</td>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>f73f95fe-76d3-41e0-8469-c8456874f172,6e115d22-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>11|0|0|0|0|15|0|0|0|4|89|91|2|0|0|0|1|9|93|89|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6,-14,-3,-4,-29,-24,-17,-8,-69</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>dbdc51c7-868d-4c1e-aef8-54b4be7616b1,ee73bd80-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149631582</td>\n",
       "      <td>149631792</td>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-21,-19</td>\n",
       "      <td>...</td>\n",
       "      <td>3a23193f-a7dc-4723-82a1-80cdcf61dd63,87c620df-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>100|4|30|7|40|97|0,0|0|0|0|2|0|0|1|0|0|0|0|2|2...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-23,-13,-2,-8,-21,.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0bdcf4ce-425a-4468-9796-785e0a2817f8,8d6dac54-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150363318</td>\n",
       "      <td>150363530</td>\n",
       "      <td>MAMLD1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1fd78472-ace3-40c8-ac5e-a8177b6a7284,0a85a084-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1|0|1|0|0|0|4|0|3|61|9|12|46|79|84|75|0|1|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-8,-3,-11,-2,-17,8,-5,-2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>b0ddb187-c71b-4f31-8e70-da57f39b8cff,9b1c13e4-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150568622</td>\n",
       "      <td>150568662</td>\n",
       "      <td>MTM1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0,0,0,0,0</td>\n",
       "      <td>...</td>\n",
       "      <td>76f39933-8fcc-46eb-a59a-6a6cd09427dc,a36f05c0-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>40|36|9|41|79|30|22|2|45,4|2|0|0|0|0|5,0|4|0|1...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>9,-1,-2,2,6</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7178be63-b143-43d2-b646-7f250eee984a,3152af87-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150693397</td>\n",
       "      <td>150693530</td>\n",
       "      <td>MTMR1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1,0,0</td>\n",
       "      <td>...</td>\n",
       "      <td>676248ca-67b4-40c0-9fc5-32e5b3373c0b,f81f4122-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|9|0|1|0|0|0|1|13|0|16|0|0|2|0|5,55|70|9|2|1|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-10,-2,5,0,-3,3,5,-5,-16,-4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>33acaac3-8252-404b-b9e8-e6c5fde090e9,21350c97-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150898590</td>\n",
       "      <td>150898668</td>\n",
       "      <td>CD99L2_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0,0</td>\n",
       "      <td>...</td>\n",
       "      <td>ca01fc96-4616-4d64-9ad7-2f89ec68a1e5,3484b9c0-...</td>\n",
       "      <td>33|9|91|41|0|2|1|6</td>\n",
       "      <td>1|0|0|7|7|0|0|0|2,45|0|21|0|0|2|0|2</td>\n",
       "      <td>46|85,0|49|2|0|0|4|0|0,0|80|0|2|0|3|0|36,10|0</td>\n",
       "      <td>-3</td>\n",
       "      <td>4,0</td>\n",
       "      <td>-6,-6,-5,.</td>\n",
       "      <td>5af74e3f-fde8-469d-9c44-8ca2cce771d2</td>\n",
       "      <td>34a1e889-e5f6-4400-8f70-0d32c5afd4d6,7d1b38e0-...</td>\n",
       "      <td>1ae72adc-1d37-4141-bfa6-590e0ec211d1,46f0b406-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150983336</td>\n",
       "      <td>150983376</td>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>58341e7d-ff66-4acc-8aaa-0dbc00054c82,b0d7c6e3-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|12|0,0|0|0|0|2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1,-6</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>519f9439-750b-421f-9828-4705fb07f845,94933348-...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 2434 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom        pos        end      region_id REACH000026_methyl_h1  \\\n",
       "1   chrX  147911920  147912179      FMR1_5UTR                     .   \n",
       "7   chrX  149505139  149505306       IDS_5UTR                     .   \n",
       "8   chrX  149631582  149631792  TMEM185A_5UTR                     .   \n",
       "9   chrX  150363318  150363530    MAMLD1_5UTR                     .   \n",
       "10  chrX  150568622  150568662      MTM1_5UTR                     .   \n",
       "11  chrX  150693397  150693530     MTMR1_5UTR                     .   \n",
       "12  chrX  150898590  150898668    CD99L2_5UTR                     .   \n",
       "13  chrX  150983336  150983376     HMGB3_5UTR                     .   \n",
       "\n",
       "   REACH000026_methyl_h2                              REACH000026_methyl_h0  \\\n",
       "1                      .                                                  .   \n",
       "7                      .                                                  .   \n",
       "8                      .  5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...   \n",
       "9                      .  0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...   \n",
       "10                     .  1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...   \n",
       "11                     .  6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...   \n",
       "12                     .                 0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69   \n",
       "13                     .                                                  .   \n",
       "\n",
       "   REACH000026_bp_dev_h1 REACH000026_bp_dev_h2 REACH000026_bp_dev_h0  ...  \\\n",
       "1                      .                     .                     .  ...   \n",
       "7                      .                     .                     .  ...   \n",
       "8                      .                     .               -21,-19  ...   \n",
       "9                      .                     .                    -1  ...   \n",
       "10                     .                     .             0,0,0,0,0  ...   \n",
       "11                     .                     .                 1,0,0  ...   \n",
       "12                     .                     .                   0,0  ...   \n",
       "13                     .                     .                     .  ...   \n",
       "\n",
       "                                    REACH000684_rn_h0 REACH000685_methyl_h1  \\\n",
       "1                                                   .                     .   \n",
       "7   f73f95fe-76d3-41e0-8469-c8456874f172,6e115d22-...                     .   \n",
       "8   3a23193f-a7dc-4723-82a1-80cdcf61dd63,87c620df-...                     .   \n",
       "9   1fd78472-ace3-40c8-ac5e-a8177b6a7284,0a85a084-...                     .   \n",
       "10  76f39933-8fcc-46eb-a59a-6a6cd09427dc,a36f05c0-...                     .   \n",
       "11  676248ca-67b4-40c0-9fc5-32e5b3373c0b,f81f4122-...                     .   \n",
       "12  ca01fc96-4616-4d64-9ad7-2f89ec68a1e5,3484b9c0-...    33|9|91|41|0|2|1|6   \n",
       "13  58341e7d-ff66-4acc-8aaa-0dbc00054c82,b0d7c6e3-...                     .   \n",
       "\n",
       "                  REACH000685_methyl_h2  \\\n",
       "1                                     .   \n",
       "7                                     .   \n",
       "8                                     .   \n",
       "9                                     .   \n",
       "10                                    .   \n",
       "11                                    .   \n",
       "12  1|0|0|7|7|0|0|0|2,45|0|21|0|0|2|0|2   \n",
       "13                                    .   \n",
       "\n",
       "                                REACH000685_methyl_h0 REACH000685_bp_dev_h1  \\\n",
       "1   2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...                     .   \n",
       "7   11|0|0|0|0|15|0|0|0|4|89|91|2|0|0|0|1|9|93|89|...                     .   \n",
       "8   100|4|30|7|40|97|0,0|0|0|0|2|0|0|1|0|0|0|0|2|2...                     .   \n",
       "9   1|0|1|0|0|0|4|0|3|61|9|12|46|79|84|75|0|1|0|0|...                     .   \n",
       "10  40|36|9|41|79|30|22|2|45,4|2|0|0|0|0|5,0|4|0|1...                     .   \n",
       "11  0|9|0|1|0|0|0|1|13|0|16|0|0|2|0|5,55|70|9|2|1|...                     .   \n",
       "12      46|85,0|49|2|0|0|4|0|0,0|80|0|2|0|3|0|36,10|0                    -3   \n",
       "13                                 0|0|12|0,0|0|0|0|2                     .   \n",
       "\n",
       "   REACH000685_bp_dev_h2           REACH000685_bp_dev_h0  \\\n",
       "1                      .                  13,17,33,29,16   \n",
       "7                      .  6,-14,-3,-4,-29,-24,-17,-8,-69   \n",
       "8                      .             -23,-13,-2,-8,-21,.   \n",
       "9                      .        -8,-3,-11,-2,-17,8,-5,-2   \n",
       "10                     .                     9,-1,-2,2,6   \n",
       "11                     .     -10,-2,5,0,-3,3,5,-5,-16,-4   \n",
       "12                   4,0                      -6,-6,-5,.   \n",
       "13                     .                           -1,-6   \n",
       "\n",
       "                       REACH000685_rn_h1  \\\n",
       "1                                      .   \n",
       "7                                      .   \n",
       "8                                      .   \n",
       "9                                      .   \n",
       "10                                     .   \n",
       "11                                     .   \n",
       "12  5af74e3f-fde8-469d-9c44-8ca2cce771d2   \n",
       "13                                     .   \n",
       "\n",
       "                                    REACH000685_rn_h2  \\\n",
       "1                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12  34a1e889-e5f6-4400-8f70-0d32c5afd4d6,7d1b38e0-...   \n",
       "13                                                  .   \n",
       "\n",
       "                                    REACH000685_rn_h0  \n",
       "1   658567b8-0bb0-48a8-a85d-619f26f01f60,ec6386b7-...  \n",
       "7   dbdc51c7-868d-4c1e-aef8-54b4be7616b1,ee73bd80-...  \n",
       "8   0bdcf4ce-425a-4468-9796-785e0a2817f8,8d6dac54-...  \n",
       "9   b0ddb187-c71b-4f31-8e70-da57f39b8cff,9b1c13e4-...  \n",
       "10  7178be63-b143-43d2-b646-7f250eee984a,3152af87-...  \n",
       "11  33acaac3-8252-404b-b9e8-e6c5fde090e9,21350c97-...  \n",
       "12  1ae72adc-1d37-4141-bfa6-590e0ec211d1,46f0b406-...  \n",
       "13  519f9439-750b-421f-9828-4705fb07f845,94933348-...  \n",
       "\n",
       "[8 rows x 2434 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_exnd = df[['chrom' ,'pos', 'end', 'region_id']].copy()\n",
    "cols = df.columns.tolist()[4:]\n",
    "#print(cols)\n",
    "\n",
    "for idx, sample in enumerate(sample_order):\n",
    "    #if sample == 'REACH000450' or sample == 'REACH000452':\n",
    "    #    continue\n",
    "    for col in cols:\n",
    "        df_exnd[f'{sample}_{col}'] = df[col].str.split(';').apply(lambda row: row[idx])\n",
    "display(df_exnd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d9172bd9-c93c-4ea4-89d2-9c21ad10224d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>MEAN_COVERAGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG004</td>\n",
       "      <td>ONT</td>\n",
       "      <td>81.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000626</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACH000546</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACH000683</td>\n",
       "      <td>ONT</td>\n",
       "      <td>9.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACH000563</td>\n",
       "      <td>ONT</td>\n",
       "      <td>8.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>REACH000284</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>REACH000672</td>\n",
       "      <td>PB</td>\n",
       "      <td>5.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>REACH000292</td>\n",
       "      <td>PB</td>\n",
       "      <td>9.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>REACH000066</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>2602-2-1</td>\n",
       "      <td>PB</td>\n",
       "      <td>0.89</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>288 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SAMPLE COHORT  MEAN_COVERAGE\n",
       "0          HG004    ONT          81.44\n",
       "1    REACH000626    ONT          10.51\n",
       "2    REACH000546    ONT          10.87\n",
       "3    REACH000683    ONT           9.97\n",
       "4    REACH000563    ONT           8.68\n",
       "..           ...    ...            ...\n",
       "283  REACH000284     PB           3.43\n",
       "284  REACH000672     PB           5.02\n",
       "285  REACH000292     PB           9.00\n",
       "286  REACH000066     PB           1.61\n",
       "287     2602-2-1     PB           0.89\n",
       "\n",
       "[288 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "file_cov = '/expanse/projects/sebat1/s3/data/sebat/long_reads_pipeline_results/coverages_mod.tsv'\n",
    "#file_cov = '/tscc/projects/ps-sebat1/s3/data/sebat/long_reads_pipeline_results/coverages_mod.tsv'\n",
    "df_cov = pd.read_table(file_cov, header=0, sep='\\t')\n",
    "multiplat_samples = ['REACH000236', 'REACH000530', 'REACH000531', 'REACH000532', 'REACH000436']\n",
    "df_cov['SAMPLE'] = df_cov.apply(lambda row: row['SAMPLE']+'_'+row['COHORT'] if row['SAMPLE'] in multiplat_samples else row['SAMPLE'], axis=1)\n",
    "df_cov_high = df_cov.loc[df_cov.SAMPLE.str.startswith('REACH000') & (df_cov.MEAN_COVERAGE > 9)]\n",
    "display(df_cov)\n",
    "high_cov_samples = df_cov_high.SAMPLE.tolist()\n",
    "#print(high_cov_samples)\n",
    "\n",
    "def get_plat(sample):\n",
    "    try:\n",
    "        ret = df_cov[df_cov.SAMPLE == sample]['COHORT'].values[0]\n",
    "    except:\n",
    "        print(f'problem with samples: {sample}')\n",
    "        ret = ''\n",
    "    return ret\n",
    "\n",
    "def get_cov(sample):\n",
    "    try:\n",
    "        ret = df_cov[df_cov.SAMPLE == sample]['MEAN_COVERAGE'].values[0]\n",
    "    except:\n",
    "        print(f'problem with samples: {sample} in get_cov')\n",
    "        ret = ''\n",
    "    return ret\n",
    "\n",
    "plat_dict = {sample: get_plat(sample) for sample in df_cov.SAMPLE.tolist()}\n",
    "plat_dict['REACH000226_blood'] = plat_dict['REACH000226']\n",
    "plat_dict['REACH000226_sperm'] = plat_dict['REACH000226']\n",
    "#print('plat_dict:')\n",
    "#print(plat_dict)\n",
    "\n",
    "cov_dict = {sample: get_cov(sample) for sample in df_cov.SAMPLE.tolist()}\n",
    "#print('cov_dict:')\n",
    "#print(cov_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "b13becbf-2f34-4a72-9c29-ff8d1a02f39d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_trios:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(117, 6)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_samples:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>plat</th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>6.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>REACH000065</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>REACH000086</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>2.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>REACH000089</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>REACH000651</td>\n",
       "      <td>M</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>5.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>REACH000195</td>\n",
       "      <td>M</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>2.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>REACH000285</td>\n",
       "      <td>M</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REACH000226_blood</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>ONT</td>\n",
       "      <td>1.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>ONT</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>245 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample_id rel  sex plat  coverage\n",
       "10        REACH000026   C    1   PB      3.22\n",
       "11        REACH000058   C    1   PB      6.19\n",
       "12        REACH000065   C    1   PB      1.22\n",
       "15        REACH000086   C    1   PB      2.70\n",
       "18        REACH000089   C    2   PB      3.27\n",
       "..                ...  ..  ...  ...       ...\n",
       "60        REACH000651   M    2   PB      5.75\n",
       "61        REACH000195   M    2   PB      2.67\n",
       "62        REACH000285   M    2   PB      3.00\n",
       "0   REACH000226_blood   D    1  ONT      1.92\n",
       "1   REACH000226_sperm   D    1  ONT     27.87\n",
       "\n",
       "[245 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sex\n",
      "1    139\n",
      "2    106\n",
      "Name: count, dtype: int64\n",
      "plat\n",
      "PB     142\n",
      "ONT    103\n",
      "Name: count, dtype: int64\n",
      "rel\n",
      "C    117\n",
      "D     65\n",
      "M     63\n",
      "Name: count, dtype: int64\n",
      "===============================\n",
      "df_samples_flt:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(206, 5)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===============================\n",
      "df_samples_flt_female_kids:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(39, 5)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "plat\n",
      "ONT    23\n",
      "PB     16\n",
      "Name: count, dtype: int64\n",
      "===============================\n",
      "df_samples_flt_females:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(93, 5)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "plat\n",
      "PB     47\n",
      "ONT    46\n",
      "Name: count, dtype: int64\n",
      "rel\n",
      "M    54\n",
      "C    39\n",
      "Name: count, dtype: int64\n",
      "===============================\n",
      "df_samples_flt_males:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(113, 5)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "plat\n",
      "PB     57\n",
      "ONT    56\n",
      "Name: count, dtype: int64\n",
      "rel\n",
      "C    62\n",
      "D    51\n",
      "Name: count, dtype: int64\n",
      "['REACH000026', 'REACH000058', 'REACH000086', 'REACH000097', 'REACH000120', 'REACH000159', 'REACH000162', 'REACH000182', 'REACH000185', 'REACH000194', 'REACH000197', 'REACH000214', 'REACH000222', 'REACH000233', 'REACH000287', 'REACH000288', 'REACH000291', 'REACH000292', 'REACH000293', 'REACH000356', 'REACH000441', 'REACH000537', 'REACH000643', 'REACH000650', 'REACH000653', 'REACH000671', 'REACH000708', 'REACH000730', 'REACH000747', 'REACH000750', 'REACH000762', 'REACH000767', 'REACH000001', 'REACH000154', 'REACH000400', 'REACH000403', 'REACH000414', 'REACH000419', 'REACH000422', 'REACH000423', 'REACH000426', 'REACH000427', 'REACH000428', 'REACH000448', 'REACH000476', 'REACH000483', 'REACH000501', 'REACH000514', 'REACH000517', 'REACH000529', 'REACH000532_ONT', 'REACH000540', 'REACH000544', 'REACH000558', 'REACH000559', 'REACH000560', 'REACH000564', 'REACH000626', 'REACH000663', 'REACH000681', 'REACH000684', 'REACH000685', 'REACH000503', 'REACH000642', 'REACH000646', 'REACH000707', 'REACH000562', 'REACH000156', 'REACH000542', 'REACH000440', 'REACH000481', 'REACH000557', 'REACH000099', 'REACH000749', 'REACH000662', 'REACH000295', 'REACH000094', 'REACH000442', 'REACH000566', 'REACH000402', 'REACH000161', 'REACH000437', 'REACH000753', 'REACH000539', 'REACH000452', 'REACH000524', 'REACH000088', 'REACH000315', 'REACH000516', 'REACH000196', 'REACH000683', 'REACH000580', 'REACH000122', 'REACH000411', 'REACH000629', 'REACH000290', 'REACH000478', 'REACH000425', 'REACH000235', 'REACH000226', 'REACH000184', 'REACH000421', 'REACH000216', 'REACH000546', 'REACH000286', 'REACH000673', 'REACH000531_ONT', 'REACH000604', 'REACH000527', 'REACH000512', 'REACH000221', 'REACH000416', 'REACH000226_sperm']\n"
     ]
    }
   ],
   "source": [
    "file_psam = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/process_IL_LR/HWE/REACH_LR_platform.psam'\n",
    "df_p = pd.read_table(file_psam, sep='\\t', header=None, names = ['famid', 'sample_id', 'dad', 'mom', 'sex', 'phen'])\n",
    "#print('df_p:')\n",
    "#display(df_p)\n",
    "ds_sex = df_p.set_index('sample_id').sex\n",
    "#display(ds_sex)\n",
    "\n",
    "# make mom and dad dictionary\n",
    "sample_dad_dict = {}\n",
    "sample_mom_dict = {}\n",
    "for sample, dad, mom in zip(df_p.sample_id, df_p.dad, df_p.mom):\n",
    "    #print(sample, dad, mom)\n",
    "    sample_dad_dict[sample] = dad\n",
    "    sample_mom_dict[sample] = mom\n",
    "    \n",
    "#print('complete trios:')\n",
    "df_trios = df_p.loc[df_p.dad.isin(df_p.sample_id) & df_p.mom.isin(df_p.sample_id)]\n",
    "print('df_trios:')\n",
    "display(df_trios.shape)\n",
    "\n",
    "### make a samples dataframe containing all samples in complete trios\n",
    "df_samples = pd.DataFrame()\n",
    "df_samples['sample_id'] = df_trios['sample_id']\n",
    "df_samples['rel'] = 'C'\n",
    "\n",
    "df_dads = pd.DataFrame()\n",
    "df_dads['sample_id'] =  list(set(df_trios['dad'].tolist()))\n",
    "df_dads['rel'] = 'D'\n",
    "df_samples = pd.concat([df_samples, df_dads])\n",
    "\n",
    "df_moms = pd.DataFrame()\n",
    "df_moms['sample_id'] =  list(set(df_trios['mom'].tolist()))\n",
    "df_moms['rel'] = 'M'\n",
    "df_samples = pd.concat([df_samples, df_moms])\n",
    "\n",
    "df_samples['sex'] = df_samples.sample_id.map(ds_sex)\n",
    "df_samples['plat'] = df_samples.sample_id.map(plat_dict)\n",
    "df_samples['coverage'] = df_samples.sample_id.map(cov_dict)\n",
    "### add blood/sperm sections for REACH000226\n",
    "row_add = pd.DataFrame({'sample_id':['REACH000226_blood', 'REACH000226_sperm'] ,'rel':['D', 'D'] ,'sex':[1, 1] ,'plat':['ONT', 'ONT'] ,'coverage':[1.92, 27.87]})\n",
    "df_samples = pd.concat([df_samples, row_add])\n",
    "pd.set_option('display.max_rows', 10)\n",
    "print('df_samples:')\n",
    "display(df_samples)\n",
    "\n",
    "print(df_samples.sex.value_counts())\n",
    "print(df_samples.plat.value_counts())\n",
    "print(df_samples.rel.value_counts())\n",
    "\n",
    "print('===============================')\n",
    "df_samples_flt = df_samples.loc[(df_samples.coverage>=2)]\n",
    "print('df_samples_flt:')\n",
    "display(df_samples_flt.shape)\n",
    "\n",
    "print('===============================')\n",
    "df_samples_flt_female_kids = df_samples_flt.loc[(df_samples_flt.sex==2)&(df_samples_flt.rel=='C')]\n",
    "print('df_samples_flt_female_kids:')\n",
    "display(df_samples_flt_female_kids.shape)\n",
    "print(df_samples_flt_female_kids.plat.value_counts())\n",
    "\n",
    "print('===============================')\n",
    "df_samples_flt_females = df_samples_flt.loc[(df_samples_flt.sex==2)]\n",
    "print('df_samples_flt_females:')\n",
    "display(df_samples_flt_females.shape)\n",
    "print(df_samples_flt_females.plat.value_counts())\n",
    "print(df_samples_flt_females.rel.value_counts())\n",
    "\n",
    "print('===============================')\n",
    "df_samples_flt_males = df_samples_flt.loc[(df_samples_flt.sex==1)]\n",
    "print('df_samples_flt_males:')\n",
    "display(df_samples_flt_males.shape)\n",
    "print(df_samples_flt_males.plat.value_counts())\n",
    "print(df_samples_flt_males.rel.value_counts())\n",
    "print(df_samples_flt_males.sample_id.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "582e536e-b8e1-4d39-a945-96f337eef2c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_file = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/REACH_sample_info.tsv'\n",
    "df_meta = pd.read_table(meta_file, sep='\\t', header=0)\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "df_meta['Sample_ID'] = df_meta.apply(lambda row: rename_samples_dict[row['Sample_ID']] \n",
    "                                     if row['Sample_ID'] in rename_samples_dict else row['Sample_ID'], axis=1)\n",
    "\n",
    "aff_dict = {}\n",
    "for sample, aff in zip(df_meta['Sample_ID'].tolist(), df_meta['Affected'].tolist()):\n",
    "    aff_dict[sample] = aff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a075a41a-5db3-4be3-a147-58b9063b5c20",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_pca_flt:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phen</th>\n",
       "      <th>sex</th>\n",
       "      <th>cohort</th>\n",
       "      <th>duo</th>\n",
       "      <th>family</th>\n",
       "      <th>is_eur</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>...</th>\n",
       "      <th>PC2_MAT</th>\n",
       "      <th>PC3_MAT</th>\n",
       "      <th>PC4_MAT</th>\n",
       "      <th>PC5_MAT</th>\n",
       "      <th>PC6_MAT</th>\n",
       "      <th>PC7_MAT</th>\n",
       "      <th>PC8_MAT</th>\n",
       "      <th>PC9_MAT</th>\n",
       "      <th>PC10_MAT</th>\n",
       "      <th>rel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>F0001</td>\n",
       "      <td>REACH000001</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.008627</td>\n",
       "      <td>-0.005607</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.002845</td>\n",
       "      <td>-0.019312</td>\n",
       "      <td>-0.013769</td>\n",
       "      <td>-0.014384</td>\n",
       "      <td>0.007249</td>\n",
       "      <td>0.002096</td>\n",
       "      <td>0.005399</td>\n",
       "      <td>-0.014983</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.095364</td>\n",
       "      <td>-0.045230</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047780</td>\n",
       "      <td>-0.017435</td>\n",
       "      <td>0.153686</td>\n",
       "      <td>0.005292</td>\n",
       "      <td>-0.009487</td>\n",
       "      <td>-0.020773</td>\n",
       "      <td>0.034190</td>\n",
       "      <td>0.015895</td>\n",
       "      <td>-0.011925</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>F0058</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002422</td>\n",
       "      <td>-0.009216</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003289</td>\n",
       "      <td>0.003757</td>\n",
       "      <td>-0.004828</td>\n",
       "      <td>0.007650</td>\n",
       "      <td>0.005633</td>\n",
       "      <td>0.032103</td>\n",
       "      <td>0.009510</td>\n",
       "      <td>0.020007</td>\n",
       "      <td>0.016383</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>F0065</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.252774</td>\n",
       "      <td>0.107933</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049799</td>\n",
       "      <td>0.158347</td>\n",
       "      <td>0.011411</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.007690</td>\n",
       "      <td>-0.006605</td>\n",
       "      <td>0.008261</td>\n",
       "      <td>-0.012000</td>\n",
       "      <td>-0.009906</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.044785</td>\n",
       "      <td>0.038065</td>\n",
       "      <td>...</td>\n",
       "      <td>0.080387</td>\n",
       "      <td>0.132056</td>\n",
       "      <td>0.029583</td>\n",
       "      <td>0.020156</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>-0.002383</td>\n",
       "      <td>0.014468</td>\n",
       "      <td>0.006253</td>\n",
       "      <td>-0.018303</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0288</td>\n",
       "      <td>REACH000756</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.204041</td>\n",
       "      <td>0.127559</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.030630</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000764</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.027327</td>\n",
       "      <td>-0.004878</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000768</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.091442</td>\n",
       "      <td>-0.016917</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000769</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.086006</td>\n",
       "      <td>-0.003008</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       fid          iid phen     sex  cohort   duo  family  is_eur       PC1  \\\n",
       "136  F0001  REACH000001  ASD    Male       1  trio     1.0       1  0.008627   \n",
       "137  F0026  REACH000026  ASD    Male       1  trio     1.0       0  0.095364   \n",
       "139  F0058  REACH000058  ASD    Male       1  trio     1.0       1  0.002422   \n",
       "140  F0065  REACH000065  ASD    Male       1  trio     1.0       0  0.252774   \n",
       "146  F0078  REACH000086  ASD    Male       1  trio     2.0       0  0.044785   \n",
       "..     ...          ...  ...     ...     ...   ...     ...     ...       ...   \n",
       "0    F0288  REACH000756    .    Male       1     .     NaN       0  0.204041   \n",
       "0    F0290  REACH000763    .  Female       1     .     NaN       1 -0.030630   \n",
       "0    F0290  REACH000764    .    Male       1     .     NaN       1 -0.027327   \n",
       "0    F0291  REACH000768    .  Female       1     .     NaN       0  0.091442   \n",
       "0    F0291  REACH000769    .    Male       1     .     NaN       0  0.086006   \n",
       "\n",
       "          PC2  ...   PC2_MAT   PC3_MAT   PC4_MAT   PC5_MAT   PC6_MAT  \\\n",
       "136 -0.005607  ...  0.000410  0.002845 -0.019312 -0.013769 -0.014384   \n",
       "137 -0.045230  ... -0.047780 -0.017435  0.153686  0.005292 -0.009487   \n",
       "139 -0.009216  ... -0.003289  0.003757 -0.004828  0.007650  0.005633   \n",
       "140  0.107933  ...  0.049799  0.158347  0.011411  0.004728  0.007690   \n",
       "146  0.038065  ...  0.080387  0.132056  0.029583  0.020156  0.007971   \n",
       "..        ...  ...       ...       ...       ...       ...       ...   \n",
       "0    0.127559  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0    0.003052  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0   -0.004878  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0   -0.016917  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0   -0.003008  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "      PC7_MAT   PC8_MAT   PC9_MAT  PC10_MAT  rel  \n",
       "136  0.007249  0.002096  0.005399 -0.014983    C  \n",
       "137 -0.020773  0.034190  0.015895 -0.011925    C  \n",
       "139  0.032103  0.009510  0.020007  0.016383    C  \n",
       "140 -0.006605  0.008261 -0.012000 -0.009906    C  \n",
       "146 -0.002383  0.014468  0.006253 -0.018303    C  \n",
       "..        ...       ...       ...       ...  ...  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "\n",
       "[243 rows x 39 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "file_pca = '/expanse/projects/sebat1/j3guevar/SFARI/aspera-sf/SFARI_DS468588/data/master_phen_4.SBayesR.20210915.csv'\n",
    "\n",
    "df_pca = pd.read_table(file_pca, sep=',', header=0)\n",
    "cols = ['fid', 'iid', 'phen','sex','cohort','duo','family','is_eur','PC1','PC2','PC3','PC4','PC5','PC6','PC7','PC8','PC9','PC10', \n",
    "        'PC1_PAT','PC2_PAT','PC3_PAT','PC4_PAT','PC5_PAT','PC6_PAT','PC7_PAT','PC8_PAT','PC9_PAT','PC10_PAT', \n",
    "        'PC1_MAT','PC2_MAT','PC3_MAT','PC4_MAT','PC5_MAT','PC6_MAT','PC7_MAT','PC8_MAT','PC9_MAT','PC10_MAT']\n",
    "df_pca = df_pca[cols]\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "df_pca['iid'] = df_pca.apply(lambda row: rename_samples_dict[row['iid']] if row['iid'] in rename_samples_dict else row['iid'], axis=1)\n",
    "#display(df_pca)\n",
    "\n",
    "# first filter for the kids, then add parents to them\n",
    "df_pca_flt = df_pca.loc[df_pca.iid.isin(df_trios.sample_id)].copy()\n",
    "df_pca_flt['rel'] = 'C'\n",
    "#print('df_pca_flt:')\n",
    "#display(df_pca_flt)\n",
    "\n",
    "df_parents = pd.DataFrame()\n",
    "for index, row in df_pca_flt.iterrows():\n",
    "    sample = row['iid']\n",
    "    fid = row['fid']\n",
    "    phen = row['phen']\n",
    "    is_eur = row['is_eur']\n",
    "    pc_mat = [row[f'PC{x}_MAT'] for x in range(1,11)]\n",
    "    pc_pat = [row[f'PC{x}_PAT'] for x in range(1,11)]\n",
    "    dad = sample_dad_dict[sample]\n",
    "    mom = sample_mom_dict[sample]\n",
    "    \n",
    "    dict_mat = {'fid': [fid], 'iid': [mom], 'phen': ['.'], 'sex': ['Female'], 'cohort': [1], 'duo': ['.'], 'is_eur': [is_eur]}\n",
    "    dict_mat.update({f'PC{x}': [pc_mat[x-1]] for x in range(1,11)})\n",
    "    df_mat = pd.DataFrame(dict_mat)\n",
    "\n",
    "    dict_pat = {'fid': [fid], 'iid': [dad], 'phen': ['.'], 'sex': ['Male'], 'cohort': [1], 'duo': ['.'], 'is_eur': [is_eur]}\n",
    "    dict_pat.update({f'PC{x}': [pc_pat[x-1]] for x in range(1,11)})\n",
    "    df_pat = pd.DataFrame(dict_pat)\n",
    "    \n",
    "    if df_parents.shape[0] == 0 or mom not in df_parents.iid.tolist():\n",
    "        df_parents = pd.concat([df_parents, df_mat])\n",
    "    if df_parents.shape[0] == 0 or dad not in df_parents.iid.tolist():\n",
    "        df_parents = pd.concat([df_parents, df_pat])\n",
    "\n",
    "df_parents['rel'] = 'P'\n",
    "#print('df_parents:')\n",
    "#display(df_parents)\n",
    "\n",
    "df_pca_flt = pd.concat([df_pca_flt, df_parents])\n",
    "print('df_pca_flt:')\n",
    "display(df_pca_flt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5ef9c2a5-9ca0-4bcb-9dfb-95075df31906",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_id</th>\n",
       "      <th>GB</th>\n",
       "      <th>Z</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Fam_id</th>\n",
       "      <th>sex</th>\n",
       "      <th>rel</th>\n",
       "      <th>GB_min</th>\n",
       "      <th>GB_max</th>\n",
       "      <th>repeats_min</th>\n",
       "      <th>repeats_max</th>\n",
       "      <th>Affected</th>\n",
       "      <th>Platform</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REACH000001</td>\n",
       "      <td>29|31</td>\n",
       "      <td>0.20|0.32</td>\n",
       "      <td>0.99</td>\n",
       "      <td>F0001</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>29</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ONT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000107</td>\n",
       "      <td>0|34</td>\n",
       "      <td>-1.64|0.51</td>\n",
       "      <td>1</td>\n",
       "      <td>F0001</td>\n",
       "      <td>2</td>\n",
       "      <td>P</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>20</td>\n",
       "      <td>31</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ONT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACH000226</td>\n",
       "      <td>0|4</td>\n",
       "      <td>-1.64|-1.38</td>\n",
       "      <td>1</td>\n",
       "      <td>F0001</td>\n",
       "      <td>1</td>\n",
       "      <td>P</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>Yes</td>\n",
       "      <td>ONT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACH000026</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>F0026</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Yes</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACH000269</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>F0026</td>\n",
       "      <td>2</td>\n",
       "      <td>P</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>REACH000766</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>F0290</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>REACH000767</td>\n",
       "      <td>30|30</td>\n",
       "      <td>0.26|0.26</td>\n",
       "      <td>0.88</td>\n",
       "      <td>F0291</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>Yes</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>REACH000768</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>F0291</td>\n",
       "      <td>2</td>\n",
       "      <td>P</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>No</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>REACH000769</td>\n",
       "      <td>27|27</td>\n",
       "      <td>0.07|0.07</td>\n",
       "      <td>0.48</td>\n",
       "      <td>F0291</td>\n",
       "      <td>1</td>\n",
       "      <td>P</td>\n",
       "      <td>27</td>\n",
       "      <td>27</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>No</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>REACH000770</td>\n",
       "      <td>30|30</td>\n",
       "      <td>0.26|0.26</td>\n",
       "      <td>0.98</td>\n",
       "      <td>F0291</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>No</td>\n",
       "      <td>PB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>265 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample_id     GB            Z  Qual Fam_id  sex rel GB_min GB_max  \\\n",
       "0    REACH000001  29|31    0.20|0.32  0.99  F0001    1   C     29     31   \n",
       "1    REACH000107   0|34   -1.64|0.51     1  F0001    2   P      0     34   \n",
       "2    REACH000226    0|4  -1.64|-1.38     1  F0001    1   P      0      4   \n",
       "3    REACH000026      .            .     .  F0026    1   C      .      .   \n",
       "4    REACH000269      .            .     .  F0026    2   P      .      .   \n",
       "..           ...    ...          ...   ...    ...  ...  ..    ...    ...   \n",
       "260  REACH000766      .            .     .  F0290    2   C      .      .   \n",
       "261  REACH000767  30|30    0.26|0.26  0.88  F0291    1   C     30     30   \n",
       "262  REACH000768      .            .     .  F0291    2   P      .      .   \n",
       "263  REACH000769  27|27    0.07|0.07  0.48  F0291    1   P     27     27   \n",
       "264  REACH000770  30|30    0.26|0.26  0.98  F0291    2   C     30     30   \n",
       "\n",
       "    repeats_min repeats_max Affected Platform  \n",
       "0            29          30      Yes      ONT  \n",
       "1            20          31      Yes      ONT  \n",
       "2            20          21      Yes      ONT  \n",
       "3             .           .      Yes       PB  \n",
       "4             .           .       No       PB  \n",
       "..          ...         ...      ...      ...  \n",
       "260           .           .       No       PB  \n",
       "261          30          30      Yes       PB  \n",
       "262           .           .       No       PB  \n",
       "263          29          29       No       PB  \n",
       "264          30          30       No       PB  \n",
       "\n",
       "[265 rows x 13 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### not needed really\n",
    "file_tr_gts = 'fragile_x/FMR1_repeat_gts.tsv'\n",
    "df_tr_gts = pd.read_table(file_tr_gts, header=0, sep='\\t')\n",
    "display(df_tr_gts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bacac553-dbf0-4ca2-b045-7da31f8507f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "### make a dataframe for each sample to represent each bp methylation in each region\n",
    "#pd.set_option('display.max_rows', 10)\n",
    "def get_bp_df(df_in, sample):\n",
    "    df_bps = pd.DataFrame()\n",
    "    read_hps = ['h1', 'h2', 'h0']\n",
    "    for idx in df_in.index:\n",
    "    #for idx in range(1):\n",
    "        region_id = df_in.loc[idx, 'region_id']\n",
    "        for read_hp in read_hps:\n",
    "            ### example: 0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69\n",
    "            m_region = df_in.loc[idx, f'{sample}_methyl_{read_hp}']\n",
    "            #print(f'm_region: {m_region}')\n",
    "\n",
    "            ### example bp_dev: 0,0\n",
    "            bp_dev_list = df_in.loc[idx, f'{sample}_bp_dev_{read_hp}'].split(',')\n",
    "            read_name_list = df_in.loc[idx, f'{sample}_rn_{read_hp}'].split(',')\n",
    "            \n",
    "            if m_region == '.':\n",
    "                continue\n",
    "                \n",
    "            m_reads = m_region.split(',')\n",
    "            #print(f'm_reads: {m_reads}')\n",
    "    \n",
    "            m_bps = []\n",
    "            read_names = []\n",
    "            bp_devs = []\n",
    "            hp_tags = []\n",
    "            tmp = [x.split('|') for x in m_reads] ### list of lists. the inner list values are the methylation probs\n",
    "            for i_x, x in enumerate(tmp):\n",
    "                m_bps.extend(x)\n",
    "                #read_ids.extend([f'{region_id}_{read_hp}_{i_x}' for i in range(len(x))])\n",
    "                read_names.extend([read_name_list[i_x] for i in range(len(x))])\n",
    "                bp_devs.extend([bp_dev_list[i_x] for i in range(len(x))])\n",
    "                hp_tags.extend([read_hp for i in range(len(x))])\n",
    "            #print(f'm_bps: {m_bps}')\n",
    "            #print(f'read_ids: {read_ids}')\n",
    "    \n",
    "            df_tmp = pd.DataFrame()\n",
    "            df_tmp['region_id'] = [region_id for x in m_bps]\n",
    "            df_tmp['methylation'] = m_bps\n",
    "            df_tmp['read_name'] = read_names\n",
    "            df_tmp['bp_dev'] = bp_devs\n",
    "            df_tmp['hp_tag'] = hp_tags\n",
    "    \n",
    "            df_bps = pd.concat([df_bps, df_tmp], ignore_index=True)\n",
    "\n",
    "    if df_bps.shape[0]>0:\n",
    "        df_bps['methylation'] = df_bps['methylation'].astype(int)\n",
    "        df_bps = df_bps.loc[df_bps.bp_dev!='.']\n",
    "        df_bps['bp_dev'] = df_bps['bp_dev'].astype(int)\n",
    "        df_bps['subject'] = [sample for i in range(df_bps.shape[0])]\n",
    "        #df_bps['haplotype'] = df_bps.bp_dev.apply(lambda x: 'PM' if x>=bp_dev_thr else 'N')\n",
    "        #df_bps['haplotype'] = df_bps.read_name.apply(lambda x: read_haplotype_dict[x] if x in read_haplotype_dict else 'NA')\n",
    "        #df_bps = df_bps.loc[df_bps.haplotype!='NA']\n",
    "        #display(df_bps)\n",
    "    return df_bps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "347a5a54-ede1-4422-ac88-b653fa43d29b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### not needed anymore\n",
    "#file_fm_offsp_pb = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/METHYLATION/PB_SAMPLES/select_samples/pb_female_kids_cov_lt_2.txt'\n",
    "#file_fm_offsp_ont = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/METHYLATION/ONT_SAMPLES/01_pod5s/prior_samples/ont_female_kids.txt'\n",
    "#df_sam_pb = pd.read_table(file_fm_offsp_pb, header=None)\n",
    "#print(f'number of pb samples: {df_sam_pb.shape[0]}')\n",
    "#df_sam_ont = pd.read_table(file_fm_offsp_ont, header=None)\n",
    "#print(f'number of pb samples: {len([x for x in df_sam_ont[0].tolist() if x not in [\"REACH000592\", \"REACH000409\"]])}')\n",
    "#samples = df_sam_pb[0].tolist() + [x for x in df_sam_ont[0].tolist() if x not in ['REACH000592', 'REACH000409']]\n",
    "#print(f'number of samples: {len(samples)}')\n",
    "#print(samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "72fc97c3-a06b-4a39-a81d-0079bc10b3e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of samples: 244\n",
      "['REACH000026', 'REACH000058', 'REACH000065', 'REACH000086', 'REACH000089', 'REACH000092', 'REACH000095', 'REACH000097', 'REACH000120', 'REACH000159', 'REACH000162', 'REACH000182', 'REACH000185', 'REACH000194', 'REACH000197', 'REACH000214', 'REACH000217', 'REACH000219', 'REACH000222', 'REACH000224', 'REACH000229', 'REACH000232', 'REACH000233', 'REACH000246', 'REACH000284', 'REACH000287', 'REACH000288', 'REACH000291', 'REACH000292', 'REACH000293', 'REACH000316', 'REACH000353', 'REACH000356', 'REACH000364', 'REACH000367', 'REACH000441', 'REACH000450', 'REACH000453', 'REACH000525', 'REACH000528', 'REACH000537', 'REACH000630', 'REACH000640', 'REACH000643', 'REACH000644', 'REACH000650', 'REACH000653', 'REACH000654', 'REACH000671', 'REACH000674', 'REACH000701', 'REACH000705', 'REACH000708', 'REACH000727', 'REACH000730', 'REACH000747', 'REACH000750', 'REACH000751', 'REACH000754', 'REACH000762', 'REACH000765', 'REACH000766', 'REACH000767', 'REACH000770', 'REACH000154', 'REACH000400', 'REACH000403', 'REACH000404', 'REACH000412', 'REACH000413', 'REACH000414', 'REACH000417', 'REACH000418', 'REACH000419', 'REACH000422', 'REACH000423', 'REACH000426', 'REACH000427', 'REACH000428', 'REACH000435', 'REACH000448', 'REACH000476', 'REACH000479', 'REACH000482', 'REACH000483', 'REACH000501', 'REACH000510', 'REACH000513', 'REACH000514', 'REACH000517', 'REACH000518', 'REACH000519', 'REACH000529', 'REACH000532_ONT', 'REACH000540', 'REACH000543', 'REACH000544', 'REACH000547', 'REACH000555', 'REACH000558', 'REACH000559', 'REACH000560', 'REACH000563', 'REACH000564', 'REACH000567', 'REACH000568', 'REACH000578', 'REACH000581', 'REACH000602', 'REACH000626', 'REACH000660', 'REACH000663', 'REACH000681', 'REACH000684', 'REACH000685', 'REACH000503', 'REACH000642', 'REACH000366', 'REACH000646', 'REACH000707', 'REACH000562', 'REACH000156', 'REACH000542', 'REACH000440', 'REACH000481', 'REACH000557', 'REACH000099', 'REACH000749', 'REACH000662', 'REACH000295', 'REACH000067', 'REACH000094', 'REACH000442', 'REACH000566', 'REACH000248', 'REACH000402', 'REACH000161', 'REACH000437', 'REACH000652', 'REACH000756', 'REACH000355', 'REACH000753', 'REACH000703', 'REACH000539', 'REACH000769', 'REACH000270', 'REACH000452', 'REACH000764', 'REACH000524', 'REACH000088', 'REACH000315', 'REACH000516', 'REACH000196', 'REACH000683', 'REACH000580', 'REACH000122', 'REACH000411', 'REACH000231', 'REACH000629', 'REACH000290', 'REACH000478', 'REACH000425', 'REACH000268', 'REACH000235', 'REACH000184', 'REACH000421', 'REACH000216', 'REACH000546', 'REACH000729', 'REACH000286', 'REACH000673', 'REACH000531_ONT', 'REACH000604', 'REACH000527', 'REACH000512', 'REACH000221', 'REACH000416', 'REACH000565', 'REACH000401', 'REACH000121', 'REACH000556', 'REACH000728', 'REACH000511', 'REACH000451', 'REACH000561', 'REACH000220', 'REACH000641', 'REACH000183', 'REACH000230', 'REACH000706', 'REACH000502', 'REACH000755', 'REACH000247', 'REACH000768', 'REACH000538', 'REACH000289', 'REACH000087', 'REACH000603', 'REACH000294', 'REACH000269', 'REACH000093', 'REACH000314', 'REACH000410', 'REACH000415', 'REACH000420', 'REACH000424', 'REACH000436_PB', 'REACH000267', 'REACH000702', 'REACH000477', 'REACH000523', 'REACH000439', 'REACH000752', 'REACH000545', 'REACH000661', 'REACH000515', 'REACH000628', 'REACH000682', 'REACH000748', 'REACH000480', 'REACH000215', 'REACH000234', 'REACH000098', 'REACH000365', 'REACH000645', 'REACH000107', 'REACH000579', 'REACH000066', 'REACH000354', 'REACH000160', 'REACH000526', 'REACH000155', 'REACH000672', 'REACH000763', 'REACH000541', 'REACH000530_PB', 'REACH000449', 'REACH000651', 'REACH000195', 'REACH000285', 'REACH000226_blood', 'REACH000226_sperm', 'REACH000226_blood', 'REACH000226_sperm']\n",
      "df_bps:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210853 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            region_id  methylation                             read_name  \\\n",
       "0       TMEM185A_5UTR            5   m54340U_200716_001732/174262380/ccs   \n",
       "1       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "2       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "3       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "4       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "...               ...          ...                                   ...   \n",
       "210848     HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210849     HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210850     HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210851     HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210852     HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "\n",
       "        bp_dev hp_tag            subject rel  sex  coverage  \n",
       "0          -21     h0        REACH000026   C    1      3.22  \n",
       "1          -21     h0        REACH000026   C    1      3.22  \n",
       "2          -21     h0        REACH000026   C    1      3.22  \n",
       "3          -21     h0        REACH000026   C    1      3.22  \n",
       "4          -21     h0        REACH000026   C    1      3.22  \n",
       "...        ...    ...                ...  ..  ...       ...  \n",
       "210848       1     h0  REACH000226_sperm   D    1     27.87  \n",
       "210849       1     h0  REACH000226_sperm   D    1     27.87  \n",
       "210850      -4     h0  REACH000226_sperm   D    1     27.87  \n",
       "210851      -4     h0  REACH000226_sperm   D    1     27.87  \n",
       "210852      -4     h0  REACH000226_sperm   D    1     27.87  \n",
       "\n",
       "[210853 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "exclude_samples = ['REACH000592', 'REACH000409', 'REACH000001', 'REACH000226']\n",
    "samples = [x for x in df_samples.sample_id.tolist() if x not in exclude_samples] + ['REACH000226_blood', 'REACH000226_sperm']\n",
    "print(f'number of samples: {len(samples)}')\n",
    "print(samples)\n",
    "\n",
    "df_bps = pd.DataFrame()\n",
    "for sample in samples:\n",
    "    df_tmp = get_bp_df(df_exnd, sample)\n",
    "    df_bps = pd.concat([df_bps, df_tmp], ignore_index=True)\n",
    "df_bps['rel'] = df_bps.subject.map(df_samples.set_index('sample_id').rel)\n",
    "df_bps['sex'] = df_bps.subject.map(df_samples.set_index('sample_id').sex)\n",
    "df_bps['coverage'] = df_bps.subject.map(df_samples.set_index('sample_id').coverage)\n",
    "print('df_bps:')\n",
    "display(df_bps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0901d9fa-8854-485f-b146-0cea8e26e53d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_status(row, bp_dev_col, ignore_region=False):\n",
    "    if row['region_id'] != 'FMR1_5UTR' and not ignore_region:\n",
    "        status = 'XCI (other genes)'\n",
    "    else:\n",
    "        status = 'NA'\n",
    "        rep = round(row[bp_dev_col] / 3) + 20\n",
    "        rep_thrs = [20, 30, 40]\n",
    "        if rep <= rep_thrs[0]:\n",
    "            status = f'repeats<{rep_thrs[0]}'\n",
    "        for idx, rep_thr in enumerate(rep_thrs[:-1]):\n",
    "            if rep > rep_thr and rep <= rep_thrs[idx+1]:\n",
    "                status = f'{rep_thr}<repeats<{rep_thrs[idx+1]}'\n",
    "        if rep > rep_thrs[-1]:\n",
    "            status = f'{rep_thrs[-1]}<repeats'\n",
    "    return status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7ff8c01b-c5eb-488a-9438-50a2bbe8d986",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_bps:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210853 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            region_id  methylation                             read_name  \\\n",
       "0       TMEM185A_5UTR            5   m54340U_200716_001732/174262380/ccs   \n",
       "1       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "2       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "3       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "4       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "...               ...          ...                                   ...   \n",
       "210848     HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210849     HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210850     HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210851     HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210852     HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "\n",
       "        bp_dev hp_tag            subject rel  sex  coverage      bp_rep_status  \n",
       "0          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "1          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "2          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "3          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "4          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "...        ...    ...                ...  ..  ...       ...                ...  \n",
       "210848       1     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210849       1     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210850      -4     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210851      -4     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210852      -4     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "\n",
       "[210853 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_bps['bp_rep_status'] = df_bps.apply(lambda row: get_status(row, 'bp_dev'), axis=1)\n",
    "print('df_bps:')\n",
    "display(df_bps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "263ad084-a511-4d65-8a95-e6270fe2e27a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "corrected df_bps for R563 and FMR1_5UTR:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>83641</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>3dc80ce9-433b-4051-9a5a-52005e0d3627</td>\n",
       "      <td>86</td>\n",
       "      <td>h1</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>40&lt;repeats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83642</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>3dc80ce9-433b-4051-9a5a-52005e0d3627</td>\n",
       "      <td>86</td>\n",
       "      <td>h1</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>40&lt;repeats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83643</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>20</td>\n",
       "      <td>3dc80ce9-433b-4051-9a5a-52005e0d3627</td>\n",
       "      <td>86</td>\n",
       "      <td>h1</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>40&lt;repeats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83644</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>97</td>\n",
       "      <td>3dc80ce9-433b-4051-9a5a-52005e0d3627</td>\n",
       "      <td>86</td>\n",
       "      <td>h1</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>40&lt;repeats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83645</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>80</td>\n",
       "      <td>3dc80ce9-433b-4051-9a5a-52005e0d3627</td>\n",
       "      <td>86</td>\n",
       "      <td>h1</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>40&lt;repeats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84040</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>21</td>\n",
       "      <td>be004428-de04-4e9a-bb2f-3f5f0e54e1a1</td>\n",
       "      <td>53</td>\n",
       "      <td>h2</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84041</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>88</td>\n",
       "      <td>be004428-de04-4e9a-bb2f-3f5f0e54e1a1</td>\n",
       "      <td>53</td>\n",
       "      <td>h2</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84042</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>36</td>\n",
       "      <td>be004428-de04-4e9a-bb2f-3f5f0e54e1a1</td>\n",
       "      <td>53</td>\n",
       "      <td>h2</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84043</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>25</td>\n",
       "      <td>be004428-de04-4e9a-bb2f-3f5f0e54e1a1</td>\n",
       "      <td>53</td>\n",
       "      <td>h2</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84044</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>18</td>\n",
       "      <td>be004428-de04-4e9a-bb2f-3f5f0e54e1a1</td>\n",
       "      <td>53</td>\n",
       "      <td>h2</td>\n",
       "      <td>REACH000563</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>8.68</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>404 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       region_id  methylation                             read_name  bp_dev  \\\n",
       "83641  FMR1_5UTR            0  3dc80ce9-433b-4051-9a5a-52005e0d3627      86   \n",
       "83642  FMR1_5UTR            0  3dc80ce9-433b-4051-9a5a-52005e0d3627      86   \n",
       "83643  FMR1_5UTR           20  3dc80ce9-433b-4051-9a5a-52005e0d3627      86   \n",
       "83644  FMR1_5UTR           97  3dc80ce9-433b-4051-9a5a-52005e0d3627      86   \n",
       "83645  FMR1_5UTR           80  3dc80ce9-433b-4051-9a5a-52005e0d3627      86   \n",
       "...          ...          ...                                   ...     ...   \n",
       "84040  FMR1_5UTR           21  be004428-de04-4e9a-bb2f-3f5f0e54e1a1      53   \n",
       "84041  FMR1_5UTR           88  be004428-de04-4e9a-bb2f-3f5f0e54e1a1      53   \n",
       "84042  FMR1_5UTR           36  be004428-de04-4e9a-bb2f-3f5f0e54e1a1      53   \n",
       "84043  FMR1_5UTR           25  be004428-de04-4e9a-bb2f-3f5f0e54e1a1      53   \n",
       "84044  FMR1_5UTR           18  be004428-de04-4e9a-bb2f-3f5f0e54e1a1      53   \n",
       "\n",
       "      hp_tag      subject rel  sex  coverage  bp_rep_status  \n",
       "83641     h1  REACH000563   C    2      8.68     40<repeats  \n",
       "83642     h1  REACH000563   C    2      8.68     40<repeats  \n",
       "83643     h1  REACH000563   C    2      8.68     40<repeats  \n",
       "83644     h1  REACH000563   C    2      8.68     40<repeats  \n",
       "83645     h1  REACH000563   C    2      8.68     40<repeats  \n",
       "...      ...          ...  ..  ...       ...            ...  \n",
       "84040     h2  REACH000563   C    2      8.68  30<repeats<40  \n",
       "84041     h2  REACH000563   C    2      8.68  30<repeats<40  \n",
       "84042     h2  REACH000563   C    2      8.68  30<repeats<40  \n",
       "84043     h2  REACH000563   C    2      8.68  30<repeats<40  \n",
       "84044     h2  REACH000563   C    2      8.68  30<repeats<40  \n",
       "\n",
       "[404 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_bps:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>h0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210853 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            region_id  methylation                             read_name  \\\n",
       "0       TMEM185A_5UTR            5   m54340U_200716_001732/174262380/ccs   \n",
       "1       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "2       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "3       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "4       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "...               ...          ...                                   ...   \n",
       "210848     HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210849     HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210850     HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210851     HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210852     HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "\n",
       "        bp_dev hp_tag            subject rel  sex  coverage      bp_rep_status  \n",
       "0          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "1          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "2          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "3          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "4          -21     h0        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "...        ...    ...                ...  ..  ...       ...                ...  \n",
       "210848       1     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210849       1     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210850      -4     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210851      -4     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210852      -4     h0  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "\n",
       "[210853 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#### correct FMR1 repeat hp_tag for REACH000563. The hp_tag is inferred from bp_dev with a bp_dev_thr\n",
    "pd.set_option('display.max_rows', 10)\n",
    "bp_dev_thr = 60\n",
    "#df_bps.loc[(df_bps.subject=='REACH000563')&(df_bps.region_id=='FMR1_5UTR')]['hp_tag'] = df_bps.loc[((df_bps.subject=='REACH000563')&\n",
    "#                                                                                                    (df_bps.region_id=='FMR1_5UTR'))].apply(lambda row: 'h1' if row['bp_dev']>bp_dev_thr else 'h2', axis=1)\n",
    "def correct_hp(row, bp_dev_thr):\n",
    "    hp = row['hp_tag']\n",
    "    if row['subject'] == 'REACH000563' and row['region_id'] == 'FMR1_5UTR':\n",
    "        if row['bp_dev'] >= bp_dev_thr:\n",
    "            hp = 'h1'\n",
    "        else:\n",
    "            hp = 'h2'\n",
    "    return hp\n",
    "df_bps['hp_tag'] = df_bps.apply(lambda row: correct_hp(row, bp_dev_thr), axis=1)\n",
    "print('corrected df_bps for R563 and FMR1_5UTR:')\n",
    "display(df_bps.loc[(df_bps.subject=='REACH000563')&(df_bps.region_id=='FMR1_5UTR')])\n",
    "print('df_bps:')\n",
    "display(df_bps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "51589bb7-bfad-494e-9d7c-a64cc8b10b1b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_bps:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210853 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            region_id  methylation                             read_name  \\\n",
       "0       TMEM185A_5UTR            5   m54340U_200716_001732/174262380/ccs   \n",
       "1       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "2       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "3       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "4       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "...               ...          ...                                   ...   \n",
       "210848     HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210849     HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210850     HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210851     HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210852     HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "\n",
       "        bp_dev hp_tag            subject rel  sex  coverage      bp_rep_status  \n",
       "0          -21     hm        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "1          -21     hm        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "2          -21     hm        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "3          -21     hm        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "4          -21     hm        REACH000026   C    1      3.22  XCI (other genes)  \n",
       "...        ...    ...                ...  ..  ...       ...                ...  \n",
       "210848       1     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210849       1     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210850      -4     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210851      -4     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "210852      -4     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)  \n",
       "\n",
       "[210853 rows x 10 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#### correct male haplotypes to hemizygous: hm\n",
    "pd.set_option('display.max_rows', 10)\n",
    "def correct_hp_male(row):\n",
    "    hp = row['hp_tag']\n",
    "    sex = row['sex']\n",
    "    if sex == 1:\n",
    "        hp = 'hm'\n",
    "    return hp\n",
    "df_bps['hp_tag'] = df_bps.apply(lambda row: correct_hp_male(row), axis=1)\n",
    "print('df_bps:')\n",
    "display(df_bps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0e829e25-f6e6-4e7a-9a93-3726fd01fafd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of samples: 242\n",
      "{'REACH000026': {}, 'REACH000058': {'hm': 51}, 'REACH000065': {'hm': 30}, 'REACH000086': {'hm': 30}, 'REACH000089': {'h1': 30, 'h2': 30}, 'REACH000092': {'h1': 34, 'h2': 27}, 'REACH000095': {'h0': 4}, 'REACH000097': {'hm': 30}, 'REACH000120': {'hm': 11}, 'REACH000159': {}, 'REACH000162': {}, 'REACH000182': {'hm': 30}, 'REACH000185': {'hm': 31}, 'REACH000194': {}, 'REACH000197': {}, 'REACH000214': {'hm': 30}, 'REACH000217': {'h0': 2}, 'REACH000219': {'h0': 30}, 'REACH000222': {'hm': 27}, 'REACH000224': {}, 'REACH000229': {}, 'REACH000232': {}, 'REACH000233': {'hm': 30}, 'REACH000246': {'hm': 32}, 'REACH000284': {'h0': 16}, 'REACH000287': {'hm': 15}, 'REACH000288': {'hm': 30}, 'REACH000291': {'hm': 30}, 'REACH000292': {'hm': 30}, 'REACH000293': {'hm': 9}, 'REACH000316': {'h2': 28}, 'REACH000353': {}, 'REACH000356': {'hm': 30}, 'REACH000364': {'hm': 27}, 'REACH000367': {'h0': 27}, 'REACH000441': {'hm': 51}, 'REACH000450': {}, 'REACH000453': {'h1': 33, 'h2': 90}, 'REACH000525': {}, 'REACH000528': {'h0': 3}, 'REACH000537': {'hm': 15}, 'REACH000630': {'h0': 30}, 'REACH000640': {'h0': 30}, 'REACH000643': {'hm': 34}, 'REACH000644': {'hm': 31}, 'REACH000650': {'hm': 28}, 'REACH000653': {'hm': 54}, 'REACH000654': {'h1': 30, 'h2': 27}, 'REACH000671': {'hm': 30}, 'REACH000674': {'h1': 30, 'h2': 30}, 'REACH000701': {}, 'REACH000705': {'hm': 36}, 'REACH000708': {'hm': 36}, 'REACH000727': {'h0': 33, 'h1': 31}, 'REACH000730': {'hm': 36}, 'REACH000747': {'hm': 34}, 'REACH000750': {}, 'REACH000751': {}, 'REACH000754': {'hm': 30}, 'REACH000762': {'hm': 35}, 'REACH000765': {}, 'REACH000766': {'h2': 33}, 'REACH000767': {'hm': 30}, 'REACH000770': {'h0': 33}, 'REACH000154': {'hm': 35}, 'REACH000400': {'hm': 8}, 'REACH000403': {'hm': -4}, 'REACH000404': {'h0': -12, 'h1': 26, 'h2': -2}, 'REACH000412': {'h0': 32}, 'REACH000413': {'h0': 44}, 'REACH000414': {'hm': 23}, 'REACH000417': {'h0': 30, 'h1': 23, 'h2': 26}, 'REACH000418': {'h1': 36, 'h2': 28}, 'REACH000419': {'hm': 28}, 'REACH000422': {'hm': 22}, 'REACH000423': {'hm': 28}, 'REACH000426': {'hm': 28}, 'REACH000427': {'hm': 25}, 'REACH000428': {'hm': 29}, 'REACH000435': {'h0': 21, 'h1': 22, 'h2': 5}, 'REACH000448': {'hm': 20}, 'REACH000476': {'hm': 29}, 'REACH000479': {'h0': 1, 'h1': 48, 'h2': -4}, 'REACH000482': {'h0': 41, 'h1': 36, 'h2': 0}, 'REACH000483': {}, 'REACH000501': {'hm': 30}, 'REACH000510': {'h0': 27, 'h1': 36, 'h2': 38}, 'REACH000513': {'h0': 33, 'h1': 32, 'h2': 29}, 'REACH000514': {'hm': 23}, 'REACH000517': {'hm': 24}, 'REACH000518': {'h0': 25, 'h1': 28, 'h2': 27}, 'REACH000519': {'h0': 21, 'h1': 58, 'h2': 2}, 'REACH000529': {'hm': 26}, 'REACH000532_ONT': {'hm': 32}, 'REACH000540': {'hm': 26}, 'REACH000543': {'h0': 32}, 'REACH000544': {'hm': 28}, 'REACH000547': {'h1': 23, 'h2': 42}, 'REACH000555': {'h0': 28}, 'REACH000558': {'hm': 28}, 'REACH000559': {'hm': 33}, 'REACH000560': {'hm': 63}, 'REACH000563': {'h1': 88, 'h2': 37}, 'REACH000564': {'hm': 37}, 'REACH000567': {'h0': 26, 'h1': 32, 'h2': 30}, 'REACH000568': {'h0': 30, 'h1': 29}, 'REACH000578': {'h1': 26, 'h2': 17}, 'REACH000581': {'h0': 26, 'h1': 24, 'h2': 12}, 'REACH000602': {'h0': 12}, 'REACH000626': {'hm': 21}, 'REACH000660': {'h0': 28, 'h1': 24, 'h2': 18}, 'REACH000663': {'hm': 42}, 'REACH000681': {'hm': -6}, 'REACH000684': {'hm': 24}, 'REACH000685': {'hm': 17}, 'REACH000503': {'hm': 22}, 'REACH000642': {'hm': 30}, 'REACH000366': {'hm': 27}, 'REACH000646': {'hm': 30}, 'REACH000707': {'hm': 0}, 'REACH000562': {'hm': 14}, 'REACH000156': {'hm': 22}, 'REACH000542': {'hm': 23}, 'REACH000440': {'hm': 32}, 'REACH000481': {'hm': 37}, 'REACH000557': {'hm': 26}, 'REACH000099': {'hm': 32}, 'REACH000749': {'hm': 27}, 'REACH000662': {'hm': 22}, 'REACH000295': {'hm': 35}, 'REACH000067': {}, 'REACH000094': {'hm': 34}, 'REACH000442': {'hm': 27}, 'REACH000566': {'hm': 27}, 'REACH000248': {}, 'REACH000402': {'hm': 26}, 'REACH000161': {'hm': 0}, 'REACH000437': {'hm': 16}, 'REACH000652': {}, 'REACH000756': {}, 'REACH000355': {}, 'REACH000753': {'hm': 30}, 'REACH000703': {}, 'REACH000539': {'hm': 32}, 'REACH000769': {'hm': 27}, 'REACH000270': {}, 'REACH000452': {}, 'REACH000764': {'hm': 0}, 'REACH000524': {'hm': 39}, 'REACH000088': {'hm': 32}, 'REACH000315': {'hm': 30}, 'REACH000516': {'hm': 28}, 'REACH000196': {'hm': 29}, 'REACH000683': {'hm': 28}, 'REACH000580': {'hm': 23}, 'REACH000122': {'hm': 30}, 'REACH000411': {'hm': 26}, 'REACH000231': {}, 'REACH000629': {'hm': 30}, 'REACH000290': {'hm': 27}, 'REACH000478': {'hm': 6}, 'REACH000425': {'hm': 29}, 'REACH000268': {}, 'REACH000235': {'hm': 27}, 'REACH000184': {'hm': 26}, 'REACH000421': {'hm': 30}, 'REACH000216': {'hm': -1}, 'REACH000546': {'hm': 23}, 'REACH000729': {}, 'REACH000286': {'hm': 7}, 'REACH000673': {'hm': 30}, 'REACH000531_ONT': {}, 'REACH000604': {'hm': -6}, 'REACH000527': {'hm': 30}, 'REACH000512': {'hm': 20}, 'REACH000221': {'hm': 36}, 'REACH000416': {'hm': 29}, 'REACH000565': {'h0': 34, 'h1': 28, 'h2': 37}, 'REACH000401': {'h0': 21, 'h1': -5, 'h2': 14}, 'REACH000121': {'h0': 14, 'h2': 10}, 'REACH000556': {'h0': 27, 'h1': 20}, 'REACH000728': {}, 'REACH000511': {'h0': 32, 'h1': 10}, 'REACH000451': {'h1': 27, 'h2': 86}, 'REACH000561': {'h1': 25, 'h2': 90}, 'REACH000220': {'h1': 27, 'h2': 27}, 'REACH000641': {'h1': 32, 'h2': 32}, 'REACH000183': {'h0': 32}, 'REACH000230': {'h1': 30, 'h2': 0}, 'REACH000706': {'h0': 36, 'h2': 36}, 'REACH000502': {'h0': 23, 'h2': 27}, 'REACH000755': {'h0': 0, 'h1': 0}, 'REACH000247': {'h0': 34}, 'REACH000768': {}, 'REACH000538': {'h0': 33}, 'REACH000289': {'h0': 30, 'h1': 30, 'h2': 30}, 'REACH000087': {'h1': 30, 'h2': 29}, 'REACH000603': {'h0': 33}, 'REACH000294': {'h1': 7, 'h2': 9}, 'REACH000269': {}, 'REACH000093': {'h1': 27}, 'REACH000314': {'h0': 38, 'h1': 28, 'h2': 34}, 'REACH000410': {'h0': 3}, 'REACH000415': {'h0': 30, 'h2': 33}, 'REACH000420': {'h0': 20, 'h1': -1, 'h2': 17}, 'REACH000424': {'h0': 28, 'h1': 16, 'h2': 26}, 'REACH000436_PB': {'h1': 10, 'h2': 30}, 'REACH000267': {'h1': 27, 'h2': 30}, 'REACH000702': {'h1': 27, 'h2': 30}, 'REACH000477': {'h0': 34, 'h1': 28, 'h2': 24}, 'REACH000523': {'h1': 10, 'h2': 26}, 'REACH000439': {'h0': 50}, 'REACH000752': {'h1': 38}, 'REACH000545': {}, 'REACH000661': {'h0': 30, 'h1': 20, 'h2': 32}, 'REACH000515': {'h0': 34, 'h1': 19, 'h2': 25}, 'REACH000628': {'h1': 30, 'h2': 27}, 'REACH000682': {'h0': 26, 'h1': 28, 'h2': 13}, 'REACH000748': {}, 'REACH000480': {'h0': 14}, 'REACH000215': {'h0': 32}, 'REACH000234': {'h1': 27, 'h2': 32}, 'REACH000098': {'h0': 31, 'h1': 36}, 'REACH000365': {'h1': 30, 'h2': 27}, 'REACH000645': {'h1': 33}, 'REACH000107': {'h0': 26, 'h1': 26, 'h2': 26}, 'REACH000579': {'h1': 38, 'h2': 25}, 'REACH000066': {'h1': 32, 'h2': 31}, 'REACH000354': {'h0': 30}, 'REACH000160': {'h0': 12}, 'REACH000526': {'h0': 3}, 'REACH000155': {'h0': 37, 'h2': 36}, 'REACH000672': {'h1': 30, 'h2': 30}, 'REACH000763': {'h2': 32}, 'REACH000541': {'h0': 32, 'h1': 28, 'h2': 19}, 'REACH000530_PB': {'h0': 36, 'h2': 34}, 'REACH000449': {'h0': 30}, 'REACH000651': {'h1': 52, 'h2': 27}, 'REACH000195': {'h1': 36, 'h2': 9}, 'REACH000285': {'h1': 37, 'h2': 15}, 'REACH000226_blood': {}, 'REACH000226_sperm': {'hm': 0}}\n",
      "number of samples with both h1 and h2: 53\n",
      "{'REACH000089': {'h1': 30, 'h2': 30}, 'REACH000092': {'h1': 34, 'h2': 27}, 'REACH000453': {'h1': 33, 'h2': 90}, 'REACH000654': {'h1': 30, 'h2': 27}, 'REACH000674': {'h1': 30, 'h2': 30}, 'REACH000404': {'h0': -12, 'h1': 26, 'h2': -2}, 'REACH000417': {'h0': 30, 'h1': 23, 'h2': 26}, 'REACH000418': {'h1': 36, 'h2': 28}, 'REACH000435': {'h0': 21, 'h1': 22, 'h2': 5}, 'REACH000479': {'h0': 1, 'h1': 48, 'h2': -4}, 'REACH000482': {'h0': 41, 'h1': 36, 'h2': 0}, 'REACH000510': {'h0': 27, 'h1': 36, 'h2': 38}, 'REACH000513': {'h0': 33, 'h1': 32, 'h2': 29}, 'REACH000518': {'h0': 25, 'h1': 28, 'h2': 27}, 'REACH000519': {'h0': 21, 'h1': 58, 'h2': 2}, 'REACH000547': {'h1': 23, 'h2': 42}, 'REACH000563': {'h1': 88, 'h2': 37}, 'REACH000567': {'h0': 26, 'h1': 32, 'h2': 30}, 'REACH000578': {'h1': 26, 'h2': 17}, 'REACH000581': {'h0': 26, 'h1': 24, 'h2': 12}, 'REACH000660': {'h0': 28, 'h1': 24, 'h2': 18}, 'REACH000565': {'h0': 34, 'h1': 28, 'h2': 37}, 'REACH000401': {'h0': 21, 'h1': -5, 'h2': 14}, 'REACH000451': {'h1': 27, 'h2': 86}, 'REACH000561': {'h1': 25, 'h2': 90}, 'REACH000220': {'h1': 27, 'h2': 27}, 'REACH000641': {'h1': 32, 'h2': 32}, 'REACH000230': {'h1': 30, 'h2': 0}, 'REACH000289': {'h0': 30, 'h1': 30, 'h2': 30}, 'REACH000087': {'h1': 30, 'h2': 29}, 'REACH000294': {'h1': 7, 'h2': 9}, 'REACH000314': {'h0': 38, 'h1': 28, 'h2': 34}, 'REACH000420': {'h0': 20, 'h1': -1, 'h2': 17}, 'REACH000424': {'h0': 28, 'h1': 16, 'h2': 26}, 'REACH000436_PB': {'h1': 10, 'h2': 30}, 'REACH000267': {'h1': 27, 'h2': 30}, 'REACH000702': {'h1': 27, 'h2': 30}, 'REACH000477': {'h0': 34, 'h1': 28, 'h2': 24}, 'REACH000523': {'h1': 10, 'h2': 26}, 'REACH000661': {'h0': 30, 'h1': 20, 'h2': 32}, 'REACH000515': {'h0': 34, 'h1': 19, 'h2': 25}, 'REACH000628': {'h1': 30, 'h2': 27}, 'REACH000682': {'h0': 26, 'h1': 28, 'h2': 13}, 'REACH000234': {'h1': 27, 'h2': 32}, 'REACH000365': {'h1': 30, 'h2': 27}, 'REACH000107': {'h0': 26, 'h1': 26, 'h2': 26}, 'REACH000579': {'h1': 38, 'h2': 25}, 'REACH000066': {'h1': 32, 'h2': 31}, 'REACH000672': {'h1': 30, 'h2': 30}, 'REACH000541': {'h0': 32, 'h1': 28, 'h2': 19}, 'REACH000651': {'h1': 52, 'h2': 27}, 'REACH000195': {'h1': 36, 'h2': 9}, 'REACH000285': {'h1': 37, 'h2': 15}}\n"
     ]
    }
   ],
   "source": [
    "this_df = df_bps.loc[df_bps.region_id=='FMR1_5UTR']\n",
    "#display(this_df)\n",
    "hp_info_dict = {}\n",
    "#for sample in samples[:1]:\n",
    "for sample in samples:\n",
    "    #print(sample)\n",
    "    hp_info_dict[sample] = {}\n",
    "    df_tmp = this_df.loc[this_df.subject==sample]\n",
    "    #print(df_tmp.shape)\n",
    "        \n",
    "    ### reduce redundant bps to each read/hp\n",
    "    collapse_1 = df_tmp[['read_name', 'hp_tag', 'bp_dev']].groupby(by=['read_name', 'hp_tag']).median()\n",
    "    #print(collapse_1)\n",
    "    ### take the median of bp in each hp\n",
    "    collapse_2 = collapse_1.groupby('hp_tag').median()\n",
    "    #print(collapse_2)\n",
    "    hps = collapse_2.index.tolist()\n",
    "    for hp in hps:\n",
    "        hp_info_dict[sample][hp] = round(collapse_2.loc[hp,'bp_dev'])\n",
    "    #print(collapse_2.index.tolist())\n",
    "    #print(df_tmp[['read_name', 'hp_tag', 'bp_dev']].groupby(by=['read_name', 'hp_tag']).mean().groupby('hp_tag').median())\n",
    "    #print(df_tmp[['hp_tag', 'bp_dev']].groupby(by='hp_tag').median())\n",
    "print(f'number of samples: {len(hp_info_dict)}')\n",
    "print(hp_info_dict)\n",
    "\n",
    "### thi is used for long vs short plots\n",
    "hp_info_dict_comp = {}\n",
    "for sample in hp_info_dict:\n",
    "    if 'h1' in hp_info_dict[sample] and 'h2' in hp_info_dict[sample]:\n",
    "        hp_info_dict_comp[sample] = hp_info_dict[sample]\n",
    "print(f'number of samples with both h1 and h2: {len(hp_info_dict_comp)}')\n",
    "print(hp_info_dict_comp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e348c77d-0ce2-494f-ac8d-310bb4857c9f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'REACH000026': {}, 'REACH000058': {'hm': 51}, 'REACH000086': {'hm': 30}, 'REACH000097': {'hm': 30}, 'REACH000120': {'hm': 11}, 'REACH000159': {}, 'REACH000162': {}, 'REACH000182': {'hm': 30}, 'REACH000185': {'hm': 31}, 'REACH000194': {}, 'REACH000197': {}, 'REACH000214': {'hm': 30}, 'REACH000222': {'hm': 27}, 'REACH000233': {'hm': 30}, 'REACH000287': {'hm': 15}, 'REACH000288': {'hm': 30}, 'REACH000291': {'hm': 30}, 'REACH000292': {'hm': 30}, 'REACH000293': {'hm': 9}, 'REACH000356': {'hm': 30}, 'REACH000441': {'hm': 51}, 'REACH000537': {'hm': 15}, 'REACH000643': {'hm': 34}, 'REACH000650': {'hm': 28}, 'REACH000653': {'hm': 54}, 'REACH000671': {'hm': 30}, 'REACH000708': {'hm': 36}, 'REACH000730': {'hm': 36}, 'REACH000747': {'hm': 34}, 'REACH000750': {}, 'REACH000762': {'hm': 35}, 'REACH000767': {'hm': 30}, 'REACH000154': {'hm': 35}, 'REACH000400': {'hm': 8}, 'REACH000403': {'hm': -4}, 'REACH000414': {'hm': 23}, 'REACH000419': {'hm': 28}, 'REACH000422': {'hm': 22}, 'REACH000423': {'hm': 28}, 'REACH000426': {'hm': 28}, 'REACH000427': {'hm': 25}, 'REACH000428': {'hm': 29}, 'REACH000448': {'hm': 20}, 'REACH000476': {'hm': 29}, 'REACH000483': {}, 'REACH000501': {'hm': 30}, 'REACH000514': {'hm': 23}, 'REACH000517': {'hm': 24}, 'REACH000529': {'hm': 26}, 'REACH000532_ONT': {'hm': 32}, 'REACH000540': {'hm': 26}, 'REACH000544': {'hm': 28}, 'REACH000558': {'hm': 28}, 'REACH000559': {'hm': 33}, 'REACH000560': {'hm': 63}, 'REACH000564': {'hm': 37}, 'REACH000626': {'hm': 21}, 'REACH000663': {'hm': 42}, 'REACH000681': {'hm': -6}, 'REACH000684': {'hm': 24}, 'REACH000685': {'hm': 17}, 'REACH000503': {'hm': 22}, 'REACH000642': {'hm': 30}, 'REACH000646': {'hm': 30}, 'REACH000707': {'hm': 0}, 'REACH000562': {'hm': 14}, 'REACH000156': {'hm': 22}, 'REACH000542': {'hm': 23}, 'REACH000440': {'hm': 32}, 'REACH000481': {'hm': 37}, 'REACH000557': {'hm': 26}, 'REACH000099': {'hm': 32}, 'REACH000749': {'hm': 27}, 'REACH000662': {'hm': 22}, 'REACH000295': {'hm': 35}, 'REACH000094': {'hm': 34}, 'REACH000442': {'hm': 27}, 'REACH000566': {'hm': 27}, 'REACH000402': {'hm': 26}, 'REACH000161': {'hm': 0}, 'REACH000437': {'hm': 16}, 'REACH000753': {'hm': 30}, 'REACH000539': {'hm': 32}, 'REACH000452': {}, 'REACH000524': {'hm': 39}, 'REACH000088': {'hm': 32}, 'REACH000315': {'hm': 30}, 'REACH000516': {'hm': 28}, 'REACH000196': {'hm': 29}, 'REACH000683': {'hm': 28}, 'REACH000580': {'hm': 23}, 'REACH000122': {'hm': 30}, 'REACH000411': {'hm': 26}, 'REACH000629': {'hm': 30}, 'REACH000290': {'hm': 27}, 'REACH000478': {'hm': 6}, 'REACH000425': {'hm': 29}, 'REACH000235': {'hm': 27}, 'REACH000184': {'hm': 26}, 'REACH000421': {'hm': 30}, 'REACH000216': {'hm': -1}, 'REACH000546': {'hm': 23}, 'REACH000286': {'hm': 7}, 'REACH000673': {'hm': 30}, 'REACH000531_ONT': {}, 'REACH000604': {'hm': -6}, 'REACH000527': {'hm': 30}, 'REACH000512': {'hm': 20}, 'REACH000221': {'hm': 36}, 'REACH000416': {'hm': 29}, 'REACH000226_sperm': {'hm': 0}}\n"
     ]
    }
   ],
   "source": [
    "hp_info_dict_male = {s: hp_info_dict[s] for s in hp_info_dict if s in df_samples_flt_males.sample_id.tolist()}\n",
    "print(hp_info_dict_male)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a232c306-5156-4b81-92c1-4057de1bef5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_hp_info(row, info_dict):\n",
    "    ### we set the hp info for hp_tag or h1 and h2 only. because for h0 we don't know the haplotype and the bp_dev can be wrong by taking the median of them\n",
    "    this_hp = row['hp_tag']\n",
    "    if this_hp == 'h0':\n",
    "        return 'NA', 'NA'\n",
    "    if this_hp in info_dict[row['subject']]:\n",
    "        this_hp_bp_dev = info_dict[row['subject']][this_hp]\n",
    "        this_hp_rep = round(this_hp_bp_dev / 3) + 20\n",
    "        return this_hp_bp_dev, this_hp_rep\n",
    "    else:\n",
    "        ### this could happen when you are looking at other genes with a haplotype that was not present in the FMR1 gene\n",
    "        ### in these cases the information for that hp in the other genes is not known, so we leave NA\n",
    "        return 'NA', 'NA'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "64c10dfe-579e-4d3b-8aaf-bfdd03916bef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "      <th>hp_bp_dev</th>\n",
       "      <th>hp_rep</th>\n",
       "      <th>gene_tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs</td>\n",
       "      <td>-21</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3.22</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210853 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            region_id  methylation                             read_name  \\\n",
       "0       TMEM185A_5UTR            5   m54340U_200716_001732/174262380/ccs   \n",
       "1       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "2       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "3       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "4       TMEM185A_5UTR            0   m54340U_200716_001732/174262380/ccs   \n",
       "...               ...          ...                                   ...   \n",
       "210848     HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210849     HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d   \n",
       "210850     HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210851     HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "210852     HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628   \n",
       "\n",
       "        bp_dev hp_tag            subject rel  sex  coverage  \\\n",
       "0          -21     hm        REACH000026   C    1      3.22   \n",
       "1          -21     hm        REACH000026   C    1      3.22   \n",
       "2          -21     hm        REACH000026   C    1      3.22   \n",
       "3          -21     hm        REACH000026   C    1      3.22   \n",
       "4          -21     hm        REACH000026   C    1      3.22   \n",
       "...        ...    ...                ...  ..  ...       ...   \n",
       "210848       1     hm  REACH000226_sperm   D    1     27.87   \n",
       "210849       1     hm  REACH000226_sperm   D    1     27.87   \n",
       "210850      -4     hm  REACH000226_sperm   D    1     27.87   \n",
       "210851      -4     hm  REACH000226_sperm   D    1     27.87   \n",
       "210852      -4     hm  REACH000226_sperm   D    1     27.87   \n",
       "\n",
       "            bp_rep_status hp_bp_dev hp_rep           gene_tag  \n",
       "0       XCI (other genes)        NA     NA  XCI (other genes)  \n",
       "1       XCI (other genes)        NA     NA  XCI (other genes)  \n",
       "2       XCI (other genes)        NA     NA  XCI (other genes)  \n",
       "3       XCI (other genes)        NA     NA  XCI (other genes)  \n",
       "4       XCI (other genes)        NA     NA  XCI (other genes)  \n",
       "...                   ...       ...    ...                ...  \n",
       "210848  XCI (other genes)         0     20  XCI (other genes)  \n",
       "210849  XCI (other genes)         0     20  XCI (other genes)  \n",
       "210850  XCI (other genes)         0     20  XCI (other genes)  \n",
       "210851  XCI (other genes)         0     20  XCI (other genes)  \n",
       "210852  XCI (other genes)         0     20  XCI (other genes)  \n",
       "\n",
       "[210853 rows x 13 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_bps[['hp_bp_dev', 'hp_rep']] = df_bps.apply(lambda row: get_hp_info(row, hp_info_dict), \n",
    "                                                                                 axis=1, result_type='expand')\n",
    "df_bps['gene_tag'] = df_bps.apply(lambda row: 'FMR1 UTR' if row['region_id']=='FMR1_5UTR' else 'XCI (other genes)', axis=1)\n",
    "display(df_bps)\n",
    "#display(df_bps.loc[df_bps.hp_bp_dev==-10000])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4bfe1208-a60a-415f-9231-80c809e7eaba",
   "metadata": {},
   "outputs": [],
   "source": [
    "###\n",
    "### for females we require that the haplotype is h1 or h2, and we filter out h0 becuase we don't know which haplotype it is\n",
    "### and taking the median will not be accurate\n",
    "df_bps_fm = df_bps.loc[(df_bps.subject.isin(df_samples_flt_females.sample_id))] # &(df_bps.hp_tag!='h0')\n",
    "df_bps_fm_kid = df_bps.loc[(df_bps.subject.isin(df_samples_flt_female_kids.sample_id))]\n",
    "df_bps_male = df_bps.loc[df_bps.subject.isin(df_samples_flt_males.sample_id)]\n",
    "df_bps_flt = df_bps.loc[df_bps.subject.isin(df_samples_flt.sample_id)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a8729ea8-9019-493b-8d92-ad3083f8bec4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_flt:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "      <th>hp_bp_dev</th>\n",
       "      <th>hp_rep</th>\n",
       "      <th>gene_tag</th>\n",
       "      <th>hp_rep_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130088 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         region_id  methylation                             read_name  bp_dev  \\\n",
       "194      FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "195      FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "196      FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "197      FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "198      FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "...            ...          ...                                   ...     ...   \n",
       "210848  HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d       1   \n",
       "210849  HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d       1   \n",
       "210850  HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628      -4   \n",
       "210851  HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628      -4   \n",
       "210852  HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628      -4   \n",
       "\n",
       "       hp_tag            subject rel  sex  coverage      bp_rep_status  \\\n",
       "194        hm        REACH000058   C    1      6.19      30<repeats<40   \n",
       "195        hm        REACH000058   C    1      6.19      30<repeats<40   \n",
       "196        hm        REACH000058   C    1      6.19      30<repeats<40   \n",
       "197        hm        REACH000058   C    1      6.19      30<repeats<40   \n",
       "198        hm        REACH000058   C    1      6.19      30<repeats<40   \n",
       "...       ...                ...  ..  ...       ...                ...   \n",
       "210848     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210849     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210850     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210851     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210852     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "\n",
       "       hp_bp_dev hp_rep           gene_tag  hp_rep_status  \n",
       "194           51     37           FMR1 UTR  30<repeats<40  \n",
       "195           51     37           FMR1 UTR  30<repeats<40  \n",
       "196           51     37           FMR1 UTR  30<repeats<40  \n",
       "197           51     37           FMR1 UTR  30<repeats<40  \n",
       "198           51     37           FMR1 UTR  30<repeats<40  \n",
       "...          ...    ...                ...            ...  \n",
       "210848         0     20  XCI (other genes)     repeats<20  \n",
       "210849         0     20  XCI (other genes)     repeats<20  \n",
       "210850         0     20  XCI (other genes)     repeats<20  \n",
       "210851         0     20  XCI (other genes)     repeats<20  \n",
       "210852         0     20  XCI (other genes)     repeats<20  \n",
       "\n",
       "[130088 rows x 14 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "### make tables for all subjects with at least 2x coverage\n",
    "df_flt = df_bps_flt.loc[df_bps_flt.hp_bp_dev!='NA'].copy()\n",
    "df_flt['hp_rep_status'] = df_flt.apply(lambda row: get_status(row, 'hp_bp_dev', ignore_region=True), axis=1)\n",
    "print('df_flt:')\n",
    "display(df_flt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "af0aad33-f01d-47e2-95c5-15a1e2243b40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_flt_fmr1:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "      <th>hp_bp_dev</th>\n",
       "      <th>hp_rep</th>\n",
       "      <th>gene_tag</th>\n",
       "      <th>hp_rep_status</th>\n",
       "      <th>bp_rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_210523_044800/52758275/ccs</td>\n",
       "      <td>51</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209228</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>1074c822-d8c7-4eb6-9210-72ca8bfe9602</td>\n",
       "      <td>-3</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209229</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>46</td>\n",
       "      <td>1074c822-d8c7-4eb6-9210-72ca8bfe9602</td>\n",
       "      <td>-3</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209230</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>7</td>\n",
       "      <td>1074c822-d8c7-4eb6-9210-72ca8bfe9602</td>\n",
       "      <td>-3</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209231</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>1074c822-d8c7-4eb6-9210-72ca8bfe9602</td>\n",
       "      <td>-3</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209232</th>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>1</td>\n",
       "      <td>1074c822-d8c7-4eb6-9210-72ca8bfe9602</td>\n",
       "      <td>-3</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>FMR1 UTR</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42511 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        region_id  methylation                             read_name  bp_dev  \\\n",
       "194     FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "195     FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "196     FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "197     FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "198     FMR1_5UTR            0    m54340U_210523_044800/52758275/ccs      51   \n",
       "...           ...          ...                                   ...     ...   \n",
       "209228  FMR1_5UTR            0  1074c822-d8c7-4eb6-9210-72ca8bfe9602      -3   \n",
       "209229  FMR1_5UTR           46  1074c822-d8c7-4eb6-9210-72ca8bfe9602      -3   \n",
       "209230  FMR1_5UTR            7  1074c822-d8c7-4eb6-9210-72ca8bfe9602      -3   \n",
       "209231  FMR1_5UTR            0  1074c822-d8c7-4eb6-9210-72ca8bfe9602      -3   \n",
       "209232  FMR1_5UTR            1  1074c822-d8c7-4eb6-9210-72ca8bfe9602      -3   \n",
       "\n",
       "       hp_tag            subject rel  sex  coverage  bp_rep_status hp_bp_dev  \\\n",
       "194        hm        REACH000058   C    1      6.19  30<repeats<40        51   \n",
       "195        hm        REACH000058   C    1      6.19  30<repeats<40        51   \n",
       "196        hm        REACH000058   C    1      6.19  30<repeats<40        51   \n",
       "197        hm        REACH000058   C    1      6.19  30<repeats<40        51   \n",
       "198        hm        REACH000058   C    1      6.19  30<repeats<40        51   \n",
       "...       ...                ...  ..  ...       ...            ...       ...   \n",
       "209228     hm  REACH000226_sperm   D    1     27.87     repeats<20         0   \n",
       "209229     hm  REACH000226_sperm   D    1     27.87     repeats<20         0   \n",
       "209230     hm  REACH000226_sperm   D    1     27.87     repeats<20         0   \n",
       "209231     hm  REACH000226_sperm   D    1     27.87     repeats<20         0   \n",
       "209232     hm  REACH000226_sperm   D    1     27.87     repeats<20         0   \n",
       "\n",
       "       hp_rep  gene_tag  hp_rep_status  bp_rep  \n",
       "194        37  FMR1 UTR  30<repeats<40      37  \n",
       "195        37  FMR1 UTR  30<repeats<40      37  \n",
       "196        37  FMR1 UTR  30<repeats<40      37  \n",
       "197        37  FMR1 UTR  30<repeats<40      37  \n",
       "198        37  FMR1 UTR  30<repeats<40      37  \n",
       "...       ...       ...            ...     ...  \n",
       "209228     20  FMR1 UTR     repeats<20      19  \n",
       "209229     20  FMR1 UTR     repeats<20      19  \n",
       "209230     20  FMR1 UTR     repeats<20      19  \n",
       "209231     20  FMR1 UTR     repeats<20      19  \n",
       "209232     20  FMR1 UTR     repeats<20      19  \n",
       "\n",
       "[42511 rows x 15 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_reads_fmr1:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bp_rep</th>\n",
       "      <th>methylation_mean</th>\n",
       "      <th>methylation_std</th>\n",
       "      <th>methylation_mean_binary</th>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>hp_rep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>read_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>00f9b098-78b4-4f41-b956-cbde27cee4a9</th>\n",
       "      <td>35</td>\n",
       "      <td>81.431034</td>\n",
       "      <td>33.684872</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000523</td>\n",
       "      <td>h2</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022a04bb-5a93-4714-8f50-808c07dad312</th>\n",
       "      <td>18</td>\n",
       "      <td>20.133333</td>\n",
       "      <td>31.325708</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000604</td>\n",
       "      <td>hm</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>02cf1093-a6d1-4ed9-b452-89ea2c2b661d</th>\n",
       "      <td>22</td>\n",
       "      <td>10.000000</td>\n",
       "      <td>21.982511</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000626</td>\n",
       "      <td>hm</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>03498a1d-13a3-435a-9bb4-78b3a1852470</th>\n",
       "      <td>29</td>\n",
       "      <td>4.824561</td>\n",
       "      <td>12.376710</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000503</td>\n",
       "      <td>hm</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>036caaab-f5dd-4b1b-b88c-264465c960d6</th>\n",
       "      <td>23</td>\n",
       "      <td>6.693878</td>\n",
       "      <td>15.192109</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000562</td>\n",
       "      <td>hm</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220618_125405/16974869/ccs</th>\n",
       "      <td>30</td>\n",
       "      <td>2.719298</td>\n",
       "      <td>5.608522</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000289</td>\n",
       "      <td>h1</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220618_125405/2752607/ccs</th>\n",
       "      <td>30</td>\n",
       "      <td>0.035088</td>\n",
       "      <td>0.185637</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000289</td>\n",
       "      <td>h2</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220618_125405/28051053/ccs</th>\n",
       "      <td>30</td>\n",
       "      <td>0.192982</td>\n",
       "      <td>0.953164</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000289</td>\n",
       "      <td>h1</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/129305076/ccs</th>\n",
       "      <td>29</td>\n",
       "      <td>11.910714</td>\n",
       "      <td>10.455580</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/82970194/ccs</th>\n",
       "      <td>29</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>0.942503</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>756 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      bp_rep  methylation_mean  \\\n",
       "read_name                                                        \n",
       "00f9b098-78b4-4f41-b956-cbde27cee4a9      35         81.431034   \n",
       "022a04bb-5a93-4714-8f50-808c07dad312      18         20.133333   \n",
       "02cf1093-a6d1-4ed9-b452-89ea2c2b661d      22         10.000000   \n",
       "03498a1d-13a3-435a-9bb4-78b3a1852470      29          4.824561   \n",
       "036caaab-f5dd-4b1b-b88c-264465c960d6      23          6.693878   \n",
       "...                                      ...               ...   \n",
       "m64278e_220618_125405/16974869/ccs        30          2.719298   \n",
       "m64278e_220618_125405/2752607/ccs         30          0.035088   \n",
       "m64278e_220618_125405/28051053/ccs        30          0.192982   \n",
       "m64278e_220704_012231/129305076/ccs       29         11.910714   \n",
       "m64278e_220704_012231/82970194/ccs        29          0.357143   \n",
       "\n",
       "                                      methylation_std  \\\n",
       "read_name                                               \n",
       "00f9b098-78b4-4f41-b956-cbde27cee4a9        33.684872   \n",
       "022a04bb-5a93-4714-8f50-808c07dad312        31.325708   \n",
       "02cf1093-a6d1-4ed9-b452-89ea2c2b661d        21.982511   \n",
       "03498a1d-13a3-435a-9bb4-78b3a1852470        12.376710   \n",
       "036caaab-f5dd-4b1b-b88c-264465c960d6        15.192109   \n",
       "...                                               ...   \n",
       "m64278e_220618_125405/16974869/ccs           5.608522   \n",
       "m64278e_220618_125405/2752607/ccs            0.185637   \n",
       "m64278e_220618_125405/28051053/ccs           0.953164   \n",
       "m64278e_220704_012231/129305076/ccs         10.455580   \n",
       "m64278e_220704_012231/82970194/ccs           0.942503   \n",
       "\n",
       "                                      methylation_mean_binary      subject  \\\n",
       "read_name                                                                    \n",
       "00f9b098-78b4-4f41-b956-cbde27cee4a9                        1  REACH000523   \n",
       "022a04bb-5a93-4714-8f50-808c07dad312                        0  REACH000604   \n",
       "02cf1093-a6d1-4ed9-b452-89ea2c2b661d                        0  REACH000626   \n",
       "03498a1d-13a3-435a-9bb4-78b3a1852470                        0  REACH000503   \n",
       "036caaab-f5dd-4b1b-b88c-264465c960d6                        0  REACH000562   \n",
       "...                                                       ...          ...   \n",
       "m64278e_220618_125405/16974869/ccs                          0  REACH000289   \n",
       "m64278e_220618_125405/2752607/ccs                           0  REACH000289   \n",
       "m64278e_220618_125405/28051053/ccs                          0  REACH000289   \n",
       "m64278e_220704_012231/129305076/ccs                         0  REACH000290   \n",
       "m64278e_220704_012231/82970194/ccs                          0  REACH000290   \n",
       "\n",
       "                                     hp_tag  hp_rep  \n",
       "read_name                                            \n",
       "00f9b098-78b4-4f41-b956-cbde27cee4a9     h2      29  \n",
       "022a04bb-5a93-4714-8f50-808c07dad312     hm      18  \n",
       "02cf1093-a6d1-4ed9-b452-89ea2c2b661d     hm      27  \n",
       "03498a1d-13a3-435a-9bb4-78b3a1852470     hm      27  \n",
       "036caaab-f5dd-4b1b-b88c-264465c960d6     hm      25  \n",
       "...                                     ...     ...  \n",
       "m64278e_220618_125405/16974869/ccs       h1      30  \n",
       "m64278e_220618_125405/2752607/ccs        h2      30  \n",
       "m64278e_220618_125405/28051053/ccs       h1      30  \n",
       "m64278e_220704_012231/129305076/ccs      hm      29  \n",
       "m64278e_220704_012231/82970194/ccs       hm      29  \n",
       "\n",
       "[756 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_hps_fmr1:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>hp_rep</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>methylation_mean</th>\n",
       "      <th>methylation_std</th>\n",
       "      <th>methylation_mean_binary</th>\n",
       "      <th>methylation_std_binary</th>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <th>hm</th>\n",
       "      <td>37</td>\n",
       "      <td>7</td>\n",
       "      <td>12.742508</td>\n",
       "      <td>6.459954</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000086</th>\n",
       "      <th>hm</th>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>8.070175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">REACH000087</th>\n",
       "      <th>h1</th>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>18.859649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>REACH000087</td>\n",
       "      <td>h1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>h2</th>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "      <td>59.431413</td>\n",
       "      <td>31.915171</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>REACH000087</td>\n",
       "      <td>h2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000088</th>\n",
       "      <th>hm</th>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>7.419643</td>\n",
       "      <td>4.760344</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>REACH000088</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000755</th>\n",
       "      <th>h1</th>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>30.477754</td>\n",
       "      <td>49.626784</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>REACH000755</td>\n",
       "      <td>h1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000762</th>\n",
       "      <th>hm</th>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>23.824561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>REACH000762</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000763</th>\n",
       "      <th>h2</th>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>15.763158</td>\n",
       "      <td>1.501051</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>h2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000766</th>\n",
       "      <th>h2</th>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>REACH000766</td>\n",
       "      <td>h2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000767</th>\n",
       "      <th>hm</th>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "      <td>1.307018</td>\n",
       "      <td>1.175788</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>REACH000767</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>219 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    hp_rep  num_reads  methylation_mean  methylation_std  \\\n",
       "subject     hp_tag                                                         \n",
       "REACH000058 hm          37          7         12.742508         6.459954   \n",
       "REACH000086 hm          30          1          8.070175              NaN   \n",
       "REACH000087 h1          30          1         18.859649              NaN   \n",
       "            h2          30          4         59.431413        31.915171   \n",
       "REACH000088 hm          31          2          7.419643         4.760344   \n",
       "...                    ...        ...               ...              ...   \n",
       "REACH000755 h1          20          3         30.477754        49.626784   \n",
       "REACH000762 hm          32          1         23.824561              NaN   \n",
       "REACH000763 h2          31          2         15.763158         1.501051   \n",
       "REACH000766 h2          31          1          0.000000              NaN   \n",
       "REACH000767 hm          30          4          1.307018         1.175788   \n",
       "\n",
       "                    methylation_mean_binary  methylation_std_binary  \\\n",
       "subject     hp_tag                                                    \n",
       "REACH000058 hm                     0.000000                 0.00000   \n",
       "REACH000086 hm                     0.000000                     NaN   \n",
       "REACH000087 h1                     0.000000                     NaN   \n",
       "            h2                     0.750000                 0.50000   \n",
       "REACH000088 hm                     0.000000                 0.00000   \n",
       "...                                     ...                     ...   \n",
       "REACH000755 h1                     0.333333                 0.57735   \n",
       "REACH000762 hm                     0.000000                     NaN   \n",
       "REACH000763 h2                     0.000000                 0.00000   \n",
       "REACH000766 h2                     0.000000                     NaN   \n",
       "REACH000767 hm                     0.000000                 0.00000   \n",
       "\n",
       "                        subject hp_tag  \n",
       "subject     hp_tag                      \n",
       "REACH000058 hm      REACH000058     hm  \n",
       "REACH000086 hm      REACH000086     hm  \n",
       "REACH000087 h1      REACH000087     h1  \n",
       "            h2      REACH000087     h2  \n",
       "REACH000088 hm      REACH000088     hm  \n",
       "...                         ...    ...  \n",
       "REACH000755 h1      REACH000755     h1  \n",
       "REACH000762 hm      REACH000762     hm  \n",
       "REACH000763 h2      REACH000763     h2  \n",
       "REACH000766 h2      REACH000766     h2  \n",
       "REACH000767 hm      REACH000767     hm  \n",
       "\n",
       "[219 rows x 8 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_flt_xci:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region_id</th>\n",
       "      <th>methylation</th>\n",
       "      <th>read_name</th>\n",
       "      <th>bp_dev</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>subject</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>bp_rep_status</th>\n",
       "      <th>hp_bp_dev</th>\n",
       "      <th>hp_rep</th>\n",
       "      <th>gene_tag</th>\n",
       "      <th>hp_rep_status</th>\n",
       "      <th>bp_rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200806_070905/88277013/ccs</td>\n",
       "      <td>0</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>m54340U_200806_070905/88277013/ccs</td>\n",
       "      <td>0</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>639</th>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200806_070905/88277013/ccs</td>\n",
       "      <td>0</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>1</td>\n",
       "      <td>m54340U_200806_070905/88277013/ccs</td>\n",
       "      <td>0</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>m54340U_200806_070905/88277013/ccs</td>\n",
       "      <td>0</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>51</td>\n",
       "      <td>37</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>30&lt;repeats&lt;40</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210848</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>12</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210849</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>61e50e11-513c-490f-90fb-7f2473f7576d</td>\n",
       "      <td>1</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210850</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210851</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210852</th>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>3</td>\n",
       "      <td>7ce83f3e-5461-49fe-9c4a-c0304d719628</td>\n",
       "      <td>-4</td>\n",
       "      <td>hm</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>27.87</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>XCI (other genes)</td>\n",
       "      <td>repeats&lt;20</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87577 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         region_id  methylation                             read_name  bp_dev  \\\n",
       "637       IDS_5UTR            0    m54340U_200806_070905/88277013/ccs       0   \n",
       "638       IDS_5UTR            2    m54340U_200806_070905/88277013/ccs       0   \n",
       "639       IDS_5UTR            0    m54340U_200806_070905/88277013/ccs       0   \n",
       "640       IDS_5UTR            1    m54340U_200806_070905/88277013/ccs       0   \n",
       "641       IDS_5UTR            0    m54340U_200806_070905/88277013/ccs       0   \n",
       "...            ...          ...                                   ...     ...   \n",
       "210848  HMGB3_5UTR           12  61e50e11-513c-490f-90fb-7f2473f7576d       1   \n",
       "210849  HMGB3_5UTR            0  61e50e11-513c-490f-90fb-7f2473f7576d       1   \n",
       "210850  HMGB3_5UTR            2  7ce83f3e-5461-49fe-9c4a-c0304d719628      -4   \n",
       "210851  HMGB3_5UTR            0  7ce83f3e-5461-49fe-9c4a-c0304d719628      -4   \n",
       "210852  HMGB3_5UTR            3  7ce83f3e-5461-49fe-9c4a-c0304d719628      -4   \n",
       "\n",
       "       hp_tag            subject rel  sex  coverage      bp_rep_status  \\\n",
       "637        hm        REACH000058   C    1      6.19  XCI (other genes)   \n",
       "638        hm        REACH000058   C    1      6.19  XCI (other genes)   \n",
       "639        hm        REACH000058   C    1      6.19  XCI (other genes)   \n",
       "640        hm        REACH000058   C    1      6.19  XCI (other genes)   \n",
       "641        hm        REACH000058   C    1      6.19  XCI (other genes)   \n",
       "...       ...                ...  ..  ...       ...                ...   \n",
       "210848     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210849     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210850     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210851     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "210852     hm  REACH000226_sperm   D    1     27.87  XCI (other genes)   \n",
       "\n",
       "       hp_bp_dev hp_rep           gene_tag  hp_rep_status  bp_rep  \n",
       "637           51     37  XCI (other genes)  30<repeats<40      20  \n",
       "638           51     37  XCI (other genes)  30<repeats<40      20  \n",
       "639           51     37  XCI (other genes)  30<repeats<40      20  \n",
       "640           51     37  XCI (other genes)  30<repeats<40      20  \n",
       "641           51     37  XCI (other genes)  30<repeats<40      20  \n",
       "...          ...    ...                ...            ...     ...  \n",
       "210848         0     20  XCI (other genes)     repeats<20      20  \n",
       "210849         0     20  XCI (other genes)     repeats<20      20  \n",
       "210850         0     20  XCI (other genes)     repeats<20      19  \n",
       "210851         0     20  XCI (other genes)     repeats<20      19  \n",
       "210852         0     20  XCI (other genes)     repeats<20      19  \n",
       "\n",
       "[87577 rows x 15 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_reads_xci:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bp_rep</th>\n",
       "      <th>methylation_mean</th>\n",
       "      <th>methylation_std</th>\n",
       "      <th>methylation_mean_binary</th>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>hp_rep</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>read_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0027fca9-122a-43ab-b29c-4625eeafb4f3</th>\n",
       "      <td>14</td>\n",
       "      <td>84.310345</td>\n",
       "      <td>32.317734</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000482</td>\n",
       "      <td>h2</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0034a01f-16d8-4bfd-a654-cade57eb50e6</th>\n",
       "      <td>20</td>\n",
       "      <td>2.461538</td>\n",
       "      <td>6.610481</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000427</td>\n",
       "      <td>hm</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0040855e-b99b-4eff-9f45-852cb1b377c7</th>\n",
       "      <td>20</td>\n",
       "      <td>7.875000</td>\n",
       "      <td>15.615354</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000684</td>\n",
       "      <td>hm</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0044cef1-af81-458e-aa05-04768f4d5da3</th>\n",
       "      <td>17</td>\n",
       "      <td>3.275862</td>\n",
       "      <td>8.526834</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000476</td>\n",
       "      <td>hm</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>008b5a8e-89f5-4ee0-984d-a8a1d2296fe7</th>\n",
       "      <td>22</td>\n",
       "      <td>4.526316</td>\n",
       "      <td>11.874881</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>hm</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/6686007/ccs</th>\n",
       "      <td>20</td>\n",
       "      <td>4.200000</td>\n",
       "      <td>4.868265</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/81855701/ccs</th>\n",
       "      <td>21</td>\n",
       "      <td>1.666667</td>\n",
       "      <td>1.751190</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/83362719/ccs</th>\n",
       "      <td>20</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>1.388730</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/9176454/ccs</th>\n",
       "      <td>20</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>1.164965</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>m64278e_220704_012231/98568707/ccs</th>\n",
       "      <td>20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000290</td>\n",
       "      <td>hm</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4984 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      bp_rep  methylation_mean  \\\n",
       "read_name                                                        \n",
       "0027fca9-122a-43ab-b29c-4625eeafb4f3      14         84.310345   \n",
       "0034a01f-16d8-4bfd-a654-cade57eb50e6      20          2.461538   \n",
       "0040855e-b99b-4eff-9f45-852cb1b377c7      20          7.875000   \n",
       "0044cef1-af81-458e-aa05-04768f4d5da3      17          3.275862   \n",
       "008b5a8e-89f5-4ee0-984d-a8a1d2296fe7      22          4.526316   \n",
       "...                                      ...               ...   \n",
       "m64278e_220704_012231/6686007/ccs         20          4.200000   \n",
       "m64278e_220704_012231/81855701/ccs        21          1.666667   \n",
       "m64278e_220704_012231/83362719/ccs        20          0.750000   \n",
       "m64278e_220704_012231/9176454/ccs         20          0.750000   \n",
       "m64278e_220704_012231/98568707/ccs        20          0.000000   \n",
       "\n",
       "                                      methylation_std  \\\n",
       "read_name                                               \n",
       "0027fca9-122a-43ab-b29c-4625eeafb4f3        32.317734   \n",
       "0034a01f-16d8-4bfd-a654-cade57eb50e6         6.610481   \n",
       "0040855e-b99b-4eff-9f45-852cb1b377c7        15.615354   \n",
       "0044cef1-af81-458e-aa05-04768f4d5da3         8.526834   \n",
       "008b5a8e-89f5-4ee0-984d-a8a1d2296fe7        11.874881   \n",
       "...                                               ...   \n",
       "m64278e_220704_012231/6686007/ccs            4.868265   \n",
       "m64278e_220704_012231/81855701/ccs           1.751190   \n",
       "m64278e_220704_012231/83362719/ccs           1.388730   \n",
       "m64278e_220704_012231/9176454/ccs            1.164965   \n",
       "m64278e_220704_012231/98568707/ccs           0.000000   \n",
       "\n",
       "                                      methylation_mean_binary  \\\n",
       "read_name                                                       \n",
       "0027fca9-122a-43ab-b29c-4625eeafb4f3                        1   \n",
       "0034a01f-16d8-4bfd-a654-cade57eb50e6                        0   \n",
       "0040855e-b99b-4eff-9f45-852cb1b377c7                        0   \n",
       "0044cef1-af81-458e-aa05-04768f4d5da3                        0   \n",
       "008b5a8e-89f5-4ee0-984d-a8a1d2296fe7                        0   \n",
       "...                                                       ...   \n",
       "m64278e_220704_012231/6686007/ccs                           0   \n",
       "m64278e_220704_012231/81855701/ccs                          0   \n",
       "m64278e_220704_012231/83362719/ccs                          0   \n",
       "m64278e_220704_012231/9176454/ccs                           0   \n",
       "m64278e_220704_012231/98568707/ccs                          0   \n",
       "\n",
       "                                                subject hp_tag  hp_rep  \n",
       "read_name                                                               \n",
       "0027fca9-122a-43ab-b29c-4625eeafb4f3        REACH000482     h2      20  \n",
       "0034a01f-16d8-4bfd-a654-cade57eb50e6        REACH000427     hm      28  \n",
       "0040855e-b99b-4eff-9f45-852cb1b377c7        REACH000684     hm      28  \n",
       "0044cef1-af81-458e-aa05-04768f4d5da3        REACH000476     hm      30  \n",
       "008b5a8e-89f5-4ee0-984d-a8a1d2296fe7  REACH000226_sperm     hm      20  \n",
       "...                                                 ...    ...     ...  \n",
       "m64278e_220704_012231/6686007/ccs           REACH000290     hm      29  \n",
       "m64278e_220704_012231/81855701/ccs          REACH000290     hm      29  \n",
       "m64278e_220704_012231/83362719/ccs          REACH000290     hm      29  \n",
       "m64278e_220704_012231/9176454/ccs           REACH000290     hm      29  \n",
       "m64278e_220704_012231/98568707/ccs          REACH000290     hm      29  \n",
       "\n",
       "[4984 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_hps_xci:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>hp_rep</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>methylation_mean</th>\n",
       "      <th>methylation_std</th>\n",
       "      <th>methylation_mean_binary</th>\n",
       "      <th>methylation_std_binary</th>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <th>hm</th>\n",
       "      <td>37</td>\n",
       "      <td>28</td>\n",
       "      <td>7.594767</td>\n",
       "      <td>8.684887</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000086</th>\n",
       "      <th>hm</th>\n",
       "      <td>30</td>\n",
       "      <td>8</td>\n",
       "      <td>6.067812</td>\n",
       "      <td>9.957010</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">REACH000087</th>\n",
       "      <th>h1</th>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>35.944841</td>\n",
       "      <td>39.181304</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>REACH000087</td>\n",
       "      <td>h1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>h2</th>\n",
       "      <td>30</td>\n",
       "      <td>7</td>\n",
       "      <td>59.863159</td>\n",
       "      <td>34.534200</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.487950</td>\n",
       "      <td>REACH000087</td>\n",
       "      <td>h2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000088</th>\n",
       "      <th>hm</th>\n",
       "      <td>31</td>\n",
       "      <td>18</td>\n",
       "      <td>8.140036</td>\n",
       "      <td>8.183474</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>REACH000088</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000755</th>\n",
       "      <th>h1</th>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>51.713706</td>\n",
       "      <td>40.257137</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.507093</td>\n",
       "      <td>REACH000755</td>\n",
       "      <td>h1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000762</th>\n",
       "      <th>hm</th>\n",
       "      <td>32</td>\n",
       "      <td>16</td>\n",
       "      <td>4.585281</td>\n",
       "      <td>6.695018</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>REACH000762</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000763</th>\n",
       "      <th>h2</th>\n",
       "      <td>31</td>\n",
       "      <td>14</td>\n",
       "      <td>69.680573</td>\n",
       "      <td>37.437119</td>\n",
       "      <td>0.785714</td>\n",
       "      <td>0.425815</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>h2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000766</th>\n",
       "      <th>h2</th>\n",
       "      <td>31</td>\n",
       "      <td>18</td>\n",
       "      <td>51.360592</td>\n",
       "      <td>40.967890</td>\n",
       "      <td>0.611111</td>\n",
       "      <td>0.501631</td>\n",
       "      <td>REACH000766</td>\n",
       "      <td>h2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000767</th>\n",
       "      <th>hm</th>\n",
       "      <td>30</td>\n",
       "      <td>26</td>\n",
       "      <td>4.172865</td>\n",
       "      <td>6.990531</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>REACH000767</td>\n",
       "      <td>hm</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>218 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    hp_rep  num_reads  methylation_mean  methylation_std  \\\n",
       "subject     hp_tag                                                         \n",
       "REACH000058 hm          37         28          7.594767         8.684887   \n",
       "REACH000086 hm          30          8          6.067812         9.957010   \n",
       "REACH000087 h1          30          9         35.944841        39.181304   \n",
       "            h2          30          7         59.863159        34.534200   \n",
       "REACH000088 hm          31         18          8.140036         8.183474   \n",
       "...                    ...        ...               ...              ...   \n",
       "REACH000755 h1          20         15         51.713706        40.257137   \n",
       "REACH000762 hm          32         16          4.585281         6.695018   \n",
       "REACH000763 h2          31         14         69.680573        37.437119   \n",
       "REACH000766 h2          31         18         51.360592        40.967890   \n",
       "REACH000767 hm          30         26          4.172865         6.990531   \n",
       "\n",
       "                    methylation_mean_binary  methylation_std_binary  \\\n",
       "subject     hp_tag                                                    \n",
       "REACH000058 hm                     0.000000                0.000000   \n",
       "REACH000086 hm                     0.000000                0.000000   \n",
       "REACH000087 h1                     0.333333                0.500000   \n",
       "            h2                     0.714286                0.487950   \n",
       "REACH000088 hm                     0.000000                0.000000   \n",
       "...                                     ...                     ...   \n",
       "REACH000755 h1                     0.600000                0.507093   \n",
       "REACH000762 hm                     0.000000                0.000000   \n",
       "REACH000763 h2                     0.785714                0.425815   \n",
       "REACH000766 h2                     0.611111                0.501631   \n",
       "REACH000767 hm                     0.000000                0.000000   \n",
       "\n",
       "                        subject hp_tag  \n",
       "subject     hp_tag                      \n",
       "REACH000058 hm      REACH000058     hm  \n",
       "REACH000086 hm      REACH000086     hm  \n",
       "REACH000087 h1      REACH000087     h1  \n",
       "            h2      REACH000087     h2  \n",
       "REACH000088 hm      REACH000088     hm  \n",
       "...                         ...    ...  \n",
       "REACH000755 h1      REACH000755     h1  \n",
       "REACH000762 hm      REACH000762     hm  \n",
       "REACH000763 h2      REACH000763     h2  \n",
       "REACH000766 h2      REACH000766     h2  \n",
       "REACH000767 hm      REACH000767     hm  \n",
       "\n",
       "[218 rows x 8 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### FMR1 haplotypes\n",
    "df_flt_fmr1 = df_flt.loc[df_flt.region_id=='FMR1_5UTR'].copy()\n",
    "df_flt_fmr1['bp_rep'] = (df_flt_fmr1['bp_dev'] / 3).round().astype(int) + 20\n",
    "print('df_flt_fmr1:')\n",
    "display(df_flt_fmr1)\n",
    "df_reads_fmr1 = pd.DataFrame()\n",
    "df_reads_fmr1['bp_rep'] = df_flt_fmr1.groupby(by='read_name')['bp_rep'].median().astype(int) # we first get the repeats for each read (the same for each bp)\n",
    "df_reads_fmr1['methylation_mean'] = df_reads_fmr1.index.map(df_flt_fmr1.groupby(by='read_name')['methylation'].mean()) # average methylation over bps for each read\n",
    "df_reads_fmr1['methylation_std'] = df_reads_fmr1.index.map(df_flt_fmr1.groupby(by='read_name')['methylation'].std())\n",
    "df_reads_fmr1['methylation_mean_binary'] = df_reads_fmr1['methylation_mean'].apply(lambda x: 1 if x>=50 else 0) # decide if a read is methylated or not (make binary)\n",
    "df_reads_fmr1['subject'] = df_reads_fmr1.index.map(df_flt_fmr1.groupby(by='read_name')['subject'].first())\n",
    "df_reads_fmr1['hp_tag'] = df_reads_fmr1.index.map(df_flt_fmr1.groupby(by='read_name')['hp_tag'].first())\n",
    "df_reads_fmr1['hp_rep'] = df_reads_fmr1.index.map(df_flt_fmr1.groupby(by='read_name')['hp_rep'].first())\n",
    "print('df_reads_fmr1:')\n",
    "display(df_reads_fmr1)\n",
    "#pd.set_option('display.max_rows', 10)\n",
    "#display(df_reads_fmr1.subject.value_counts())\n",
    "\n",
    "df_hps_fmr1 = pd.DataFrame()\n",
    "df_hps_fmr1['hp_rep'] = df_reads_fmr1.groupby(by=['subject', 'hp_tag'])['hp_rep'].first()\n",
    "df_hps_fmr1['num_reads'] = df_reads_fmr1.groupby(by=['subject', 'hp_tag'])['methylation_mean'].size()\n",
    "df_hps_fmr1['methylation_mean'] = df_reads_fmr1.groupby(by=['subject', 'hp_tag'])['methylation_mean'].mean()\n",
    "df_hps_fmr1['methylation_std'] = df_reads_fmr1.groupby(by=['subject', 'hp_tag'])['methylation_mean'].std()\n",
    "df_hps_fmr1['methylation_mean_binary'] = df_reads_fmr1.groupby(by=['subject', 'hp_tag'])['methylation_mean_binary'].mean()\n",
    "df_hps_fmr1['methylation_std_binary'] = df_reads_fmr1.groupby(by=['subject', 'hp_tag'])['methylation_mean_binary'].std()\n",
    "df_hps_fmr1['subject'] = df_hps_fmr1.index.map(lambda x: x[0])\n",
    "df_hps_fmr1['hp_tag'] = df_hps_fmr1.index.map(lambda x: x[1])\n",
    "pd.set_option('display.max_rows', 10)\n",
    "print('df_hps_fmr1:')\n",
    "display(df_hps_fmr1)\n",
    "\n",
    "### XCI other genes haplotypes\n",
    "df_flt_xci = df_flt.loc[df_flt.region_id!='FMR1_5UTR'].copy()\n",
    "df_flt_xci['bp_rep'] = (df_flt_xci['bp_dev'] / 3).round().astype(int) + 20\n",
    "print('df_flt_xci:')\n",
    "display(df_flt_xci)\n",
    "\n",
    "df_reads_xci = pd.DataFrame()\n",
    "df_reads_xci['bp_rep'] = df_flt_xci.groupby(by='read_name')['bp_rep'].median().astype(int) # we first get the repeats for each read (the same for each read)\n",
    "df_reads_xci['methylation_mean'] = df_reads_xci.index.map(df_flt_xci.groupby(by='read_name')['methylation'].mean()) # average methylation over bps for each read\n",
    "df_reads_xci['methylation_std'] = df_reads_xci.index.map(df_flt_xci.groupby(by='read_name')['methylation'].std())\n",
    "df_reads_xci['methylation_mean_binary'] = df_reads_xci['methylation_mean'].apply(lambda x: 1 if x>=50 else 0) # decide if a read is methylated or not (make binary)\n",
    "df_reads_xci['subject'] = df_reads_xci.index.map(df_flt_xci.groupby(by='read_name')['subject'].first())\n",
    "df_reads_xci['hp_tag'] = df_reads_xci.index.map(df_flt_xci.groupby(by='read_name')['hp_tag'].first())\n",
    "df_reads_xci['hp_rep'] = df_reads_xci.index.map(df_flt_xci.groupby(by='read_name')['hp_rep'].first())\n",
    "print('df_reads_xci:')\n",
    "display(df_reads_xci)\n",
    "#display(df_reads_xci.subject.value_counts())\n",
    "\n",
    "df_hps_xci = pd.DataFrame()\n",
    "df_hps_xci['hp_rep'] = df_reads_xci.groupby(by=['subject', 'hp_tag'])['hp_rep'].first()\n",
    "df_hps_xci['num_reads'] = df_reads_xci.groupby(by=['subject', 'hp_tag'])['methylation_mean'].size()\n",
    "df_hps_xci['methylation_mean'] = df_reads_xci.groupby(by=['subject', 'hp_tag'])['methylation_mean'].mean()\n",
    "df_hps_xci['methylation_std'] = df_reads_xci.groupby(by=['subject', 'hp_tag'])['methylation_mean'].std()\n",
    "df_hps_xci['methylation_mean_binary'] = df_reads_xci.groupby(by=['subject', 'hp_tag'])['methylation_mean_binary'].mean()\n",
    "df_hps_xci['methylation_std_binary'] = df_reads_xci.groupby(by=['subject', 'hp_tag'])['methylation_mean_binary'].std()\n",
    "df_hps_xci['subject'] = df_hps_xci.index.map(lambda x: x[0])\n",
    "df_hps_xci['hp_tag'] = df_hps_xci.index.map(lambda x: x[1])\n",
    "pd.set_option('display.max_rows', 10)\n",
    "print('df_hps_xci:')\n",
    "display(df_hps_xci)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "4fd116f3-d15a-4427-8437-f60eb82e3cbf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_merge:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th>fid</th>\n",
       "      <th>Affected</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>sex</th>\n",
       "      <th>coverage</th>\n",
       "      <th>rel</th>\n",
       "      <th>hp_rep_fmr1</th>\n",
       "      <th>num_reads_fmr1</th>\n",
       "      <th>...</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>PC9</th>\n",
       "      <th>PC10</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subject</th>\n",
       "      <th>hp_tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <th>hm</th>\n",
       "      <td>REACH000058</td>\n",
       "      <td>hm</td>\n",
       "      <td>F0058</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.19</td>\n",
       "      <td>C</td>\n",
       "      <td>37</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002422</td>\n",
       "      <td>-0.009216</td>\n",
       "      <td>-0.007762</td>\n",
       "      <td>0.073817</td>\n",
       "      <td>0.015720</td>\n",
       "      <td>0.006303</td>\n",
       "      <td>0.055403</td>\n",
       "      <td>-0.038327</td>\n",
       "      <td>-0.011381</td>\n",
       "      <td>0.032498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000086</th>\n",
       "      <th>hm</th>\n",
       "      <td>REACH000086</td>\n",
       "      <td>hm</td>\n",
       "      <td>F0078</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.70</td>\n",
       "      <td>C</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.044785</td>\n",
       "      <td>0.038065</td>\n",
       "      <td>0.054606</td>\n",
       "      <td>0.014648</td>\n",
       "      <td>0.018967</td>\n",
       "      <td>0.014095</td>\n",
       "      <td>-0.008569</td>\n",
       "      <td>0.010836</td>\n",
       "      <td>0.005924</td>\n",
       "      <td>-0.012543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">REACH000087</th>\n",
       "      <th>h1</th>\n",
       "      <td>REACH000087</td>\n",
       "      <td>h1</td>\n",
       "      <td>F0078</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2.75</td>\n",
       "      <td>M</td>\n",
       "      <td>30</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.129968</td>\n",
       "      <td>0.080387</td>\n",
       "      <td>0.132056</td>\n",
       "      <td>0.029583</td>\n",
       "      <td>0.020156</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>-0.002383</td>\n",
       "      <td>0.014468</td>\n",
       "      <td>0.006253</td>\n",
       "      <td>-0.018303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>h2</th>\n",
       "      <td>REACH000087</td>\n",
       "      <td>h2</td>\n",
       "      <td>F0078</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2.75</td>\n",
       "      <td>M</td>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>0.129968</td>\n",
       "      <td>0.080387</td>\n",
       "      <td>0.132056</td>\n",
       "      <td>0.029583</td>\n",
       "      <td>0.020156</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>-0.002383</td>\n",
       "      <td>0.014468</td>\n",
       "      <td>0.006253</td>\n",
       "      <td>-0.018303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000088</th>\n",
       "      <th>hm</th>\n",
       "      <td>REACH000088</td>\n",
       "      <td>hm</td>\n",
       "      <td>F0078</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3.30</td>\n",
       "      <td>D</td>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.034480</td>\n",
       "      <td>-0.003222</td>\n",
       "      <td>-0.007194</td>\n",
       "      <td>-0.001006</td>\n",
       "      <td>0.007012</td>\n",
       "      <td>0.020830</td>\n",
       "      <td>-0.009793</td>\n",
       "      <td>0.015290</td>\n",
       "      <td>-0.020905</td>\n",
       "      <td>-0.000667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000755</th>\n",
       "      <th>h1</th>\n",
       "      <td>REACH000755</td>\n",
       "      <td>h1</td>\n",
       "      <td>F0288</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>8.76</td>\n",
       "      <td>M</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.106540</td>\n",
       "      <td>0.060475</td>\n",
       "      <td>0.110081</td>\n",
       "      <td>0.040933</td>\n",
       "      <td>0.025530</td>\n",
       "      <td>0.026022</td>\n",
       "      <td>0.015113</td>\n",
       "      <td>0.014570</td>\n",
       "      <td>-0.019842</td>\n",
       "      <td>-0.013599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000762</th>\n",
       "      <th>hm</th>\n",
       "      <td>REACH000762</td>\n",
       "      <td>hm</td>\n",
       "      <td>F0290</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.79</td>\n",
       "      <td>C</td>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.027901</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>-0.004413</td>\n",
       "      <td>-0.008514</td>\n",
       "      <td>-0.003313</td>\n",
       "      <td>-0.000918</td>\n",
       "      <td>-0.001420</td>\n",
       "      <td>0.008548</td>\n",
       "      <td>0.001257</td>\n",
       "      <td>-0.000569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000763</th>\n",
       "      <th>h2</th>\n",
       "      <td>REACH000763</td>\n",
       "      <td>h2</td>\n",
       "      <td>F0290</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3.36</td>\n",
       "      <td>M</td>\n",
       "      <td>31</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.030630</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>-0.009075</td>\n",
       "      <td>-0.014136</td>\n",
       "      <td>-0.005499</td>\n",
       "      <td>-0.008555</td>\n",
       "      <td>0.015644</td>\n",
       "      <td>0.004869</td>\n",
       "      <td>0.011004</td>\n",
       "      <td>-0.019684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000766</th>\n",
       "      <th>h2</th>\n",
       "      <td>REACH000766</td>\n",
       "      <td>h2</td>\n",
       "      <td>F0290</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5.83</td>\n",
       "      <td>C</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.028469</td>\n",
       "      <td>-0.005167</td>\n",
       "      <td>-0.005808</td>\n",
       "      <td>-0.009013</td>\n",
       "      <td>-0.008074</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.007456</td>\n",
       "      <td>0.005629</td>\n",
       "      <td>0.002762</td>\n",
       "      <td>-0.007944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000767</th>\n",
       "      <th>hm</th>\n",
       "      <td>REACH000767</td>\n",
       "      <td>hm</td>\n",
       "      <td>F0291</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.31</td>\n",
       "      <td>C</td>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>0.087690</td>\n",
       "      <td>-0.012273</td>\n",
       "      <td>-0.017086</td>\n",
       "      <td>0.152272</td>\n",
       "      <td>-0.022385</td>\n",
       "      <td>0.008105</td>\n",
       "      <td>-0.002083</td>\n",
       "      <td>0.017964</td>\n",
       "      <td>-0.004649</td>\n",
       "      <td>-0.006845</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>218 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        subject hp_tag    fid Affected  Phenotype  sex  \\\n",
       "subject     hp_tag                                                       \n",
       "REACH000058 hm      REACH000058     hm  F0058      Yes          1    1   \n",
       "REACH000086 hm      REACH000086     hm  F0078      Yes          1    1   \n",
       "REACH000087 h1      REACH000087     h1  F0078       No          0    2   \n",
       "            h2      REACH000087     h2  F0078       No          0    2   \n",
       "REACH000088 hm      REACH000088     hm  F0078       No          0    1   \n",
       "...                         ...    ...    ...      ...        ...  ...   \n",
       "REACH000755 h1      REACH000755     h1  F0288       No          0    2   \n",
       "REACH000762 hm      REACH000762     hm  F0290      Yes          1    1   \n",
       "REACH000763 h2      REACH000763     h2  F0290       No          0    2   \n",
       "REACH000766 h2      REACH000766     h2  F0290       No          0    2   \n",
       "REACH000767 hm      REACH000767     hm  F0291      Yes          1    1   \n",
       "\n",
       "                    coverage rel  hp_rep_fmr1  num_reads_fmr1  ...       PC1  \\\n",
       "subject     hp_tag                                             ...             \n",
       "REACH000058 hm          6.19   C           37               7  ...  0.002422   \n",
       "REACH000086 hm          2.70   C           30               1  ...  0.044785   \n",
       "REACH000087 h1          2.75   M           30               1  ...  0.129968   \n",
       "            h2          2.75   M           30               4  ...  0.129968   \n",
       "REACH000088 hm          3.30   D           31               2  ... -0.034480   \n",
       "...                      ...  ..          ...             ...  ...       ...   \n",
       "REACH000755 h1          8.76   M           20               3  ...  0.106540   \n",
       "REACH000762 hm          3.79   C           32               1  ... -0.027901   \n",
       "REACH000763 h2          3.36   M           31               2  ... -0.030630   \n",
       "REACH000766 h2          5.83   C           31               1  ... -0.028469   \n",
       "REACH000767 hm          6.31   C           30               4  ...  0.087690   \n",
       "\n",
       "                         PC2       PC3       PC4       PC5       PC6  \\\n",
       "subject     hp_tag                                                     \n",
       "REACH000058 hm     -0.009216 -0.007762  0.073817  0.015720  0.006303   \n",
       "REACH000086 hm      0.038065  0.054606  0.014648  0.018967  0.014095   \n",
       "REACH000087 h1      0.080387  0.132056  0.029583  0.020156  0.007971   \n",
       "            h2      0.080387  0.132056  0.029583  0.020156  0.007971   \n",
       "REACH000088 hm     -0.003222 -0.007194 -0.001006  0.007012  0.020830   \n",
       "...                      ...       ...       ...       ...       ...   \n",
       "REACH000755 h1      0.060475  0.110081  0.040933  0.025530  0.026022   \n",
       "REACH000762 hm      0.000946 -0.004413 -0.008514 -0.003313 -0.000918   \n",
       "REACH000763 h2      0.003052 -0.009075 -0.014136 -0.005499 -0.008555   \n",
       "REACH000766 h2     -0.005167 -0.005808 -0.009013 -0.008074  0.000038   \n",
       "REACH000767 hm     -0.012273 -0.017086  0.152272 -0.022385  0.008105   \n",
       "\n",
       "                         PC7       PC8       PC9      PC10  \n",
       "subject     hp_tag                                          \n",
       "REACH000058 hm      0.055403 -0.038327 -0.011381  0.032498  \n",
       "REACH000086 hm     -0.008569  0.010836  0.005924 -0.012543  \n",
       "REACH000087 h1     -0.002383  0.014468  0.006253 -0.018303  \n",
       "            h2     -0.002383  0.014468  0.006253 -0.018303  \n",
       "REACH000088 hm     -0.009793  0.015290 -0.020905 -0.000667  \n",
       "...                      ...       ...       ...       ...  \n",
       "REACH000755 h1      0.015113  0.014570 -0.019842 -0.013599  \n",
       "REACH000762 hm     -0.001420  0.008548  0.001257 -0.000569  \n",
       "REACH000763 h2      0.015644  0.004869  0.011004 -0.019684  \n",
       "REACH000766 h2      0.007456  0.005629  0.002762 -0.007944  \n",
       "REACH000767 hm     -0.002083  0.017964 -0.004649 -0.006845  \n",
       "\n",
       "[218 rows x 26 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fragile_x_tables//methylation_table.tsv\n"
     ]
    }
   ],
   "source": [
    "#pd.set_option('display.max_rows', None)\n",
    "cols = ['hp_rep', 'num_reads', 'methylation_mean', 'methylation_mean_binary']\n",
    "#pd.set_option('display.max_rows', None)\n",
    "df_merge = pd.merge(df_hps_fmr1[cols], df_hps_xci[cols], left_index=True, right_index=True, how='inner', suffixes=('_fmr1', '_xci'))\n",
    "#df_merge = df_merge.loc[(df_merge.num_reads_fmr1>2)&(df_merge.num_reads_xci>2)]\n",
    "df_merge['subject'] = df_merge.index.map(lambda x: x[0])\n",
    "### correct subject for REACH000226\n",
    "df_merge['subject'] = df_merge.subject.apply(lambda x: 'REACH000226' if x == 'REACH000226_sperm' else x)\n",
    "df_merge['hp_tag'] = df_merge.index.map(lambda x: x[1])\n",
    "df_merge['fid'] = df_merge.subject.map(df_pca_flt.set_index('iid').fid)\n",
    "df_merge['Affected'] = df_merge.subject.map(aff_dict)\n",
    "df_merge['Phenotype'] = df_merge.Affected.map({'Yes': 1, 'No': 0})\n",
    "df_merge['sex'] = df_merge.subject.map(ds_sex)\n",
    "df_merge['coverage'] = df_merge.subject.map(cov_dict)\n",
    "df_merge['rel'] = df_merge.subject.map(df_samples.set_index('sample_id').rel)\n",
    "for i in range(1,11):\n",
    "    df_merge[f'PC{i}'] = df_merge.subject.map(df_pca_flt.set_index('iid')[f'PC{i}'])\n",
    "cols_old_order = df_merge.columns.tolist()\n",
    "cols_new_order = cols_old_order[-10-8:-10] + cols_old_order[:-10-8] + cols_old_order[-10:]\n",
    "df_merge = df_merge[cols_new_order]\n",
    "print('df_merge:')\n",
    "display(df_merge)\n",
    "pd.set_option('display.max_rows', 10)\n",
    "\n",
    "file_name = f'{dir_fig}/methylation_table.tsv'\n",
    "print(file_name)\n",
    "df_merge.to_csv(file_name, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f51cabd-cea8-490b-8d5d-88134624e119",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21003fe1-da4a-4804-87bf-aced24e12570",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fb275bc-5f6c-4f47-93e9-fd697b7e75eb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4117da0-22dd-46f8-82ab-4b39e4af69d0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b35558b-9f96-49b9-9ec5-cb0017a29ef4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcb4996a-5d1a-48a5-852b-d6b9d4cfd1f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d651f7e-2c22-44a0-b3aa-85d8eb78ba2f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9378adfd-7c54-45a8-a28f-daa368208bf3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "578f3f61-db15-4c44-b4a7-07b2de5671e9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "599df710-1de5-4ca1-8a06-f5c47a58b4ff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ac96e26-3aab-4b66-86bb-4d4f68444b29",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c7d43a0-6e2d-416a-8059-5c3c01e828df",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a9d09a0-68a5-4b24-86b0-40466220fa47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36dff0c4-66d2-44d0-b6f7-975b6689aad6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da5db62f-4cea-470e-bdc8-bc7d5b75bc17",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
