{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "73ddd28b-c089-4ad9-83f9-10cda1888539",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6cef5309-af72-4c15-8246-421bef454a50",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25\n"
     ]
    }
   ],
   "source": [
    "#dir_out = 'combined_tables/'\n",
    "#dir_out = 'combined_tables_snooptr/'\n",
    "dir_out = 'combined_tables_missing/'\n",
    "\n",
    "#file_trs = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/TRs_LongTR/FAM/variant_prior/main_df_toR_ZS3_5_7_9_len50_TRmissingness20_fltHomPol_famGT_DELINS_GW.csv'\n",
    "file_trs = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/TRs_LongTR/FAM/variant_prior/main_df_toR_ZS3_5_7_9_len50_TRmissingness25_fltHomPol_famGT_DELINS_GW.csv'\n",
    "tr_miss_level = file_trs.split('_TRmissingness')[1].rstrip('_fltHomPol_famGT_DELINS_GW.csv')\n",
    "print(tr_miss_level)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7a281a6d-40e9-4e93-9578-4bbc8296e1bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#input = 'sbayesr'\n",
    "input = 'prscs'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8c74538e-d146-4818-8da5-03e6b3065f0e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>NAMED_ALLELE_DOSAGE_SUM</th>\n",
       "      <th>SCORE1_AVG</th>\n",
       "      <th>PRS_ASD</th>\n",
       "      <th>PRS_ASD_Z</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000001</th>\n",
       "      <td>0</td>\n",
       "      <td>1644466</td>\n",
       "      <td>382413</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>5.44642</td>\n",
       "      <td>-0.971701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000026</th>\n",
       "      <td>0</td>\n",
       "      <td>1644484</td>\n",
       "      <td>397822</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>6.97789</td>\n",
       "      <td>0.303645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000050</th>\n",
       "      <td>0</td>\n",
       "      <td>1644468</td>\n",
       "      <td>379916</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>5.59728</td>\n",
       "      <td>-0.846071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <td>0</td>\n",
       "      <td>1644454</td>\n",
       "      <td>382247</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>7.32853</td>\n",
       "      <td>0.595643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000065</th>\n",
       "      <td>0</td>\n",
       "      <td>1644492</td>\n",
       "      <td>411824</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>8.01366</td>\n",
       "      <td>1.166192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000772</th>\n",
       "      <td>0</td>\n",
       "      <td>1644340</td>\n",
       "      <td>377760</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>6.72874</td>\n",
       "      <td>0.096163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000773</th>\n",
       "      <td>0</td>\n",
       "      <td>1644310</td>\n",
       "      <td>377377</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>6.07153</td>\n",
       "      <td>-0.451135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000774</th>\n",
       "      <td>0</td>\n",
       "      <td>1644376</td>\n",
       "      <td>379981</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>6.47407</td>\n",
       "      <td>-0.115916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000775</th>\n",
       "      <td>0</td>\n",
       "      <td>1644404</td>\n",
       "      <td>379560</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>6.72222</td>\n",
       "      <td>0.090733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000776</th>\n",
       "      <td>0</td>\n",
       "      <td>1644390</td>\n",
       "      <td>378641</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>5.70677</td>\n",
       "      <td>-0.754892</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>588 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             FID  ALLELE_CT  NAMED_ALLELE_DOSAGE_SUM  SCORE1_AVG  PRS_ASD  \\\n",
       "IID                                                                         \n",
       "REACH000001    0    1644466                   382413    0.000003  5.44642   \n",
       "REACH000026    0    1644484                   397822    0.000004  6.97789   \n",
       "REACH000050    0    1644468                   379916    0.000003  5.59728   \n",
       "REACH000058    0    1644454                   382247    0.000004  7.32853   \n",
       "REACH000065    0    1644492                   411824    0.000005  8.01366   \n",
       "...          ...        ...                      ...         ...      ...   \n",
       "REACH000772    0    1644340                   377760    0.000004  6.72874   \n",
       "REACH000773    0    1644310                   377377    0.000004  6.07153   \n",
       "REACH000774    0    1644376                   379981    0.000004  6.47407   \n",
       "REACH000775    0    1644404                   379560    0.000004  6.72222   \n",
       "REACH000776    0    1644390                   378641    0.000003  5.70677   \n",
       "\n",
       "             PRS_ASD_Z  \n",
       "IID                     \n",
       "REACH000001  -0.971701  \n",
       "REACH000026   0.303645  \n",
       "REACH000050  -0.846071  \n",
       "REACH000058   0.595643  \n",
       "REACH000065   1.166192  \n",
       "...                ...  \n",
       "REACH000772   0.096163  \n",
       "REACH000773  -0.451135  \n",
       "REACH000774  -0.115916  \n",
       "REACH000775   0.090733  \n",
       "REACH000776  -0.754892  \n",
       "\n",
       "[588 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>NAMED_ALLELE_DOSAGE_SUM</th>\n",
       "      <th>SCORE1_AVG</th>\n",
       "      <th>PRS_SCZ</th>\n",
       "      <th>PRS_SCZ_Z</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000001</th>\n",
       "      <td>0</td>\n",
       "      <td>1679560</td>\n",
       "      <td>385096</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>2.75926</td>\n",
       "      <td>-0.510485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000026</th>\n",
       "      <td>0</td>\n",
       "      <td>1679580</td>\n",
       "      <td>400980</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>4.61725</td>\n",
       "      <td>0.817056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000050</th>\n",
       "      <td>0</td>\n",
       "      <td>1679562</td>\n",
       "      <td>382494</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>2.23503</td>\n",
       "      <td>-0.885050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <td>0</td>\n",
       "      <td>1679550</td>\n",
       "      <td>384819</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>3.04624</td>\n",
       "      <td>-0.305437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000065</th>\n",
       "      <td>0</td>\n",
       "      <td>1679588</td>\n",
       "      <td>415682</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>4.79451</td>\n",
       "      <td>0.943709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000772</th>\n",
       "      <td>0</td>\n",
       "      <td>1679434</td>\n",
       "      <td>380199</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>3.33420</td>\n",
       "      <td>-0.099688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000773</th>\n",
       "      <td>0</td>\n",
       "      <td>1679406</td>\n",
       "      <td>379724</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>2.73289</td>\n",
       "      <td>-0.529327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000774</th>\n",
       "      <td>0</td>\n",
       "      <td>1679472</td>\n",
       "      <td>382283</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>3.20694</td>\n",
       "      <td>-0.190616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000775</th>\n",
       "      <td>0</td>\n",
       "      <td>1679498</td>\n",
       "      <td>381953</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>3.44894</td>\n",
       "      <td>-0.017706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000776</th>\n",
       "      <td>0</td>\n",
       "      <td>1679482</td>\n",
       "      <td>381088</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>3.56207</td>\n",
       "      <td>0.063126</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>588 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             FID  ALLELE_CT  NAMED_ALLELE_DOSAGE_SUM  SCORE1_AVG  PRS_SCZ  \\\n",
       "IID                                                                         \n",
       "REACH000001    0    1679560                   385096    0.000002  2.75926   \n",
       "REACH000026    0    1679580                   400980    0.000003  4.61725   \n",
       "REACH000050    0    1679562                   382494    0.000001  2.23503   \n",
       "REACH000058    0    1679550                   384819    0.000002  3.04624   \n",
       "REACH000065    0    1679588                   415682    0.000003  4.79451   \n",
       "...          ...        ...                      ...         ...      ...   \n",
       "REACH000772    0    1679434                   380199    0.000002  3.33420   \n",
       "REACH000773    0    1679406                   379724    0.000002  2.73289   \n",
       "REACH000774    0    1679472                   382283    0.000002  3.20694   \n",
       "REACH000775    0    1679498                   381953    0.000002  3.44894   \n",
       "REACH000776    0    1679482                   381088    0.000002  3.56207   \n",
       "\n",
       "             PRS_SCZ_Z  \n",
       "IID                     \n",
       "REACH000001  -0.510485  \n",
       "REACH000026   0.817056  \n",
       "REACH000050  -0.885050  \n",
       "REACH000058  -0.305437  \n",
       "REACH000065   0.943709  \n",
       "...                ...  \n",
       "REACH000772  -0.099688  \n",
       "REACH000773  -0.529327  \n",
       "REACH000774  -0.190616  \n",
       "REACH000775  -0.017706  \n",
       "REACH000776   0.063126  \n",
       "\n",
       "[588 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>NAMED_ALLELE_DOSAGE_SUM</th>\n",
       "      <th>SCORE1_AVG</th>\n",
       "      <th>PRS_EDU</th>\n",
       "      <th>PRS_EDU_Z</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000001</th>\n",
       "      <td>0</td>\n",
       "      <td>1682848</td>\n",
       "      <td>385163</td>\n",
       "      <td>-1.601140e-07</td>\n",
       "      <td>-0.269458</td>\n",
       "      <td>0.821629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000026</th>\n",
       "      <td>0</td>\n",
       "      <td>1682868</td>\n",
       "      <td>401083</td>\n",
       "      <td>-3.094060e-07</td>\n",
       "      <td>-0.520703</td>\n",
       "      <td>0.063124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000050</th>\n",
       "      <td>0</td>\n",
       "      <td>1682850</td>\n",
       "      <td>382503</td>\n",
       "      <td>-3.417750e-07</td>\n",
       "      <td>-0.575177</td>\n",
       "      <td>-0.101332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <td>0</td>\n",
       "      <td>1682838</td>\n",
       "      <td>384860</td>\n",
       "      <td>6.060490e-08</td>\n",
       "      <td>0.101993</td>\n",
       "      <td>1.943035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000065</th>\n",
       "      <td>0</td>\n",
       "      <td>1682878</td>\n",
       "      <td>415752</td>\n",
       "      <td>-8.707660e-07</td>\n",
       "      <td>-1.465420</td>\n",
       "      <td>-2.788964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000772</th>\n",
       "      <td>0</td>\n",
       "      <td>1682720</td>\n",
       "      <td>380244</td>\n",
       "      <td>-2.805610e-07</td>\n",
       "      <td>-0.472160</td>\n",
       "      <td>0.209675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000773</th>\n",
       "      <td>0</td>\n",
       "      <td>1682694</td>\n",
       "      <td>379734</td>\n",
       "      <td>-1.557100e-07</td>\n",
       "      <td>-0.262047</td>\n",
       "      <td>0.844003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000774</th>\n",
       "      <td>0</td>\n",
       "      <td>1682760</td>\n",
       "      <td>382325</td>\n",
       "      <td>-2.574510e-07</td>\n",
       "      <td>-0.433268</td>\n",
       "      <td>0.327089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000775</th>\n",
       "      <td>0</td>\n",
       "      <td>1682784</td>\n",
       "      <td>381961</td>\n",
       "      <td>-5.100050e-08</td>\n",
       "      <td>-0.085829</td>\n",
       "      <td>1.376002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000776</th>\n",
       "      <td>0</td>\n",
       "      <td>1682770</td>\n",
       "      <td>381172</td>\n",
       "      <td>-1.362810e-07</td>\n",
       "      <td>-0.229349</td>\n",
       "      <td>0.942718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>588 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             FID  ALLELE_CT  NAMED_ALLELE_DOSAGE_SUM    SCORE1_AVG   PRS_EDU  \\\n",
       "IID                                                                            \n",
       "REACH000001    0    1682848                   385163 -1.601140e-07 -0.269458   \n",
       "REACH000026    0    1682868                   401083 -3.094060e-07 -0.520703   \n",
       "REACH000050    0    1682850                   382503 -3.417750e-07 -0.575177   \n",
       "REACH000058    0    1682838                   384860  6.060490e-08  0.101993   \n",
       "REACH000065    0    1682878                   415752 -8.707660e-07 -1.465420   \n",
       "...          ...        ...                      ...           ...       ...   \n",
       "REACH000772    0    1682720                   380244 -2.805610e-07 -0.472160   \n",
       "REACH000773    0    1682694                   379734 -1.557100e-07 -0.262047   \n",
       "REACH000774    0    1682760                   382325 -2.574510e-07 -0.433268   \n",
       "REACH000775    0    1682784                   381961 -5.100050e-08 -0.085829   \n",
       "REACH000776    0    1682770                   381172 -1.362810e-07 -0.229349   \n",
       "\n",
       "             PRS_EDU_Z  \n",
       "IID                     \n",
       "REACH000001   0.821629  \n",
       "REACH000026   0.063124  \n",
       "REACH000050  -0.101332  \n",
       "REACH000058   1.943035  \n",
       "REACH000065  -2.788964  \n",
       "...                ...  \n",
       "REACH000772   0.209675  \n",
       "REACH000773   0.844003  \n",
       "REACH000774   0.327089  \n",
       "REACH000775   1.376002  \n",
       "REACH000776   0.942718  \n",
       "\n",
       "[588 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if (input == 'sbayesr'):\n",
    "    file_prs_asd = '/expanse/projects/sebat1/j3guevar/misc/asd_prs/prs_scripts_combined_analysis/plink.asd.scores.reach_rename.txt'\n",
    "    file_prs_scz = '/expanse/projects/sebat1/j3guevar/misc/asd_prs/prs_scripts_combined_analysis/plink.scz.scores.reach_rename.txt'\n",
    "    file_prs_edu = '/expanse/projects/sebat1/j3guevar/misc/asd_prs/prs_scripts_combined_analysis/plink.edu.scores.reach_rename.txt'\n",
    "elif (input == 'prscs'):\n",
    "    #FID    IID ALLELE_CT   NAMED_ALLELE_DOSAGE_SUM SCORE1_AVG  SCORE1_SUM\n",
    "    file_prs_asd = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/PRS/REACH_daner_PGCiPSYCH_loSSC_cleaned_pst_eff_a1_b0.5_phi1e-02_all_chrs.txt.sscore'\n",
    "    file_prs_scz = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/PRS/REACH_ckqny.scz2snpres_pst_eff_a1_b0.5_phi1e-02_all_chrs.txt.sscore'\n",
    "    file_prs_edu = '/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/PRS/REACH_GWAS_EA_excl23andMe_pst_eff_a1_b0.5_phi1e-02_all_chrs.txt.sscore'\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "\n",
    "df_prs_asd = pd.read_table(file_prs_asd, sep='\\t', header=0)\n",
    "if (input == 'sbayesr'):\n",
    "    df_prs_asd.columns = ['IID', 'PRS_ASD']\n",
    "    df_prs_asd = df_prs_asd.loc[df_prs_asd.IID.str.contains('REACH000')]\n",
    "elif (input == 'prscs'):\n",
    "    df_prs_asd.columns = ['FID', 'IID', 'ALLELE_CT', 'NAMED_ALLELE_DOSAGE_SUM', 'SCORE1_AVG', 'PRS_ASD']\n",
    "\n",
    "this_mean = df_prs_asd.PRS_ASD.mean()\n",
    "this_std = df_prs_asd.PRS_ASD.std()\n",
    "df_prs_asd['PRS_ASD_Z'] = (df_prs_asd['PRS_ASD'] - this_mean) / this_std\n",
    "df_prs_asd['IID'] = df_prs_asd.apply(lambda row: rename_samples_dict[row['IID']] \n",
    "                                     if row['IID'] in rename_samples_dict else row['IID'], axis=1)\n",
    "df_prs_asd.set_index('IID', inplace=True)\n",
    "display(df_prs_asd)\n",
    "\n",
    "\n",
    "df_prs_scz = pd.read_table(file_prs_scz, sep='\\t', header=0)\n",
    "if (input == 'sbayesr'):\n",
    "    df_prs_scz.columns = ['IID', 'PRS_SCZ']\n",
    "    df_prs_scz = df_prs_scz.loc[df_prs_scz.IID.str.contains('REACH000')]\n",
    "elif (input == 'prscs'):\n",
    "    df_prs_scz.columns = ['FID', 'IID', 'ALLELE_CT', 'NAMED_ALLELE_DOSAGE_SUM', 'SCORE1_AVG', 'PRS_SCZ']\n",
    "\n",
    "this_mean = df_prs_scz.PRS_SCZ.mean()\n",
    "this_std = df_prs_scz.PRS_SCZ.std()\n",
    "df_prs_scz['PRS_SCZ_Z'] = (df_prs_scz['PRS_SCZ'] - this_mean) / this_std\n",
    "df_prs_scz['IID'] = df_prs_scz.apply(lambda row: rename_samples_dict[row['IID']] \n",
    "                                     if row['IID'] in rename_samples_dict else row['IID'], axis=1)\n",
    "df_prs_scz.set_index('IID', inplace=True)\n",
    "display(df_prs_scz)\n",
    "\n",
    "df_prs_edu = pd.read_table(file_prs_edu, sep='\\t', header=0)\n",
    "if (input == 'sbayesr'):\n",
    "    df_prs_edu.columns = ['IID', 'PRS_EDU']\n",
    "    df_prs_edu = df_prs_edu.loc[df_prs_edu.IID.str.contains('REACH000')]\n",
    "elif (input == 'prscs'):\n",
    "    df_prs_edu.columns = ['FID', 'IID', 'ALLELE_CT', 'NAMED_ALLELE_DOSAGE_SUM', 'SCORE1_AVG', 'PRS_EDU']\n",
    "\n",
    "this_mean = df_prs_edu.PRS_EDU.mean()\n",
    "this_std = df_prs_edu.PRS_EDU.std()\n",
    "df_prs_edu['PRS_EDU_Z'] = (df_prs_edu['PRS_EDU'] - this_mean) / this_std\n",
    "df_prs_edu['IID'] = df_prs_edu.apply(lambda row: rename_samples_dict[row['IID']] \n",
    "                                     if row['IID'] in rename_samples_dict else row['IID'], axis=1)\n",
    "df_prs_edu.set_index('IID', inplace=True)\n",
    "display(df_prs_edu)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f399027a-1897-4765-8fea-c00cd2173e05",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5824, 32)\n",
      "5824\n",
      "3045\n",
      "1983\n",
      "1392\n",
      "1078\n"
     ]
    }
   ],
   "source": [
    "file_brainspan = '/expanse/projects/sebat1/miladm/UCSD/resources/brainspan_rnaseq/process/gene_list_RPKM_10.csv'\n",
    "dtype = {'gene_id': str, 'entrez_id': str}\n",
    "df_br = pd.read_table(file_brainspan, sep=',', header=0, dtype=dtype)\n",
    "br_flt_10 = df_br.loc[df_br.mean_rpkm>10].gene_symbol.tolist()\n",
    "br_flt_20 = df_br.loc[df_br.mean_rpkm>20].gene_symbol.tolist()\n",
    "br_flt_30 = df_br.loc[df_br.mean_rpkm>30].gene_symbol.tolist()\n",
    "br_flt_40 = df_br.loc[df_br.mean_rpkm>40].gene_symbol.tolist()\n",
    "br_flt_50 = df_br.loc[df_br.mean_rpkm>50].gene_symbol.tolist()\n",
    "print(df_br.shape)\n",
    "print(len(br_flt_10))\n",
    "print(len(br_flt_20))\n",
    "print(len(br_flt_30))\n",
    "print(len(br_flt_40))\n",
    "print(len(br_flt_50))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0b554be3-fc66-470d-a268-066209296b68",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_fdr:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_gencodeV33</th>\n",
       "      <th>FDR_TADA_ASD</th>\n",
       "      <th>FDR_TADA_DD</th>\n",
       "      <th>FDR_TADA_NDD</th>\n",
       "      <th>p_TADA_ASD</th>\n",
       "      <th>p_TADA_DD</th>\n",
       "      <th>p_TADA_NDD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNGAP1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHD8</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADNP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SHANK3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SCN2A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18123</th>\n",
       "      <td>PSG2</td>\n",
       "      <td>0.911036</td>\n",
       "      <td>0.913971</td>\n",
       "      <td>0.879392</td>\n",
       "      <td>0.999770</td>\n",
       "      <td>0.999656</td>\n",
       "      <td>0.999762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18124</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.911038</td>\n",
       "      <td>0.913973</td>\n",
       "      <td>0.879395</td>\n",
       "      <td>0.999827</td>\n",
       "      <td>0.999713</td>\n",
       "      <td>0.999821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18125</th>\n",
       "      <td>GJB2</td>\n",
       "      <td>0.911040</td>\n",
       "      <td>0.913975</td>\n",
       "      <td>0.879399</td>\n",
       "      <td>0.999885</td>\n",
       "      <td>0.999771</td>\n",
       "      <td>0.999881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18126</th>\n",
       "      <td>PSG1</td>\n",
       "      <td>0.911042</td>\n",
       "      <td>0.913979</td>\n",
       "      <td>0.879403</td>\n",
       "      <td>0.999942</td>\n",
       "      <td>0.999885</td>\n",
       "      <td>0.999940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18127</th>\n",
       "      <td>NBPF10</td>\n",
       "      <td>0.911044</td>\n",
       "      <td>0.913981</td>\n",
       "      <td>0.879407</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999943</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18128 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_gencodeV33  FDR_TADA_ASD  FDR_TADA_DD  FDR_TADA_NDD  p_TADA_ASD  \\\n",
       "0             SYNGAP1      0.000000     0.000000      0.000000    0.000000   \n",
       "1                CHD8      0.000000     0.000000      0.000000    0.000000   \n",
       "2                ADNP      0.000000     0.000000      0.000000    0.000000   \n",
       "3              SHANK3      0.000000     0.000000      0.000000    0.000000   \n",
       "4               SCN2A      0.000000     0.000000      0.000000    0.000000   \n",
       "...               ...           ...          ...           ...         ...   \n",
       "18123            PSG2      0.911036     0.913971      0.879392    0.999770   \n",
       "18124             NaN      0.911038     0.913973      0.879395    0.999827   \n",
       "18125            GJB2      0.911040     0.913975      0.879399    0.999885   \n",
       "18126            PSG1      0.911042     0.913979      0.879403    0.999942   \n",
       "18127          NBPF10      0.911044     0.913981      0.879407    1.000000   \n",
       "\n",
       "       p_TADA_DD  p_TADA_NDD  \n",
       "0       0.000000    0.000000  \n",
       "1       0.000000    0.000000  \n",
       "2       0.000000    0.000000  \n",
       "3       0.000000    0.000000  \n",
       "4       0.000000    0.000000  \n",
       "...          ...         ...  \n",
       "18123   0.999656    0.999762  \n",
       "18124   0.999713    0.999821  \n",
       "18125   0.999771    0.999881  \n",
       "18126   0.999885    0.999940  \n",
       "18127   0.999943    1.000000  \n",
       "\n",
       "[18128 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "asd: 185\n",
      "dd: 477\n",
      "ndd: 664\n",
      "df_pli:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lof.pLI</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.814400e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>6.691700e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>7.412300e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.996200e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>1.007500e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11A</th>\n",
       "      <td>3.084800e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>8.055900e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>3.807700e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>9.990700e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18644 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              lof.pLI\n",
       "gene                 \n",
       "A1BG     1.814400e-07\n",
       "A1CF     6.691700e-03\n",
       "A2M      7.412300e-02\n",
       "A2ML1    1.996200e-01\n",
       "A3GALT2  1.007500e-06\n",
       "...               ...\n",
       "ZYG11A   3.084800e-08\n",
       "ZYG11B   8.055900e-01\n",
       "ZYX      3.807700e-02\n",
       "ZZEF1    9.990700e-01\n",
       "ZZZ3     1.000000e+00\n",
       "\n",
       "[18644 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_lf:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lof.oe_ci.upper</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>1.530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11A</th>\n",
       "      <td>1.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0.854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.290</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18644 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         lof.oe_ci.upper\n",
       "gene                    \n",
       "A1BG               1.257\n",
       "A1CF               0.789\n",
       "A2M                0.766\n",
       "A2ML1              0.829\n",
       "A3GALT2            1.530\n",
       "...                  ...\n",
       "ZYG11A             1.002\n",
       "ZYG11B             0.537\n",
       "ZYX                0.854\n",
       "ZZEF1              0.456\n",
       "ZZZ3               0.290\n",
       "\n",
       "[18644 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dnLOF and dnMIS SNV/indels\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(30, 22)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " fdr asd: 6\n",
      " fdr dd: 11\n",
      " fdr ndd: 13\n",
      " fdr brain genes: 21\n",
      "df_dn_snv:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start_hg38</th>\n",
       "      <th>end_hg38</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>alternate_allele</th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>sex</th>\n",
       "      <th>consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>gene</th>\n",
       "      <th>gnomad_loeuf</th>\n",
       "      <th>mpc</th>\n",
       "      <th>cohort</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2173976</td>\n",
       "      <td>2173977</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000089</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>0.4</td>\n",
       "      <td>2.260710742</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99979</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16251664</td>\n",
       "      <td>16251665</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0279</td>\n",
       "      <td>REACH000718</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>FBXO42</td>\n",
       "      <td>0.274</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chr1</td>\n",
       "      <td>85464960</td>\n",
       "      <td>85464961</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>F0262</td>\n",
       "      <td>REACH000650</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDAH1</td>\n",
       "      <td>0.952</td>\n",
       "      <td>3.362419638</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.15438</td>\n",
       "      <td>0.862</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>61303998</td>\n",
       "      <td>61303999</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0285</td>\n",
       "      <td>REACH000741</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDB1</td>\n",
       "      <td>0.243</td>\n",
       "      <td>3.079873048</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>chr11</td>\n",
       "      <td>64773439</td>\n",
       "      <td>64773440</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0162B</td>\n",
       "      <td>REACH000774</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>SF1</td>\n",
       "      <td>0.183</td>\n",
       "      <td>2.419035056</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>chr12</td>\n",
       "      <td>57269898</td>\n",
       "      <td>57269900</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>F0083</td>\n",
       "      <td>REACH000097</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>R3HDM2</td>\n",
       "      <td>0.176</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>chr12</td>\n",
       "      <td>94298718</td>\n",
       "      <td>94298719</td>\n",
       "      <td>T</td>\n",
       "      <td>TTTGAC</td>\n",
       "      <td>F0274</td>\n",
       "      <td>REACH000700</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>PLXNC1</td>\n",
       "      <td>0.273</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.75987</td>\n",
       "      <td>0.550</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23034800</td>\n",
       "      <td>23034802</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>F0251</td>\n",
       "      <td>REACH000605</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>PSMB5</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99989</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>chr16</td>\n",
       "      <td>2056194</td>\n",
       "      <td>2056195</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0233</td>\n",
       "      <td>REACH000529</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.074</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.168</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>chr16</td>\n",
       "      <td>67629532</td>\n",
       "      <td>67629533</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>F0242</td>\n",
       "      <td>REACH000564</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained&amp;splice_region_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>0.148</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.205</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>526</th>\n",
       "      <td>chr17</td>\n",
       "      <td>8466944</td>\n",
       "      <td>8466945</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>F0203</td>\n",
       "      <td>REACH000404</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>...</td>\n",
       "      <td>NDEL1</td>\n",
       "      <td>0.296</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99993</td>\n",
       "      <td>0.324</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>chr17</td>\n",
       "      <td>59647575</td>\n",
       "      <td>59647576</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>F0245</td>\n",
       "      <td>REACH000578</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CLTC</td>\n",
       "      <td>0.104</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.100</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>chr17</td>\n",
       "      <td>67886275</td>\n",
       "      <td>67886278</td>\n",
       "      <td>AAG</td>\n",
       "      <td>A</td>\n",
       "      <td>F0221</td>\n",
       "      <td>REACH000482</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>BPTF</td>\n",
       "      <td>0.099</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.083</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6213087</td>\n",
       "      <td>6213088</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0208</td>\n",
       "      <td>REACH000423</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>MLLT1</td>\n",
       "      <td>0.288</td>\n",
       "      <td>2.086213226</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49374625</td>\n",
       "      <td>49374626</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0284</td>\n",
       "      <td>REACH000747</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>KCNB1</td>\n",
       "      <td>0.125</td>\n",
       "      <td>3.839540893</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.309</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>chr20</td>\n",
       "      <td>63442515</td>\n",
       "      <td>63442516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>F0215</td>\n",
       "      <td>REACH000457</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>KCNQ2</td>\n",
       "      <td>0.158</td>\n",
       "      <td>4.219752135</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.251</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>chr21</td>\n",
       "      <td>37490200</td>\n",
       "      <td>37490201</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>F0210</td>\n",
       "      <td>REACH000435</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>DYRK1A</td>\n",
       "      <td>0.214</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.172</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38300756</td>\n",
       "      <td>38300757</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0230</td>\n",
       "      <td>REACH000519</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CSNK1E</td>\n",
       "      <td>0.143</td>\n",
       "      <td>2.503748791</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.92771</td>\n",
       "      <td>0.555</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41129986</td>\n",
       "      <td>41129992</td>\n",
       "      <td>TAAGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>F0226</td>\n",
       "      <td>REACH000504</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>EP300</td>\n",
       "      <td>0.099</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>chr3</td>\n",
       "      <td>70972633</td>\n",
       "      <td>70972634</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0140</td>\n",
       "      <td>REACH000189</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>FOXP1</td>\n",
       "      <td>0.175</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.250</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>chr4</td>\n",
       "      <td>46303465</td>\n",
       "      <td>46303466</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>F0221</td>\n",
       "      <td>REACH000482</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>GABRA2</td>\n",
       "      <td>0.229</td>\n",
       "      <td>2.615526082</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.98587</td>\n",
       "      <td>0.481</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>chr4</td>\n",
       "      <td>109511995</td>\n",
       "      <td>109511996</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>F0237</td>\n",
       "      <td>REACH000544</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>SEC24B</td>\n",
       "      <td>0.194</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.418</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>chr5</td>\n",
       "      <td>10255973</td>\n",
       "      <td>10255974</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>F0248</td>\n",
       "      <td>REACH000589</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_lost&amp;NMD_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CCT5</td>\n",
       "      <td>0.188</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.210</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>chr5</td>\n",
       "      <td>150251807</td>\n",
       "      <td>150251808</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0267</td>\n",
       "      <td>REACH000664</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CAMK2A</td>\n",
       "      <td>0.251</td>\n",
       "      <td>2.923086944</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.141</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>chr5</td>\n",
       "      <td>173110640</td>\n",
       "      <td>173110641</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>F0288</td>\n",
       "      <td>REACH000754</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CREBRF</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2.210301239</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.174</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>chr6</td>\n",
       "      <td>43213912</td>\n",
       "      <td>43213913</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0167</td>\n",
       "      <td>REACH000249</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CUL9</td>\n",
       "      <td>0.28</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.321</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>chr7</td>\n",
       "      <td>74700299</td>\n",
       "      <td>74700300</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>F0191</td>\n",
       "      <td>REACH000349</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>GTF2I</td>\n",
       "      <td>0.27</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99996</td>\n",
       "      <td>0.195</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>chr9</td>\n",
       "      <td>81585545</td>\n",
       "      <td>81585546</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>F0276</td>\n",
       "      <td>REACH000708</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>TLE1</td>\n",
       "      <td>0.239</td>\n",
       "      <td>2.626466068</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.269</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1324</th>\n",
       "      <td>chrX</td>\n",
       "      <td>41344087</td>\n",
       "      <td>41344090</td>\n",
       "      <td>CGA</td>\n",
       "      <td>C</td>\n",
       "      <td>F0221</td>\n",
       "      <td>REACH000479</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX3X</td>\n",
       "      <td>0.118</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1325</th>\n",
       "      <td>chrX</td>\n",
       "      <td>41346519</td>\n",
       "      <td>41346520</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>F0240</td>\n",
       "      <td>REACH000555</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX3X</td>\n",
       "      <td>0.118</td>\n",
       "      <td>2.375896712</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom  start_hg38   end_hg38 reference_allele alternate_allele     fid  \\\n",
       "0      chr1     2173976    2173977                G                A   F0078   \n",
       "17     chr1    16251664   16251665                G                A   F0279   \n",
       "54     chr1    85464960   85464961                C                G   F0262   \n",
       "208   chr11    61303998   61303999                G                A   F0285   \n",
       "219   chr11    64773439   64773440                G                A  F0162B   \n",
       "299   chr12    57269898   57269900               TG                T   F0083   \n",
       "313   chr12    94298718   94298719                T           TTTGAC   F0274   \n",
       "382   chr14    23034800   23034802               CA                C   F0251   \n",
       "465   chr16     2056194    2056195                G                A   F0233   \n",
       "501   chr16    67629532   67629533                G                T   F0242   \n",
       "526   chr17     8466944    8466945                T                C   F0203   \n",
       "580   chr17    59647575   59647576                T               TC   F0245   \n",
       "596   chr17    67886275   67886278              AAG                A   F0221   \n",
       "647   chr19     6213087    6213088                G                A   F0208   \n",
       "838   chr20    49374625   49374626                G                A   F0284   \n",
       "857   chr20    63442515   63442516                A                G   F0215   \n",
       "870   chr21    37490200   37490201                C                T   F0210   \n",
       "890   chr22    38300756   38300757                G                A   F0230   \n",
       "896   chr22    41129986   41129992           TAAGAG                T   F0226   \n",
       "948    chr3    70972633   70972634                G                A   F0140   \n",
       "995    chr4    46303465   46303466                C                G   F0221   \n",
       "1021   chr4   109511995  109511996                C                T   F0237   \n",
       "1044   chr5    10255973   10255974                A                G   F0248   \n",
       "1087   chr5   150251807  150251808                G                A   F0267   \n",
       "1100   chr5   173110640  173110641                A                T   F0288   \n",
       "1136   chr6    43213912   43213913                G                A   F0167   \n",
       "1203   chr7    74700299   74700300                C                T   F0191   \n",
       "1283   chr9    81585545   81585546                A                G   F0276   \n",
       "1324   chrX    41344087   41344090              CGA                C   F0221   \n",
       "1325   chrX    41346519   41346520                C                G   F0240   \n",
       "\n",
       "              iid phenotype     sex                        consequence  ...  \\\n",
       "0     REACH000089       ASD  Female                   missense_variant  ...   \n",
       "17    REACH000718       ASD    Male                        stop_gained  ...   \n",
       "54    REACH000650       ASD    Male                   missense_variant  ...   \n",
       "208   REACH000741   Control    Male                   missense_variant  ...   \n",
       "219   REACH000774   Control    Male                   missense_variant  ...   \n",
       "299   REACH000097       ASD    Male                 frameshift_variant  ...   \n",
       "313   REACH000700   Control    Male                 frameshift_variant  ...   \n",
       "382   REACH000605       ASD    Male                 frameshift_variant  ...   \n",
       "465   REACH000529       ASD    Male            splice_acceptor_variant  ...   \n",
       "501   REACH000564       ASD    Male  stop_gained&splice_region_variant  ...   \n",
       "526   REACH000404       ASD  Female                          stop_lost  ...   \n",
       "580   REACH000578       ASD  Female                 frameshift_variant  ...   \n",
       "596   REACH000482       ASD  Female                 frameshift_variant  ...   \n",
       "647   REACH000423       ASD    Male                   missense_variant  ...   \n",
       "838   REACH000747       ASD    Male                   missense_variant  ...   \n",
       "857   REACH000457       ASD  Female                   missense_variant  ...   \n",
       "870   REACH000435       ASD  Female                        stop_gained  ...   \n",
       "890   REACH000519       ASD  Female                   missense_variant  ...   \n",
       "896   REACH000504       ASD  Female                 frameshift_variant  ...   \n",
       "948   REACH000189       ASD    Male                        stop_gained  ...   \n",
       "995   REACH000482       ASD  Female                   missense_variant  ...   \n",
       "1021  REACH000544       ASD    Male                        stop_gained  ...   \n",
       "1044  REACH000589       ASD    Male   stop_lost&NMD_transcript_variant  ...   \n",
       "1087  REACH000664       ASD    Male                   missense_variant  ...   \n",
       "1100  REACH000754       ASD    Male                   missense_variant  ...   \n",
       "1136  REACH000249       ASD    Male               splice_donor_variant  ...   \n",
       "1203  REACH000349       ASD    Male                        stop_gained  ...   \n",
       "1283  REACH000708   Control    Male                   missense_variant  ...   \n",
       "1324  REACH000479       ASD  Female                 frameshift_variant  ...   \n",
       "1325  REACH000555       ASD  Female                   missense_variant  ...   \n",
       "\n",
       "        gene  gnomad_loeuf          mpc cohort      PLI  LOEUF  FDR_ASD  \\\n",
       "0      PRKCZ           0.4  2.260710742  REACH  0.99979  0.415        0   \n",
       "17    FBXO42         0.274           NA  REACH  1.00000  0.271        0   \n",
       "54     DDAH1         0.952  3.362419638  REACH  0.15438  0.862        0   \n",
       "208     DDB1         0.243  3.079873048  REACH  1.00000  0.183        0   \n",
       "219      SF1         0.183  2.419035056  REACH  1.00000  0.220        0   \n",
       "299   R3HDM2         0.176           NA  REACH  1.00000  0.366        0   \n",
       "313   PLXNC1         0.273           NA  REACH  0.75987  0.550        0   \n",
       "382    PSMB5           0.3           NA  REACH  0.99989  0.238        0   \n",
       "465     TSC2         0.074           NA  REACH  1.00000  0.168        0   \n",
       "501     CTCF         0.148           NA  REACH  1.00000  0.205        1   \n",
       "526    NDEL1         0.296           NA  REACH  0.99993  0.324        0   \n",
       "580     CLTC         0.104           NA  REACH  1.00000  0.100        0   \n",
       "596     BPTF         0.099           NA  REACH  1.00000  0.083        0   \n",
       "647    MLLT1         0.288  2.086213226  REACH  0.99994  0.379        0   \n",
       "838    KCNB1         0.125  3.839540893  REACH  1.00000  0.309        1   \n",
       "857    KCNQ2         0.158  4.219752135  REACH  1.00000  0.251        0   \n",
       "870   DYRK1A         0.214           NA  REACH  1.00000  0.172        1   \n",
       "890   CSNK1E         0.143  2.503748791  REACH  0.92771  0.555        1   \n",
       "896    EP300         0.099           NA  REACH  1.00000  0.163        0   \n",
       "948    FOXP1         0.175           NA  REACH  1.00000  0.250        1   \n",
       "995   GABRA2         0.229  2.615526082  REACH  0.98587  0.481        0   \n",
       "1021  SEC24B         0.194           NA  REACH  0.99994  0.418        0   \n",
       "1044    CCT5         0.188           NA  REACH  1.00000  0.210        0   \n",
       "1087  CAMK2A         0.251  2.923086944  REACH  1.00000  0.141        1   \n",
       "1100  CREBRF           0.1  2.210301239  REACH  1.00000  0.174        0   \n",
       "1136    CUL9          0.28           NA  REACH  1.00000  0.321        0   \n",
       "1203   GTF2I          0.27           NA  REACH  0.99996  0.195        0   \n",
       "1283    TLE1         0.239  2.626466068  REACH  1.00000  0.269        0   \n",
       "1324   DDX3X         0.118           NA  REACH  0.99986  0.118        0   \n",
       "1325   DDX3X         0.118  2.375896712  REACH  0.99986  0.118        0   \n",
       "\n",
       "      FDR_DD  FDR_NDD  DEV_BR_GENE_10  \n",
       "0          0        0               1  \n",
       "17         0        0               0  \n",
       "54         0        0               1  \n",
       "208        0        1               1  \n",
       "219        0        1               1  \n",
       "299        0        0               1  \n",
       "313        0        0               0  \n",
       "382        0        0               1  \n",
       "465        1        1               1  \n",
       "501        1        1               1  \n",
       "526        0        0               1  \n",
       "580        1        1               1  \n",
       "596        1        1               1  \n",
       "647        0        0               1  \n",
       "838        1        1               0  \n",
       "857        1        1               1  \n",
       "870        1        1               1  \n",
       "890        1        1               1  \n",
       "896        1        1               1  \n",
       "948        1        1               0  \n",
       "995        0        0               0  \n",
       "1021       0        0               1  \n",
       "1044       0        0               1  \n",
       "1087       1        1               0  \n",
       "1100       0        0               0  \n",
       "1136       0        0               0  \n",
       "1203       0        0               0  \n",
       "1283       0        0               1  \n",
       "1324       0        0               1  \n",
       "1325       0        0               1  \n",
       "\n",
       "[30 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this_df_dn:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(28, 9)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loss_of_function</th>\n",
       "      <th>missense</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "      <th>samples_all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99979</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99989</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.168</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.205</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>526</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99993</td>\n",
       "      <td>0.324</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.100</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.083</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.309</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.251</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.172</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.92771</td>\n",
       "      <td>0.555</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.250</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.98587</td>\n",
       "      <td>0.481</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.418</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.210</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.141</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.174</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.321</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99996</td>\n",
       "      <td>0.195</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.269</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1324</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1325</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      loss_of_function  missense      PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  \\\n",
       "0                    0         1  0.99979  0.415        0       0        0   \n",
       "17                   1         0  1.00000  0.271        0       0        0   \n",
       "208                  0         1  1.00000  0.183        0       0        1   \n",
       "219                  0         1  1.00000  0.220        0       0        1   \n",
       "299                  1         0  1.00000  0.366        0       0        0   \n",
       "382                  1         0  0.99989  0.238        0       0        0   \n",
       "465                  1         0  1.00000  0.168        0       1        1   \n",
       "501                  1         0  1.00000  0.205        1       1        1   \n",
       "526                  1         0  0.99993  0.324        0       0        0   \n",
       "580                  1         0  1.00000  0.100        0       1        1   \n",
       "596                  1         0  1.00000  0.083        0       1        1   \n",
       "647                  0         1  0.99994  0.379        0       0        0   \n",
       "838                  0         1  1.00000  0.309        1       1        1   \n",
       "857                  0         1  1.00000  0.251        0       1        1   \n",
       "870                  1         0  1.00000  0.172        1       1        1   \n",
       "890                  0         1  0.92771  0.555        1       1        1   \n",
       "896                  1         0  1.00000  0.163        0       1        1   \n",
       "948                  1         0  1.00000  0.250        1       1        1   \n",
       "995                  0         1  0.98587  0.481        0       0        0   \n",
       "1021                 1         0  0.99994  0.418        0       0        0   \n",
       "1044                 1         0  1.00000  0.210        0       0        0   \n",
       "1087                 0         1  1.00000  0.141        1       1        1   \n",
       "1100                 0         1  1.00000  0.174        0       0        0   \n",
       "1136                 1         0  1.00000  0.321        0       0        0   \n",
       "1203                 1         0  0.99996  0.195        0       0        0   \n",
       "1283                 0         1  1.00000  0.269        0       0        0   \n",
       "1324                 1         0  0.99986  0.118        0       0        0   \n",
       "1325                 0         1  0.99986  0.118        0       0        0   \n",
       "\n",
       "      DEV_BR_GENE_10  samples_all  \n",
       "0                  1  REACH000089  \n",
       "17                 0  REACH000718  \n",
       "208                1  REACH000741  \n",
       "219                1  REACH000774  \n",
       "299                1  REACH000097  \n",
       "382                1  REACH000605  \n",
       "465                1  REACH000529  \n",
       "501                1  REACH000564  \n",
       "526                1  REACH000404  \n",
       "580                1  REACH000578  \n",
       "596                1  REACH000482  \n",
       "647                1  REACH000423  \n",
       "838                0  REACH000747  \n",
       "857                1  REACH000457  \n",
       "870                1  REACH000435  \n",
       "890                1  REACH000519  \n",
       "896                1  REACH000504  \n",
       "948                0  REACH000189  \n",
       "995                0  REACH000482  \n",
       "1021               1  REACH000544  \n",
       "1044               1  REACH000589  \n",
       "1087               0  REACH000664  \n",
       "1100               0  REACH000754  \n",
       "1136               0  REACH000249  \n",
       "1203               0  REACH000349  \n",
       "1283               1  REACH000708  \n",
       "1324               1  REACH000479  \n",
       "1325               1  REACH000555  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this_df_dnL:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(16, 9)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this_df_dnM:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(12, 9)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "inhLOF SNV/indels\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(153, 21)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " fdr asd: 13\n",
      " fdr dd: 23\n",
      " fdr ndd: 29\n",
      " fdr brain genes: 56\n",
      "df_inh_snv:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start_hg38</th>\n",
       "      <th>end_hg38</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>alternate_allele</th>\n",
       "      <th>consequence</th>\n",
       "      <th>gene</th>\n",
       "      <th>gnomad_af</th>\n",
       "      <th>loeuf</th>\n",
       "      <th>cohort</th>\n",
       "      <th>...</th>\n",
       "      <th>cases</th>\n",
       "      <th>controls</th>\n",
       "      <th>fathers</th>\n",
       "      <th>mothers</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2029751</td>\n",
       "      <td>2029752</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>GABRD</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.245</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000101</td>\n",
       "      <td></td>\n",
       "      <td>REACH000103</td>\n",
       "      <td></td>\n",
       "      <td>0.001654</td>\n",
       "      <td>0.753</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21598040</td>\n",
       "      <td>21598041</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>RAP1GAP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.349</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000141</td>\n",
       "      <td></td>\n",
       "      <td>REACH000143</td>\n",
       "      <td></td>\n",
       "      <td>0.380850</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29291893</td>\n",
       "      <td>29291895</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>PTPRU</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.338</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000293</td>\n",
       "      <td></td>\n",
       "      <td>REACH000295</td>\n",
       "      <td></td>\n",
       "      <td>0.025440</td>\n",
       "      <td>0.518</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39412333</td>\n",
       "      <td>39412336</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>MACF1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.084</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000544,REACH000547</td>\n",
       "      <td></td>\n",
       "      <td>REACH000546</td>\n",
       "      <td></td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39571636</td>\n",
       "      <td>39571637</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>PABPC4</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.232</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>REACH000760</td>\n",
       "      <td>REACH000759</td>\n",
       "      <td></td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10606</th>\n",
       "      <td>chr9</td>\n",
       "      <td>136840512</td>\n",
       "      <td>136840513</td>\n",
       "      <td>G</td>\n",
       "      <td>GGAGCTCTAGGC</td>\n",
       "      <td>stop_gained&amp;frameshift_variant</td>\n",
       "      <td>RABL6</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.245</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000145</td>\n",
       "      <td></td>\n",
       "      <td>REACH000148</td>\n",
       "      <td></td>\n",
       "      <td>0.359240</td>\n",
       "      <td>0.570</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10615</th>\n",
       "      <td>chr9</td>\n",
       "      <td>136923873</td>\n",
       "      <td>136923874</td>\n",
       "      <td>C</td>\n",
       "      <td>CTA</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>TRAF2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.123</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>REACH000166,REACH000168</td>\n",
       "      <td></td>\n",
       "      <td>REACH000164</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.215</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10745</th>\n",
       "      <td>chrX</td>\n",
       "      <td>48522252</td>\n",
       "      <td>48522253</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>EBP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.342</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000586</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>REACH000587</td>\n",
       "      <td>0.936450</td>\n",
       "      <td>0.342</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10774</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67569002</td>\n",
       "      <td>67569003</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>AR</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.291</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000387</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>REACH000388</td>\n",
       "      <td>0.988370</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10819</th>\n",
       "      <td>chrX</td>\n",
       "      <td>100820619</td>\n",
       "      <td>100820620</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>CSTF2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.133</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000252</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>REACH000253</td>\n",
       "      <td>0.999780</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom  start_hg38   end_hg38 reference_allele alternate_allele  \\\n",
       "2      chr1     2029751    2029752                C                T   \n",
       "140    chr1    21598040   21598041                C                T   \n",
       "189    chr1    29291893   29291895               AC                A   \n",
       "275    chr1    39412333   39412336              CTG                C   \n",
       "281    chr1    39571636   39571637                C                T   \n",
       "...     ...         ...        ...              ...              ...   \n",
       "10606  chr9   136840512  136840513                G     GGAGCTCTAGGC   \n",
       "10615  chr9   136923873  136923874                C              CTA   \n",
       "10745  chrX    48522252   48522253                G                A   \n",
       "10774  chrX    67569002   67569003                A                G   \n",
       "10819  chrX   100820619  100820620                G                T   \n",
       "\n",
       "                          consequence     gene  gnomad_af  loeuf cohort  ...  \\\n",
       "2                         stop_gained    GABRD   0.000008  0.245  REACH  ...   \n",
       "140           splice_acceptor_variant  RAP1GAP   0.000000  0.349  REACH  ...   \n",
       "189                frameshift_variant    PTPRU   0.000012  0.338  REACH  ...   \n",
       "275                frameshift_variant    MACF1   0.000000  0.084  REACH  ...   \n",
       "281                       stop_gained   PABPC4   0.000008  0.232  REACH  ...   \n",
       "...                               ...      ...        ...    ...    ...  ...   \n",
       "10606  stop_gained&frameshift_variant    RABL6   0.000054  0.245  REACH  ...   \n",
       "10615              frameshift_variant    TRAF2   0.000000  0.123  REACH  ...   \n",
       "10745            splice_donor_variant      EBP   0.000000  0.342  REACH  ...   \n",
       "10774                      start_lost       AR   0.000017  0.291  REACH  ...   \n",
       "10819                      start_lost    CSTF2   0.000000  0.133  REACH  ...   \n",
       "\n",
       "                         cases                 controls      fathers  \\\n",
       "2                  REACH000101                           REACH000103   \n",
       "140                REACH000141                           REACH000143   \n",
       "189                REACH000293                           REACH000295   \n",
       "275    REACH000544,REACH000547                           REACH000546   \n",
       "281                                         REACH000760  REACH000759   \n",
       "...                        ...                      ...          ...   \n",
       "10606              REACH000145                           REACH000148   \n",
       "10615                           REACH000166,REACH000168                \n",
       "10745              REACH000586                                         \n",
       "10774              REACH000387                                         \n",
       "10819              REACH000252                                         \n",
       "\n",
       "           mothers       PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  DEV_BR_GENE_10  \n",
       "2                   0.001654  0.753        0       0        0               0  \n",
       "140                 0.380850  0.549        0       0        0               1  \n",
       "189                 0.025440  0.518        0       0        0               0  \n",
       "275                 1.000000  0.126        0       0        0               1  \n",
       "281                 1.000000  0.295        0       0        0               1  \n",
       "...            ...       ...    ...      ...     ...      ...             ...  \n",
       "10606               0.359240  0.570        0       0        0               0  \n",
       "10615  REACH000164  1.000000  0.215        1       0        1               0  \n",
       "10745  REACH000587  0.936450  0.342        0       0        0               1  \n",
       "10774  REACH000388  0.988370  0.291        0       0        0               0  \n",
       "10819  REACH000253  0.999780  0.124        0       0        0               0  \n",
       "\n",
       "[153 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this_df_inh:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(125, 7)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "      <th>samples_all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000544,REACH000547,REACH000546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000760,REACH000759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.131</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000650,REACH000652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.332</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000372,REACH000430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.151</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000668,REACH000670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10597</th>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.598</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000668,REACH000669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10615</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.215</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000166,REACH000168,REACH000164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10745</th>\n",
       "      <td>0.936450</td>\n",
       "      <td>0.342</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000586,REACH000587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10774</th>\n",
       "      <td>0.988370</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000387,REACH000388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10819</th>\n",
       "      <td>0.999780</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000252,REACH000253</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>125 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  DEV_BR_GENE_10  \\\n",
       "275    1.000000  0.126        0       0        0               1   \n",
       "281    1.000000  0.295        0       0        0               1   \n",
       "297    1.000000  0.131        0       1        1               1   \n",
       "589    1.000000  0.332        0       0        0               0   \n",
       "688    1.000000  0.151        1       1        1               1   \n",
       "...         ...    ...      ...     ...      ...             ...   \n",
       "10597  0.000007  0.598        0       1        1               0   \n",
       "10615  1.000000  0.215        1       0        1               0   \n",
       "10745  0.936450  0.342        0       0        0               1   \n",
       "10774  0.988370  0.291        0       0        0               0   \n",
       "10819  0.999780  0.124        0       0        0               0   \n",
       "\n",
       "                               samples_all  \n",
       "275    REACH000544,REACH000547,REACH000546  \n",
       "281                REACH000760,REACH000759  \n",
       "297                REACH000650,REACH000652  \n",
       "589                REACH000372,REACH000430  \n",
       "688                REACH000668,REACH000670  \n",
       "...                                    ...  \n",
       "10597              REACH000668,REACH000669  \n",
       "10615  REACH000166,REACH000168,REACH000164  \n",
       "10745              REACH000586,REACH000587  \n",
       "10774              REACH000387,REACH000388  \n",
       "10819              REACH000252,REACH000253  \n",
       "\n",
       "[125 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged denovo and inheritted one row per sample\n",
      "this_df_dn_inh:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loss_of_function</th>\n",
       "      <th>missense</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "      <th>samples_all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99989</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.168</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.205</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.93645</td>\n",
       "      <td>0.342</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98837</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98837</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99978</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99978</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000253</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>299 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     loss_of_function  missense      PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  \\\n",
       "0                 1.0       0.0  1.00000  0.271        0       0        0   \n",
       "1                 1.0       0.0  1.00000  0.366        0       0        0   \n",
       "2                 1.0       0.0  0.99989  0.238        0       0        0   \n",
       "3                 1.0       0.0  1.00000  0.168        0       1        1   \n",
       "4                 1.0       0.0  1.00000  0.205        1       1        1   \n",
       "..                ...       ...      ...    ...      ...     ...      ...   \n",
       "138               NaN       NaN  0.93645  0.342        0       0        0   \n",
       "139               NaN       NaN  0.98837  0.291        0       0        0   \n",
       "139               NaN       NaN  0.98837  0.291        0       0        0   \n",
       "140               NaN       NaN  0.99978  0.124        0       0        0   \n",
       "140               NaN       NaN  0.99978  0.124        0       0        0   \n",
       "\n",
       "     DEV_BR_GENE_10  samples_all  \n",
       "0                 0  REACH000718  \n",
       "1                 1  REACH000097  \n",
       "2                 1  REACH000605  \n",
       "3                 1  REACH000529  \n",
       "4                 1  REACH000564  \n",
       "..              ...          ...  \n",
       "138               1  REACH000587  \n",
       "139               0  REACH000387  \n",
       "139               0  REACH000388  \n",
       "140               0  REACH000252  \n",
       "140               0  REACH000253  \n",
       "\n",
       "[299 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### previously discovered variants\n",
    "file_fdr = '/expanse/projects/sebat1/miladm/UCSD/resources/ASD_TADA_2022/fdr_pvals_tada.tsv'\n",
    "df_fdr = pd.read_table(file_fdr, sep='\\t', header=0, index_col=None)\n",
    "print('df_fdr:')\n",
    "display(df_fdr)\n",
    "asd_genes = df_fdr.loc[df_fdr.FDR_TADA_ASD <= 0.05].gene_gencodeV33.tolist()\n",
    "dd_genes = df_fdr.loc[df_fdr.FDR_TADA_DD <= 0.05].gene_gencodeV33.tolist()\n",
    "ndd_genes = df_fdr.loc[df_fdr.FDR_TADA_NDD <= 0.05].gene_gencodeV33.tolist()\n",
    "print(f'asd: {len(asd_genes)}')\n",
    "#print(f'asd: {(asd_genes)}')\n",
    "print(f'dd: {len(dd_genes)}')\n",
    "print(f'ndd: {len(ndd_genes)}')\n",
    "\n",
    "file_pli = '/expanse/projects/sebat1/miladm/UCSD/resources/gnomAD/v4/Constraint/gene_max_pli.tsv'\n",
    "df_pli = pd.read_table(file_pli, sep='\\t', header=0, index_col=None).set_index('gene')\n",
    "print('df_pli:')\n",
    "display(df_pli)\n",
    "\n",
    "file_lf = '/expanse/projects/sebat1/miladm/UCSD/resources/gnomAD/v4/Constraint/gene_min_loeuf.tsv'\n",
    "df_lf = pd.read_table(file_lf, sep='\\t', header=0, index_col=None).set_index('gene')\n",
    "print('df_lf:')\n",
    "display(df_lf)\n",
    "\n",
    "print('dnLOF and dnMIS SNV/indels')\n",
    "file_dn_snv = '/expanse/projects/sebat1/miladm/UCSD/resources/Antaki_2022/Antaki2022_supplementary_tables_ST5.tsv'\n",
    "df_dn_snv = pd.read_table(file_dn_snv, sep='\\t', header=0, index_col=None, keep_default_na=False)\n",
    "df_dn_snv['PLI'] = df_dn_snv.gene.map(df_pli['lof.pLI'])\n",
    "df_dn_snv['LOEUF'] = df_dn_snv.gene.map(df_lf['lof.oe_ci.upper'])\n",
    "df_dn_snv['FDR_ASD'] = df_dn_snv.gene.map(lambda x: 1 if x in asd_genes else 0)\n",
    "df_dn_snv['FDR_DD'] = df_dn_snv.gene.map(lambda x: 1 if x in dd_genes else 0)\n",
    "df_dn_snv['FDR_NDD'] = df_dn_snv.gene.map(lambda x: 1 if x in ndd_genes else 0)\n",
    "df_dn_snv['DEV_BR_GENE_10'] = df_dn_snv.gene.map(lambda x: 1 if x in br_flt_10 else 0)\n",
    "df_dn_snv = df_dn_snv.loc[(df_dn_snv.iid.str.contains('REACH'))]\n",
    "display(df_dn_snv.shape)\n",
    "print(f\" fdr asd: {df_dn_snv['FDR_ASD'].sum()}\")\n",
    "print(f\" fdr dd: {df_dn_snv['FDR_DD'].sum()}\")\n",
    "print(f\" fdr ndd: {df_dn_snv['FDR_NDD'].sum()}\")\n",
    "print(f\" fdr brain genes: {df_dn_snv['DEV_BR_GENE_10'].sum()}\")\n",
    "print('df_dn_snv:')\n",
    "display(df_dn_snv)\n",
    "this_df_dn = df_dn_snv[['loss_of_function', 'missense', 'PLI', 'LOEUF', 'FDR_ASD', 'FDR_DD', 'FDR_NDD', 'DEV_BR_GENE_10']].copy()\n",
    "this_df_dn['samples_all'] = df_dn_snv['iid']\n",
    "# filter constraint genes\n",
    "this_df_dn = this_df_dn.loc[((this_df_dn.PLI>=0.9) | (this_df_dn.LOEUF<=0.37) | \n",
    "                             (this_df_dn.FDR_ASD==1) | (this_df_dn.FDR_DD==1) | (this_df_dn.FDR_NDD==1))]\n",
    "print('this_df_dn:')\n",
    "display(this_df_dn.shape)\n",
    "display(this_df_dn)\n",
    "\n",
    "this_df_dnL = this_df_dn.loc[this_df_dn.loss_of_function==1]\n",
    "this_df_dnM = this_df_dn.loc[this_df_dn.missense==1]\n",
    "print('this_df_dnL:')\n",
    "display(this_df_dnL.shape)\n",
    "print('this_df_dnM:')\n",
    "display(this_df_dnM.shape)\n",
    "\n",
    "print('inhLOF SNV/indels')\n",
    "file_inh_snv = '/expanse/projects/sebat1/miladm/UCSD/resources/Antaki_2022/Antaki2022_supplementary_tables_ST8.tsv'\n",
    "df_inh_snv = pd.read_table(file_inh_snv, sep='\\t', header=0, index_col=None, keep_default_na=False)\n",
    "df_inh_snv['PLI'] = df_inh_snv.gene.map(df_pli['lof.pLI'])\n",
    "df_inh_snv['LOEUF'] = df_inh_snv.gene.map(df_lf['lof.oe_ci.upper'])\n",
    "df_inh_snv['FDR_ASD'] = df_inh_snv.gene.map(lambda x: 1 if x in asd_genes else 0)\n",
    "df_inh_snv['FDR_DD'] = df_inh_snv.gene.map(lambda x: 1 if x in dd_genes else 0)\n",
    "df_inh_snv['FDR_NDD'] = df_inh_snv.gene.map(lambda x: 1 if x in ndd_genes else 0)\n",
    "df_inh_snv['DEV_BR_GENE_10'] = df_inh_snv.gene.map(lambda x: 1 if x in br_flt_10 else 0)\n",
    "df_inh_snv = df_inh_snv.loc[(df_inh_snv.cases.str.contains('REACH')) | ((df_inh_snv.controls.str.contains('REACH')))]\n",
    "display(df_inh_snv.shape)\n",
    "print(f\" fdr asd: {df_inh_snv['FDR_ASD'].sum()}\")\n",
    "print(f\" fdr dd: {df_inh_snv['FDR_DD'].sum()}\")\n",
    "print(f\" fdr ndd: {df_inh_snv['FDR_NDD'].sum()}\")\n",
    "print(f\" fdr brain genes: {df_inh_snv['DEV_BR_GENE_10'].sum()}\")\n",
    "print('df_inh_snv:')\n",
    "display(df_inh_snv)\n",
    "this_df_inh = df_inh_snv[['PLI', 'LOEUF', 'FDR_ASD', 'FDR_DD', 'FDR_NDD', 'DEV_BR_GENE_10']].copy()\n",
    "this_df_inh['samples_all'] = df_inh_snv.apply(lambda row: ','.join([x for x in [row.cases, row.controls, \n",
    "                                                                         row.fathers, row.mothers] if x != '']), axis=1)\n",
    "# filter constraint genes\n",
    "this_df_inh = this_df_inh.loc[((this_df_inh.PLI>=0.9) | (this_df_inh.LOEUF<=0.37) | \n",
    "                             (this_df_inh.FDR_ASD==1) | (this_df_inh.FDR_DD==1) | (this_df_inh.FDR_NDD==1))]\n",
    "print('this_df_inh:')\n",
    "display(this_df_inh.shape)\n",
    "display(this_df_inh)\n",
    "\n",
    "print('Merged denovo and inheritted one row per sample')\n",
    "#this_df_dn_inh = pd.concat([this_df_dn, this_df_inh], axis=0, ignore_index=True)\n",
    "this_df_dn_inh = pd.concat([this_df_dnL, this_df_inh], axis=0, ignore_index=True)\n",
    "this_df_dn_inh['samples_all'] = this_df_dn_inh.samples_all.str.split(',')\n",
    "this_df_dn_inh = this_df_dn_inh.explode('samples_all')\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "this_df_dn_inh['samples_all'] = this_df_dn_inh.apply(lambda row: rename_samples_dict[row['samples_all']] \n",
    "                                             if row['samples_all'] in rename_samples_dict else row['samples_all'], axis=1)\n",
    "\n",
    "print('this_df_dn_inh:')\n",
    "display(this_df_dn_inh)\n",
    "this_df_dn_inh.to_csv('investigate.tsv', sep='\\t', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "47265884-a2c1-44aa-a18f-61d3d8f821eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_AD3_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_SQ40_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_AD4_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_SQ50_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_AD5_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_SQ60_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_SQ20_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_SQ70_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n",
      "len cols_count_svs: 375\n",
      "len cols_count_trs: 4896\n",
      "sum of all dn/inh: 87\n",
      "C, case, carrier: (30, 5323)\n",
      "C, case, non-carrier: (46, 5323)\n",
      "C, control, carrier: (8, 5323)\n",
      "C, control, non-carrier: (33, 5323)\n",
      "P, case, carrier: (0, 5323)\n",
      "P, case, non-carrier: (4, 5323)\n",
      "P, control, carrier: (30, 5323)\n",
      "P, control, non-carrier: (92, 5323)\n",
      "combined_tables_missing/combined_svs_trs_snvs_prs_pc_prscs_SQ30_SAMPLES_AD2_SAMPLES_DELINS_GW_TRmissingness25_fltHomPol_famGT.csv\n"
     ]
    }
   ],
   "source": [
    "#file_svss = ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES.csv',    'main_df_toR_freq05_LRIL_SQ30_SAMPLES_SQ30_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD3_SAMPLES_AD3_SAMPLES.csv',    'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD3_SAMPLES_SQ20_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_SQ20_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES.csv',    'main_df_toR_freq05_LRIL_SQ40_SAMPLES_SQ40_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD4_SAMPLES.csv',    'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_SQ20_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_SQ20_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES.csv',    'main_df_toR_freq05_LRIL_SQ50_SAMPLES_SQ50_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD5_SAMPLES.csv',    'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_SQ20_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_SQ20_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_SQ60_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_SQ20_SAMPLES.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ70_SAMPLES_SQ20_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_SQ20_SAMPLES.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_SQ70_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD3_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD3_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD3_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD3_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ40_SAMPLES_SQ30_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ50_SAMPLES_SQ30_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ60_SAMPLES_SQ30_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ70_SAMPLES_SQ30_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ40_SAMPLES_SQ40_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ50_SAMPLES_SQ40_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ60_SAMPLES_SQ40_SAMPLES.csv'\n",
    "#,'main_df_toR_freq05_LRIL_SQ70_SAMPLES_SQ40_SAMPLES.csv']\n",
    "\n",
    "#file_svss = ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES_INSDUP.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES_INSDUP.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES_INSDUP.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES_INSDUP.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES_INSDUP.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES_INSDUP.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES_INSDUP.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES_INSDUP.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES_INSDUP.csv']\n",
    "\n",
    "#file_svss = ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES_DEL.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES_DEL.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES_DEL.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES_DEL.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES_DEL.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES_DEL.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES_DEL.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES_DEL.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES_DEL.csv']\n",
    "\n",
    "#file_svss = ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES.csv'] + ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES_HP12.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES_HP12.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES_HP12.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES_HP12.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES_HP12.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES_HP12.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES_HP12.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES_HP12.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES_HP12.csv'] + ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES_NCOM.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES_NCOM.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES_NCOM.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES_NCOM.csv',\n",
    "#'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES_NCOM.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES_NCOM.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES_NCOM.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES_NCOM.csv',\n",
    "#'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES_NCOM.csv']\n",
    "\n",
    "file_svss = ['main_df_toR_freq05_LRIL_AD3_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',   'main_df_toR_freq05_LRIL_SQ40_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',\n",
    "'main_df_toR_freq05_LRIL_AD4_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',   'main_df_toR_freq05_LRIL_SQ50_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',\n",
    "'main_df_toR_freq05_LRIL_AD5_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',   'main_df_toR_freq05_LRIL_SQ60_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',\n",
    "'main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',  'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES_DELINS_GW.csv',\n",
    "'main_df_toR_freq05_LRIL_SQ30_SAMPLES_AD2_SAMPLES_DELINS_GW.csv']\n",
    "\n",
    "#file_svs = 'main_df_toR_freq05_LRIL_SQ70_SAMPLES_AD2_SAMPLES.csv'\n",
    "\n",
    "for file_svs in file_svss:\n",
    "    print('+'*40)\n",
    "    file_suf = file_svs.lstrip('main_df_toR_freq05_LRIL_').rstrip('.csv')\n",
    "    df_svs = pd.read_table(file_svs, sep=',', header=0)\n",
    "    #display(df_svs)\n",
    "    \n",
    "    cols_count_svs = [x for x in list(df_svs.columns) if x.startswith('count_')]\n",
    "    print(f'len cols_count_svs: {len(cols_count_svs)}')\n",
    "    \n",
    "    df_trs = pd.read_table(file_trs, sep=',', header=0)\n",
    "    #display(df_trs)\n",
    "    \n",
    "    cols_count_trs = [x for x in list(df_trs.columns) if x.startswith('count_')]\n",
    "    print(f'len cols_count_trs: {len(cols_count_trs)}')\n",
    "    \n",
    "    #set_int = set(cols_count_svs).intersection(set(cols_count_trs))\n",
    "    #print(f'len set_int: {len(set_int)}')\n",
    "    \n",
    "    df_combined = df_svs.merge(df_trs[['iid']+cols_count_trs])\n",
    "    df_combined['PRS_ASD_Z'] = df_combined.iid.map(df_prs_asd.PRS_ASD_Z)\n",
    "    df_combined['PRS_SCZ_Z'] = df_combined.iid.map(df_prs_scz.PRS_SCZ_Z)\n",
    "    df_combined['PRS_EDU_Z'] = df_combined.iid.map(df_prs_edu.PRS_EDU_Z)\n",
    "    \n",
    "    count_col = []\n",
    "    samples_dn_inh = this_df_dn_inh.samples_all.tolist()\n",
    "    for sample in df_combined.iid:\n",
    "        count_col.append(samples_dn_inh.count(sample))\n",
    "    df_combined['count_dnlof_inhlof'] = count_col\n",
    "    #print('df_combined:')\n",
    "    #display(df_combined)\n",
    "    print(f'sum of all dn/inh: {df_combined.count_dnlof_inhlof.sum()}')\n",
    "    print(f'C, case, carrier: {df_combined.loc[(df_combined.rel==\"C\")&(df_combined.Affected==\"Yes\")&(df_combined.count_dnlof_inhlof>0)].shape}')\n",
    "    print(f'C, case, non-carrier: {df_combined.loc[(df_combined.rel==\"C\")&(df_combined.Affected==\"Yes\")&(df_combined.count_dnlof_inhlof==0)].shape}')\n",
    "    print(f'C, control, carrier: {df_combined.loc[(df_combined.rel==\"C\")&(df_combined.Affected==\"No\")&(df_combined.count_dnlof_inhlof>0)].shape}')\n",
    "    print(f'C, control, non-carrier: {df_combined.loc[(df_combined.rel==\"C\")&(df_combined.Affected==\"No\")&(df_combined.count_dnlof_inhlof==0)].shape}')\n",
    "    print(f'P, case, carrier: {df_combined.loc[(df_combined.rel==\"P\")&(df_combined.Affected==\"Yes\")&(df_combined.count_dnlof_inhlof>0)].shape}')\n",
    "    print(f'P, case, non-carrier: {df_combined.loc[(df_combined.rel==\"P\")&(df_combined.Affected==\"Yes\")&(df_combined.count_dnlof_inhlof==0)].shape}')\n",
    "    print(f'P, control, carrier: {df_combined.loc[(df_combined.rel==\"P\")&(df_combined.Affected==\"No\")&(df_combined.count_dnlof_inhlof>0)].shape}')\n",
    "    print(f'P, control, non-carrier: {df_combined.loc[(df_combined.rel==\"P\")&(df_combined.Affected==\"No\")&(df_combined.count_dnlof_inhlof==0)].shape}')\n",
    "    \n",
    "    if (input == 'sbayesr'):\n",
    "        file_out = dir_out + 'combined_svs_trs_snvs_prs_pc_sbayesr_' + file_suf + '_' + 'TRmissingness' + tr_miss_level + '_fltHomPol_famGT.csv'\n",
    "    elif (input == 'prscs'):\n",
    "        file_out = dir_out + 'combined_svs_trs_snvs_prs_pc_prscs_' + file_suf + '_' + 'TRmissingness' + tr_miss_level + '_fltHomPol_famGT.csv'\n",
    "    print(file_out)\n",
    "    df_combined.to_csv(file_out, sep=',', header=True, index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
